{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyN65uofOnq57eOtXXFX34Y3",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/allyssonallan/LabLuz_bioinfo_class_2020/blob/main/Aula_28_11_Bioinformatica_2.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "0pgbgbQOZCOq",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "95a2701c-ea59-4088-8a20-df6d6d59ca2f"
      },
      "source": [
        "!which python\n",
        "!python --version\n",
        "!echo $PYTHONPATH\n",
        "%env PYTHONPATH=\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/usr/local/bin/python\n",
            "Python 3.6.9\n",
            "/env/python\n",
            "env: PYTHONPATH=\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "IXTWosjCZxb6",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ad36fd29-4519-428d-cd1f-8f3a58ba7a62"
      },
      "source": [
        "%%bash\n",
        "MINICONDA_INSTALLER_SCRIPT=Miniconda3-4.5.4-Linux-x86_64.sh\n",
        "MINICONDA_PREFIX=/usr/local\n",
        "wget https://repo.continuum.io/miniconda/$MINICONDA_INSTALLER_SCRIPT\n",
        "chmod +x $MINICONDA_INSTALLER_SCRIPT\n",
        "./$MINICONDA_INSTALLER_SCRIPT -b -f -p $MINICONDA_PREFIX"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "PREFIX=/usr/local\n",
            "installing: python-3.6.5-hc3d631a_2 ...\n",
            "installing: ca-certificates-2018.03.07-0 ...\n",
            "installing: conda-env-2.6.0-h36134e3_1 ...\n",
            "installing: libgcc-ng-7.2.0-hdf63c60_3 ...\n",
            "installing: libstdcxx-ng-7.2.0-hdf63c60_3 ...\n",
            "installing: libffi-3.2.1-hd88cf55_4 ...\n",
            "installing: ncurses-6.1-hf484d3e_0 ...\n",
            "installing: openssl-1.0.2o-h20670df_0 ...\n",
            "installing: tk-8.6.7-hc745277_3 ...\n",
            "installing: xz-5.2.4-h14c3975_4 ...\n",
            "installing: yaml-0.1.7-had09818_2 ...\n",
            "installing: zlib-1.2.11-ha838bed_2 ...\n",
            "installing: libedit-3.1.20170329-h6b74fdf_2 ...\n",
            "installing: readline-7.0-ha6073c6_4 ...\n",
            "installing: sqlite-3.23.1-he433501_0 ...\n",
            "installing: asn1crypto-0.24.0-py36_0 ...\n",
            "installing: certifi-2018.4.16-py36_0 ...\n",
            "installing: chardet-3.0.4-py36h0f667ec_1 ...\n",
            "installing: idna-2.6-py36h82fb2a8_1 ...\n",
            "installing: pycosat-0.6.3-py36h0a5515d_0 ...\n",
            "installing: pycparser-2.18-py36hf9f622e_1 ...\n",
            "installing: pysocks-1.6.8-py36_0 ...\n",
            "installing: ruamel_yaml-0.15.37-py36h14c3975_2 ...\n",
            "installing: six-1.11.0-py36h372c433_1 ...\n",
            "installing: cffi-1.11.5-py36h9745a5d_0 ...\n",
            "installing: setuptools-39.2.0-py36_0 ...\n",
            "installing: cryptography-2.2.2-py36h14c3975_0 ...\n",
            "installing: wheel-0.31.1-py36_0 ...\n",
            "installing: pip-10.0.1-py36_0 ...\n",
            "installing: pyopenssl-18.0.0-py36_0 ...\n",
            "installing: urllib3-1.22-py36hbe7ace6_0 ...\n",
            "installing: requests-2.18.4-py36he2e5f8d_1 ...\n",
            "installing: conda-4.5.4-py36_0 ...\n",
            "installation finished.\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "stream",
          "text": [
            "--2020-11-28 17:39:13--  https://repo.continuum.io/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh\n",
            "Resolving repo.continuum.io (repo.continuum.io)... 104.18.200.79, 104.18.201.79, 2606:4700::6812:c84f, ...\n",
            "Connecting to repo.continuum.io (repo.continuum.io)|104.18.200.79|:443... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://repo.anaconda.com/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh [following]\n",
            "--2020-11-28 17:39:13--  https://repo.anaconda.com/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh\n",
            "Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.130.3, 104.16.131.3, 2606:4700::6810:8203, ...\n",
            "Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.130.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 58468498 (56M) [application/x-sh]\n",
            "Saving to: ‘Miniconda3-4.5.4-Linux-x86_64.sh’\n",
            "\n",
            "     0K .......... .......... .......... .......... ..........  0% 57.7M 1s\n",
            "    50K .......... .......... .......... .......... ..........  0% 3.95M 8s\n",
            "   100K .......... .......... .......... .......... ..........  0% 4.08M 10s\n",
            "   150K .......... .......... .......... .......... ..........  0% 4.01M 11s\n",
            "   200K .......... .......... .......... .......... ..........  0% 96.0M 9s\n",
            "   250K .......... .......... .......... .......... ..........  0%  101M 7s\n",
            "   300K .......... .......... .......... .......... ..........  0% 4.46M 8s\n",
            "   350K .......... .......... .......... .......... ..........  0%  127M 7s\n",
            "   400K .......... .......... .......... .......... ..........  0% 65.1M 6s\n",
            "   450K .......... .......... .......... .......... ..........  0%  178M 6s\n",
            "   500K .......... .......... .......... .......... ..........  0% 70.3M 5s\n",
            "   550K .......... .......... .......... .......... ..........  1% 69.9M 5s\n",
            "   600K .......... .......... .......... .......... ..........  1% 63.9M 5s\n",
            "   650K .......... .......... .......... .......... ..........  1%  185M 4s\n",
            "   700K .......... .......... .......... .......... ..........  1% 26.1M 4s\n",
            "   750K .......... .......... .......... .......... ..........  1% 7.45M 4s\n",
            "   800K .......... .......... .......... .......... ..........  1% 71.3M 4s\n",
            "   850K .......... .......... .......... .......... ..........  1%  164M 4s\n",
            "   900K .......... .......... .......... .......... ..........  1% 44.6M 4s\n",
            "   950K .......... .......... .......... .......... ..........  1%  156M 4s\n",
            "  1000K .......... .......... .......... .......... ..........  1% 59.2M 3s\n",
            "  1050K .......... .......... .......... .......... ..........  1%  123M 3s\n",
            "  1100K .......... .......... .......... .......... ..........  2% 56.4M 3s\n",
            "  1150K .......... .......... .......... .......... ..........  2% 65.6M 3s\n",
            "  1200K .......... .......... .......... .......... ..........  2% 64.2M 3s\n",
            "  1250K .......... .......... .......... .......... ..........  2%  196M 3s\n",
            "  1300K .......... .......... .......... .......... ..........  2% 31.6M 3s\n",
            "  1350K .......... .......... .......... .......... ..........  2%  155M 3s\n",
            "  1400K .......... .......... .......... .......... ..........  2%  188M 3s\n",
            "  1450K .......... .......... .......... .......... ..........  2% 80.8M 3s\n",
            "  1500K .......... .......... .......... .......... ..........  2% 69.3M 3s\n",
            "  1550K .......... .......... .......... .......... ..........  2% 63.8M 2s\n",
            "  1600K .......... .......... .......... .......... ..........  2% 37.1M 2s\n",
            "  1650K .......... .......... .......... .......... ..........  2%  179M 2s\n",
            "  1700K .......... .......... .......... .......... ..........  3%  192M 2s\n",
            "  1750K .......... .......... .......... .......... ..........  3% 35.3M 2s\n",
            "  1800K .......... .......... .......... .......... ..........  3%  151M 2s\n",
            "  1850K .......... .......... .......... .......... ..........  3%  130M 2s\n",
            "  1900K .......... .......... .......... .......... ..........  3%  105M 2s\n",
            "  1950K .......... .......... .......... .......... ..........  3% 59.4M 2s\n",
            "  2000K .......... .......... .......... .......... ..........  3%  182M 2s\n",
            "  2050K .......... .......... .......... .......... ..........  3%  121M 2s\n",
            "  2100K .......... .......... .......... .......... ..........  3%  116M 2s\n",
            "  2150K .......... .......... .......... .......... ..........  3%  148M 2s\n",
            "  2200K .......... .......... .......... .......... ..........  3% 76.2M 2s\n",
            "  2250K .......... .......... .......... .......... ..........  4%  176M 2s\n",
            "  2300K .......... .......... .......... .......... ..........  4%  183M 2s\n",
            "  2350K .......... .......... .......... .......... ..........  4%  108M 2s\n",
            "  2400K .......... .......... .......... .......... ..........  4% 75.3M 2s\n",
            "  2450K .......... .......... .......... .......... ..........  4%  173M 2s\n",
            "  2500K .......... .......... .......... .......... ..........  4%  136M 2s\n",
            "  2550K .......... .......... .......... .......... ..........  4%  133M 2s\n",
            "  2600K .......... .......... .......... .......... ..........  4% 93.4M 2s\n",
            "  2650K .......... .......... .......... .......... ..........  4% 30.8M 2s\n",
            "  2700K .......... .......... .......... .......... ..........  4%  136M 2s\n",
            "  2750K .......... .......... .......... .......... ..........  4%  122M 2s\n",
            "  2800K .......... .......... .......... .......... ..........  4%  152M 2s\n",
            "  2850K .......... .......... .......... .......... ..........  5%  138M 2s\n",
            "  2900K .......... .......... .......... .......... ..........  5%  109M 2s\n",
            "  2950K .......... .......... .......... .......... ..........  5%  153M 2s\n",
            "  3000K .......... .......... .......... .......... ..........  5%  199M 2s\n",
            "  3050K .......... .......... .......... .......... ..........  5% 90.2M 2s\n",
            "  3100K .......... .......... .......... .......... ..........  5%  173M 1s\n",
            "  3150K .......... .......... .......... .......... ..........  5%  167M 1s\n",
            "  3200K .......... .......... .......... .......... ..........  5%  147M 1s\n",
            "  3250K .......... .......... .......... .......... ..........  5%  160M 1s\n",
            "  3300K .......... .......... .......... .......... ..........  5%  165M 1s\n",
            "  3350K .......... .......... .......... .......... ..........  5% 36.5M 1s\n",
            "  3400K .......... .......... .......... .......... ..........  6%  170M 1s\n",
            "  3450K .......... .......... .......... .......... ..........  6%  172M 1s\n",
            "  3500K .......... .......... .......... .......... ..........  6%  178M 1s\n",
            "  3550K .......... .......... .......... .......... ..........  6%  164M 1s\n",
            "  3600K .......... .......... .......... .......... ..........  6%  171M 1s\n",
            "  3650K .......... .......... .......... .......... ..........  6%  178M 1s\n",
            "  3700K .......... .......... .......... .......... ..........  6%  180M 1s\n",
            "  3750K .......... .......... .......... .......... ..........  6%  134M 1s\n",
            "  3800K .......... .......... .......... .......... ..........  6%  193M 1s\n",
            "  3850K .......... .......... .......... .......... ..........  6%  179M 1s\n",
            "  3900K .......... .......... .......... .......... ..........  6%  171M 1s\n",
            "  3950K .......... .......... .......... .......... ..........  7%  126M 1s\n",
            "  4000K .......... .......... .......... .......... ..........  7%  199M 1s\n",
            "  4050K .......... .......... .......... .......... ..........  7%  192M 1s\n",
            "  4100K .......... .......... .......... .......... ..........  7%  182M 1s\n",
            "  4150K .......... .......... .......... .......... ..........  7%  167M 1s\n",
            "  4200K .......... .......... .......... .......... ..........  7%  192M 1s\n",
            "  4250K .......... .......... .......... .......... ..........  7%  189M 1s\n",
            "  4300K .......... .......... .......... .......... ..........  7%  182M 1s\n",
            "  4350K .......... .......... .......... .......... ..........  7%  154M 1s\n",
            "  4400K .......... .......... .......... .......... ..........  7%  113M 1s\n",
            "  4450K .......... .......... .......... .......... ..........  7%  145M 1s\n",
            "  4500K .......... .......... .......... .......... ..........  7%  196M 1s\n",
            "  4550K .......... .......... .......... .......... ..........  8%  154M 1s\n",
            "  4600K .......... .......... .......... .......... ..........  8%  181M 1s\n",
            "  4650K .......... .......... .......... .......... ..........  8%  161M 1s\n",
            "  4700K .......... .......... .......... .......... ..........  8%  163M 1s\n",
            "  4750K .......... .......... .......... .......... ..........  8%  157M 1s\n",
            "  4800K .......... .......... .......... .......... ..........  8%  146M 1s\n",
            "  4850K .......... .......... .......... .......... ..........  8%  161M 1s\n",
            "  4900K .......... .......... .......... .......... ..........  8%  172M 1s\n",
            "  4950K .......... .......... .......... .......... ..........  8%  146M 1s\n",
            "  5000K .......... .......... .......... .......... ..........  8%  157M 1s\n",
            "  5050K .......... .......... .......... .......... ..........  8%  185M 1s\n",
            "  5100K .......... .......... .......... .......... ..........  9%  172M 1s\n",
            "  5150K .......... .......... .......... .......... ..........  9%  164M 1s\n",
            "  5200K .......... .......... .......... .......... ..........  9%  196M 1s\n",
            "  5250K .......... .......... .......... .......... ..........  9%  183M 1s\n",
            "  5300K .......... .......... .......... .......... ..........  9%  173M 1s\n",
            "  5350K .......... .......... .......... .......... ..........  9%  129M 1s\n",
            "  5400K .......... .......... .......... .......... ..........  9% 41.0M 1s\n",
            "  5450K .......... .......... .......... .......... ..........  9%  171M 1s\n",
            "  5500K .......... .......... .......... .......... ..........  9%  167M 1s\n",
            "  5550K .......... .......... .......... .......... ..........  9%  142M 1s\n",
            "  5600K .......... .......... .......... .......... ..........  9%  177M 1s\n",
            "  5650K .......... .......... .......... .......... ..........  9%  176M 1s\n",
            "  5700K .......... .......... .......... .......... .......... 10%  221M 1s\n",
            "  5750K .......... .......... .......... .......... .......... 10%  141M 1s\n",
            "  5800K .......... .......... .......... .......... .......... 10%  154M 1s\n",
            "  5850K .......... .......... .......... .......... .......... 10%  181M 1s\n",
            "  5900K .......... .......... .......... .......... .......... 10%  229M 1s\n",
            "  5950K .......... .......... .......... .......... .......... 10%  289M 1s\n",
            "  6000K .......... .......... .......... .......... .......... 10%  328M 1s\n",
            "  6050K .......... .......... .......... .......... .......... 10%  266M 1s\n",
            "  6100K .......... .......... .......... .......... .......... 10%  303M 1s\n",
            "  6150K .......... .......... .......... .......... .......... 10%  270M 1s\n",
            "  6200K .......... .......... .......... .......... .......... 10%  334M 1s\n",
            "  6250K .......... .......... .......... .......... .......... 11%  333M 1s\n",
            "  6300K .......... .......... .......... .......... .......... 11%  322M 1s\n",
            "  6350K .......... .......... .......... .......... .......... 11%  228M 1s\n",
            "  6400K .......... .......... .......... .......... .......... 11%  196M 1s\n",
            "  6450K .......... .......... .......... .......... .......... 11%  197M 1s\n",
            "  6500K .......... .......... .......... .......... .......... 11%  199M 1s\n",
            "  6550K .......... .......... .......... .......... .......... 11%  153M 1s\n",
            "  6600K .......... .......... .......... .......... .......... 11%  195M 1s\n",
            "  6650K .......... .......... .......... .......... .......... 11%  193M 1s\n",
            "  6700K .......... .......... .......... .......... .......... 11%  199M 1s\n",
            "  6750K .......... .......... .......... .......... .......... 11%  153M 1s\n",
            "  6800K .......... .......... .......... .......... .......... 11%  159M 1s\n",
            "  6850K .......... .......... .......... .......... .......... 12%  165M 1s\n",
            "  6900K .......... .......... .......... .......... .......... 12%  160M 1s\n",
            "  6950K .......... .......... .......... .......... .......... 12%  147M 1s\n",
            "  7000K .......... .......... .......... .......... .......... 12%  178M 1s\n",
            "  7050K .......... .......... .......... .......... .......... 12%  188M 1s\n",
            "  7100K .......... .......... .......... .......... .......... 12%  163M 1s\n",
            "  7150K .......... .......... .......... .......... .......... 12%  159M 1s\n",
            "  7200K .......... .......... .......... .......... .......... 12%  168M 1s\n",
            "  7250K .......... .......... .......... .......... .......... 12%  171M 1s\n",
            "  7300K .......... .......... .......... .......... .......... 12%  189M 1s\n",
            "  7350K .......... .......... .......... .......... .......... 12%  162M 1s\n",
            "  7400K .......... .......... .......... .......... .......... 13%  183M 1s\n",
            "  7450K .......... .......... .......... .......... .......... 13% 38.6M 1s\n",
            "  7500K .......... .......... .......... .......... .......... 13%  197M 1s\n",
            "  7550K .......... .......... .......... .......... .......... 13%  162M 1s\n",
            "  7600K .......... .......... .......... .......... .......... 13%  185M 1s\n",
            "  7650K .......... .......... .......... .......... .......... 13%  164M 1s\n",
            "  7700K .......... .......... .......... .......... .......... 13%  163M 1s\n",
            "  7750K .......... .......... .......... .......... .......... 13%  155M 1s\n",
            "  7800K .......... .......... .......... .......... .......... 13%  179M 1s\n",
            "  7850K .......... .......... .......... .......... .......... 13%  195M 1s\n",
            "  7900K .......... .......... .......... .......... .......... 13%  197M 1s\n",
            "  7950K .......... .......... .......... .......... .......... 14%  168M 1s\n",
            "  8000K .......... .......... .......... .......... .......... 14%  166M 1s\n",
            "  8050K .......... .......... .......... .......... .......... 14%  172M 1s\n",
            "  8100K .......... .......... .......... .......... .......... 14%  193M 1s\n",
            "  8150K .......... .......... .......... .......... .......... 14%  151M 1s\n",
            "  8200K .......... .......... .......... .......... .......... 14%  199M 1s\n",
            "  8250K .......... .......... .......... .......... .......... 14%  194M 1s\n",
            "  8300K .......... .......... .......... .......... .......... 14%  192M 1s\n",
            "  8350K .......... .......... .......... .......... .......... 14%  164M 1s\n",
            "  8400K .......... .......... .......... .......... .......... 14%  195M 1s\n",
            "  8450K .......... .......... .......... .......... .......... 14%  193M 1s\n",
            "  8500K .......... .......... .......... .......... .......... 14%  198M 1s\n",
            "  8550K .......... .......... .......... .......... .......... 15%  154M 1s\n",
            "  8600K .......... .......... .......... .......... .......... 15%  195M 1s\n",
            "  8650K .......... .......... .......... .......... .......... 15%  194M 1s\n",
            "  8700K .......... .......... .......... .......... .......... 15%  172M 1s\n",
            "  8750K .......... .......... .......... .......... .......... 15%  133M 1s\n",
            "  8800K .......... .......... .......... .......... .......... 15%  166M 1s\n",
            "  8850K .......... .......... .......... .......... .......... 15%  145M 1s\n",
            "  8900K .......... .......... .......... .......... .......... 15%  164M 1s\n",
            "  8950K .......... .......... .......... .......... .......... 15%  149M 1s\n",
            "  9000K .......... .......... .......... .......... .......... 15%  193M 1s\n",
            "  9050K .......... .......... .......... .......... .......... 15%  163M 1s\n",
            "  9100K .......... .......... .......... .......... .......... 16%  182M 1s\n",
            "  9150K .......... .......... .......... .......... .......... 16%  169M 1s\n",
            "  9200K .......... .......... .......... .......... .......... 16%  181M 1s\n",
            "  9250K .......... .......... .......... .......... .......... 16%  191M 1s\n",
            "  9300K .......... .......... .......... .......... .......... 16%  193M 1s\n",
            "  9350K .......... .......... .......... .......... .......... 16%  158M 1s\n",
            "  9400K .......... .......... .......... .......... .......... 16%  179M 1s\n",
            "  9450K .......... .......... .......... .......... .......... 16%  157M 1s\n",
            "  9500K .......... .......... .......... .......... .......... 16% 34.3M 1s\n",
            "  9550K .......... .......... .......... .......... .......... 16%  160M 1s\n",
            "  9600K .......... .......... .......... .......... .......... 16%  191M 1s\n",
            "  9650K .......... .......... .......... .......... .......... 16%  179M 1s\n",
            "  9700K .......... .......... .......... .......... .......... 17%  178M 1s\n",
            "  9750K .......... .......... .......... .......... .......... 17%  161M 1s\n",
            "  9800K .......... .......... .......... .......... .......... 17%  198M 1s\n",
            "  9850K .......... .......... .......... .......... .......... 17%  198M 1s\n",
            "  9900K .......... .......... .......... .......... .......... 17%  184M 1s\n",
            "  9950K .......... .......... .......... .......... .......... 17%  161M 1s\n",
            " 10000K .......... .......... .......... .......... .......... 17%  160M 1s\n",
            " 10050K .......... .......... .......... .......... .......... 17%  194M 1s\n",
            " 10100K .......... .......... .......... .......... .......... 17%  179M 1s\n",
            " 10150K .......... .......... .......... .......... .......... 17%  163M 1s\n",
            " 10200K .......... .......... .......... .......... .......... 17%  196M 1s\n",
            " 10250K .......... .......... .......... .......... .......... 18%  182M 1s\n",
            " 10300K .......... .......... .......... .......... .......... 18%  192M 1s\n",
            " 10350K .......... .......... .......... .......... .......... 18%  173M 1s\n",
            " 10400K .......... .......... .......... .......... .......... 18%  180M 1s\n",
            " 10450K .......... .......... .......... .......... .......... 18%  178M 1s\n",
            " 10500K .......... .......... .......... .......... .......... 18%  194M 1s\n",
            " 10550K .......... .......... .......... .......... .......... 18%  160M 1s\n",
            " 10600K .......... .......... .......... .......... .......... 18%  190M 1s\n",
            " 10650K .......... .......... .......... .......... .......... 18%  151M 1s\n",
            " 10700K .......... .......... .......... .......... .......... 18%  161M 1s\n",
            " 10750K .......... .......... .......... .......... .......... 18%  165M 1s\n",
            " 10800K .......... .......... .......... .......... .......... 19%  177M 1s\n",
            " 10850K .......... .......... .......... .......... .......... 19%  164M 1s\n",
            " 10900K .......... .......... .......... .......... .......... 19%  172M 1s\n",
            " 10950K .......... .......... .......... .......... .......... 19%  152M 1s\n",
            " 11000K .......... .......... .......... .......... .......... 19%  187M 1s\n",
            " 11050K .......... .......... .......... .......... .......... 19%  195M 1s\n",
            " 11100K .......... .......... .......... .......... .......... 19%  189M 1s\n",
            " 11150K .......... .......... .......... .......... .......... 19%  161M 1s\n",
            " 11200K .......... .......... .......... .......... .......... 19%  189M 1s\n",
            " 11250K .......... .......... .......... .......... .......... 19%  196M 1s\n",
            " 11300K .......... .......... .......... .......... .......... 19%  194M 1s\n",
            " 11350K .......... .......... .......... .......... .......... 19%  147M 1s\n",
            " 11400K .......... .......... .......... .......... .......... 20%  198M 1s\n",
            " 11450K .......... .......... .......... .......... .......... 20%  308M 1s\n",
            " 11500K .......... .......... .......... .......... .......... 20%  328M 1s\n",
            " 11550K .......... .......... .......... .......... .......... 20% 37.3M 1s\n",
            " 11600K .......... .......... .......... .......... .......... 20%  187M 1s\n",
            " 11650K .......... .......... .......... .......... .......... 20%  196M 1s\n",
            " 11700K .......... .......... .......... .......... .......... 20%  190M 1s\n",
            " 11750K .......... .......... .......... .......... .......... 20%  152M 1s\n",
            " 11800K .......... .......... .......... .......... .......... 20%  199M 1s\n",
            " 11850K .......... .......... .......... .......... .......... 20%  199M 1s\n",
            " 11900K .......... .......... .......... .......... .......... 20%  198M 1s\n",
            " 11950K .......... .......... .......... .......... .......... 21%  168M 1s\n",
            " 12000K .......... .......... .......... .......... .......... 21%  185M 1s\n",
            " 12050K .......... .......... .......... .......... .......... 21%  196M 1s\n",
            " 12100K .......... .......... .......... .......... .......... 21%  194M 1s\n",
            " 12150K .......... .......... .......... .......... .......... 21%  149M 1s\n",
            " 12200K .......... .......... .......... .......... .......... 21%  161M 1s\n",
            " 12250K .......... .......... .......... .......... .......... 21%  226M 1s\n",
            " 12300K .......... .......... .......... .......... .......... 21%  181M 1s\n",
            " 12350K .......... .......... .......... .......... .......... 21%  154M 1s\n",
            " 12400K .......... .......... .......... .......... .......... 21%  190M 1s\n",
            " 12450K .......... .......... .......... .......... .......... 21%  198M 1s\n",
            " 12500K .......... .......... .......... .......... .......... 21%  178M 1s\n",
            " 12550K .......... .......... .......... .......... .......... 22%  165M 1s\n",
            " 12600K .......... .......... .......... .......... .......... 22%  162M 1s\n",
            " 12650K .......... .......... .......... .......... .......... 22%  175M 1s\n",
            " 12700K .......... .......... .......... .......... .......... 22%  184M 1s\n",
            " 12750K .......... .......... .......... .......... .......... 22%  174M 1s\n",
            " 12800K .......... .......... .......... .......... .......... 22%  184M 1s\n",
            " 12850K .......... .......... .......... .......... .......... 22%  151M 1s\n",
            " 12900K .......... .......... .......... .......... .......... 22%  168M 0s\n",
            " 12950K .......... .......... .......... .......... .......... 22%  139M 0s\n",
            " 13000K .......... .......... .......... .......... .......... 22%  165M 0s\n",
            " 13050K .......... .......... .......... .......... .......... 22%  193M 0s\n",
            " 13100K .......... .......... .......... .......... .......... 23%  197M 0s\n",
            " 13150K .......... .......... .......... .......... .......... 23%  175M 0s\n",
            " 13200K .......... .......... .......... .......... .......... 23%  174M 0s\n",
            " 13250K .......... .......... .......... .......... .......... 23%  197M 0s\n",
            " 13300K .......... .......... .......... .......... .......... 23%  192M 0s\n",
            " 13350K .......... .......... .......... .......... .......... 23%  164M 0s\n",
            " 13400K .......... .......... .......... .......... .......... 23%  176M 0s\n",
            " 13450K .......... .......... .......... .......... .......... 23%  161M 0s\n",
            " 13500K .......... .......... .......... .......... .......... 23%  182M 0s\n",
            " 13550K .......... .......... .......... .......... .......... 23%  160M 0s\n",
            " 13600K .......... .......... .......... .......... .......... 23% 41.9M 0s\n",
            " 13650K .......... .......... .......... .......... .......... 23%  195M 0s\n",
            " 13700K .......... .......... .......... .......... .......... 24%  171M 0s\n",
            " 13750K .......... .......... .......... .......... .......... 24%  153M 0s\n",
            " 13800K .......... .......... .......... .......... .......... 24%  197M 0s\n",
            " 13850K .......... .......... .......... .......... .......... 24%  200M 0s\n",
            " 13900K .......... .......... .......... .......... .......... 24%  199M 0s\n",
            " 13950K .......... .......... .......... .......... .......... 24%  163M 0s\n",
            " 14000K .......... .......... .......... .......... .......... 24%  191M 0s\n",
            " 14050K .......... .......... .......... .......... .......... 24%  199M 0s\n",
            " 14100K .......... .......... .......... .......... .......... 24%  203M 0s\n",
            " 14150K .......... .......... .......... .......... .......... 24%  154M 0s\n",
            " 14200K .......... .......... .......... .......... .......... 24%  159M 0s\n",
            " 14250K .......... .......... .......... .......... .......... 25%  168M 0s\n",
            " 14300K .......... .......... .......... .......... .......... 25%  162M 0s\n",
            " 14350K .......... .......... .......... .......... .......... 25%  165M 0s\n",
            " 14400K .......... .......... .......... .......... .......... 25%  198M 0s\n",
            " 14450K .......... .......... .......... .......... .......... 25%  195M 0s\n",
            " 14500K .......... .......... .......... .......... .......... 25%  185M 0s\n",
            " 14550K .......... .......... .......... .......... .......... 25%  156M 0s\n",
            " 14600K .......... .......... .......... .......... .......... 25%  163M 0s\n",
            " 14650K .......... .......... .......... .......... .......... 25%  183M 0s\n",
            " 14700K .......... .......... .......... .......... .......... 25%  190M 0s\n",
            " 14750K .......... .......... .......... .......... .......... 25%  180M 0s\n",
            " 14800K .......... .......... .......... .......... .......... 26%  191M 0s\n",
            " 14850K .......... .......... .......... .......... .......... 26%  142M 0s\n",
            " 14900K .......... .......... .......... .......... .......... 26%  184M 0s\n",
            " 14950K .......... .......... .......... .......... .......... 26%  162M 0s\n",
            " 15000K .......... .......... .......... .......... .......... 26%  188M 0s\n",
            " 15050K .......... .......... .......... .......... .......... 26%  186M 0s\n",
            " 15100K .......... .......... .......... .......... .......... 26%  182M 0s\n",
            " 15150K .......... .......... .......... .......... .......... 26%  153M 0s\n",
            " 15200K .......... .......... .......... .......... .......... 26%  177M 0s\n",
            " 15250K .......... .......... .......... .......... .......... 26%  185M 0s\n",
            " 15300K .......... .......... .......... .......... .......... 26%  179M 0s\n",
            " 15350K .......... .......... .......... .......... .......... 26%  162M 0s\n",
            " 15400K .......... .......... .......... .......... .......... 27% 36.1M 0s\n",
            " 15450K .......... .......... .......... .......... .......... 27%  194M 0s\n",
            " 15500K .......... .......... .......... .......... .......... 27%  192M 0s\n",
            " 15550K .......... .......... .......... .......... .......... 27%  145M 0s\n",
            " 15600K .......... .......... .......... .......... .......... 27%  160M 0s\n",
            " 15650K .......... .......... .......... .......... .......... 27%  172M 0s\n",
            " 15700K .......... .......... .......... .......... .......... 27%  185M 0s\n",
            " 15750K .......... .......... .......... .......... .......... 27%  160M 0s\n",
            " 15800K .......... .......... .......... .......... .......... 27%  198M 0s\n",
            " 15850K .......... .......... .......... .......... .......... 27%  196M 0s\n",
            " 15900K .......... .......... .......... .......... .......... 27%  184M 0s\n",
            " 15950K .......... .......... .......... .......... .......... 28%  169M 0s\n",
            " 16000K .......... .......... .......... .......... .......... 28%  201M 0s\n",
            " 16050K .......... .......... .......... .......... .......... 28%  200M 0s\n",
            " 16100K .......... .......... .......... .......... .......... 28%  181M 0s\n",
            " 16150K .......... .......... .......... .......... .......... 28%  138M 0s\n",
            " 16200K .......... .......... .......... .......... .......... 28%  165M 0s\n",
            " 16250K .......... .......... .......... .......... .......... 28%  156M 0s\n",
            " 16300K .......... .......... .......... .......... .......... 28%  177M 0s\n",
            " 16350K .......... .......... .......... .......... .......... 28%  175M 0s\n",
            " 16400K .......... .......... .......... .......... .......... 28%  197M 0s\n",
            " 16450K .......... .......... .......... .......... .......... 28%  177M 0s\n",
            " 16500K .......... .......... .......... .......... .......... 28%  178M 0s\n",
            " 16550K .......... .......... .......... .......... .......... 29%  159M 0s\n",
            " 16600K .......... .......... .......... .......... .......... 29%  322M 0s\n",
            " 16650K .......... .......... .......... .......... .......... 29%  180M 0s\n",
            " 16700K .......... .......... .......... .......... .......... 29%  201M 0s\n",
            " 16750K .......... .......... .......... .......... .......... 29%  176M 0s\n",
            " 16800K .......... .......... .......... .......... .......... 29%  176M 0s\n",
            " 16850K .......... .......... .......... .......... .......... 29%  156M 0s\n",
            " 16900K .......... .......... .......... .......... .......... 29%  201M 0s\n",
            " 16950K .......... .......... .......... .......... .......... 29%  238M 0s\n",
            " 17000K .......... .......... .......... .......... .......... 29%  327M 0s\n",
            " 17050K .......... .......... .......... .......... .......... 29%  237M 0s\n",
            " 17100K .......... .......... .......... .......... .......... 30%  209M 0s\n",
            " 17150K .......... .......... .......... .......... .......... 30%  148M 0s\n",
            " 17200K .......... .......... .......... .......... .......... 30%  161M 0s\n",
            " 17250K .......... .......... .......... .......... .......... 30%  168M 0s\n",
            " 17300K .......... .......... .......... .......... .......... 30%  173M 0s\n",
            " 17350K .......... .......... .......... .......... .......... 30%  159M 0s\n",
            " 17400K .......... .......... .......... .......... .......... 30%  178M 0s\n",
            " 17450K .......... .......... .......... .......... .......... 30% 43.0M 0s\n",
            " 17500K .......... .......... .......... .......... .......... 30%  195M 0s\n",
            " 17550K .......... .......... .......... .......... .......... 30%  172M 0s\n",
            " 17600K .......... .......... .......... .......... .......... 30%  188M 0s\n",
            " 17650K .......... .......... .......... .......... .......... 30%  187M 0s\n",
            " 17700K .......... .......... .......... .......... .......... 31%  194M 0s\n",
            " 17750K .......... .......... .......... .......... .......... 31%  162M 0s\n",
            " 17800K .......... .......... .......... .......... .......... 31%  181M 0s\n",
            " 17850K .......... .......... .......... .......... .......... 31%  172M 0s\n",
            " 17900K .......... .......... .......... .......... .......... 31%  179M 0s\n",
            " 17950K .......... .......... .......... .......... .......... 31%  165M 0s\n",
            " 18000K .......... .......... .......... .......... .......... 31%  178M 0s\n",
            " 18050K .......... .......... .......... .......... .......... 31%  192M 0s\n",
            " 18100K .......... .......... .......... .......... .......... 31%  190M 0s\n",
            " 18150K .......... .......... .......... .......... .......... 31%  148M 0s\n",
            " 18200K .......... .......... .......... .......... .......... 31%  185M 0s\n",
            " 18250K .......... .......... .......... .......... .......... 32%  169M 0s\n",
            " 18300K .......... .......... .......... .......... .......... 32%  168M 0s\n",
            " 18350K .......... .......... .......... .......... .......... 32%  148M 0s\n",
            " 18400K .......... .......... .......... .......... .......... 32%  194M 0s\n",
            " 18450K .......... .......... .......... .......... .......... 32%  190M 0s\n",
            " 18500K .......... .......... .......... .......... .......... 32%  185M 0s\n",
            " 18550K .......... .......... .......... .......... .......... 32%  156M 0s\n",
            " 18600K .......... .......... .......... .......... .......... 32%  164M 0s\n",
            " 18650K .......... .......... .......... .......... .......... 32%  170M 0s\n",
            " 18700K .......... .......... .......... .......... .......... 32%  177M 0s\n",
            " 18750K .......... .......... .......... .......... .......... 32%  169M 0s\n",
            " 18800K .......... .......... .......... .......... .......... 33%  213M 0s\n",
            " 18850K .......... .......... .......... .......... .......... 33%  201M 0s\n",
            " 18900K .......... .......... .......... .......... .......... 33%  159M 0s\n",
            " 18950K .......... .......... .......... .......... .......... 33%  154M 0s\n",
            " 19000K .......... .......... .......... .......... .......... 33%  198M 0s\n",
            " 19050K .......... .......... .......... .......... .......... 33%  178M 0s\n",
            " 19100K .......... .......... .......... .......... .......... 33%  182M 0s\n",
            " 19150K .......... .......... .......... .......... .......... 33%  153M 0s\n",
            " 19200K .......... .......... .......... .......... .......... 33%  184M 0s\n",
            " 19250K .......... .......... .......... .......... .......... 33%  173M 0s\n",
            " 19300K .......... .......... .......... .......... .......... 33%  185M 0s\n",
            " 19350K .......... .......... .......... .......... .......... 33%  157M 0s\n",
            " 19400K .......... .......... .......... .......... .......... 34%  176M 0s\n",
            " 19450K .......... .......... .......... .......... .......... 34%  193M 0s\n",
            " 19500K .......... .......... .......... .......... .......... 34% 42.2M 0s\n",
            " 19550K .......... .......... .......... .......... .......... 34%  176M 0s\n",
            " 19600K .......... .......... .......... .......... .......... 34%  157M 0s\n",
            " 19650K .......... .......... .......... .......... .......... 34%  194M 0s\n",
            " 19700K .......... .......... .......... .......... .......... 34%  193M 0s\n",
            " 19750K .......... .......... .......... .......... .......... 34%  161M 0s\n",
            " 19800K .......... .......... .......... .......... .......... 34%  180M 0s\n",
            " 19850K .......... .......... .......... .......... .......... 34%  174M 0s\n",
            " 19900K .......... .......... .......... .......... .......... 34%  168M 0s\n",
            " 19950K .......... .......... .......... .......... .......... 35%  158M 0s\n",
            " 20000K .......... .......... .......... .......... .......... 35%  190M 0s\n",
            " 20050K .......... .......... .......... .......... .......... 35%  195M 0s\n",
            " 20100K .......... .......... .......... .......... .......... 35%  192M 0s\n",
            " 20150K .......... .......... .......... .......... .......... 35%  150M 0s\n",
            " 20200K .......... .......... .......... .......... .......... 35%  165M 0s\n",
            " 20250K .......... .......... .......... .......... .......... 35%  187M 0s\n",
            " 20300K .......... .......... .......... .......... .......... 35%  158M 0s\n",
            " 20350K .......... .......... .......... .......... .......... 35%  135M 0s\n",
            " 20400K .......... .......... .......... .......... .......... 35%  175M 0s\n",
            " 20450K .......... .......... .......... .......... .......... 35%  179M 0s\n",
            " 20500K .......... .......... .......... .......... .......... 35%  175M 0s\n",
            " 20550K .......... .......... .......... .......... .......... 36%  144M 0s\n",
            " 20600K .......... .......... .......... .......... .......... 36%  175M 0s\n",
            " 20650K .......... .......... .......... .......... .......... 36%  174M 0s\n",
            " 20700K .......... .......... .......... .......... .......... 36%  195M 0s\n",
            " 20750K .......... .......... .......... .......... .......... 36%  174M 0s\n",
            " 20800K .......... .......... .......... .......... .......... 36%  184M 0s\n",
            " 20850K .......... .......... .......... .......... .......... 36%  150M 0s\n",
            " 20900K .......... .......... .......... .......... .......... 36%  196M 0s\n",
            " 20950K .......... .......... .......... .......... .......... 36%  162M 0s\n",
            " 21000K .......... .......... .......... .......... .......... 36%  184M 0s\n",
            " 21050K .......... .......... .......... .......... .......... 36%  190M 0s\n",
            " 21100K .......... .......... .......... .......... .......... 37%  168M 0s\n",
            " 21150K .......... .......... .......... .......... .......... 37%  151M 0s\n",
            " 21200K .......... .......... .......... .......... .......... 37%  179M 0s\n",
            " 21250K .......... .......... .......... .......... .......... 37%  191M 0s\n",
            " 21300K .......... .......... .......... .......... .......... 37%  195M 0s\n",
            " 21350K .......... .......... .......... .......... .......... 37%  155M 0s\n",
            " 21400K .......... .......... .......... .......... .......... 37%  184M 0s\n",
            " 21450K .......... .......... .......... .......... .......... 37%  193M 0s\n",
            " 21500K .......... .......... .......... .......... .......... 37%  179M 0s\n",
            " 21550K .......... .......... .......... .......... .......... 37% 47.8M 0s\n",
            " 21600K .......... .......... .......... .......... .......... 37%  152M 0s\n",
            " 21650K .......... .......... .......... .......... .......... 38%  187M 0s\n",
            " 21700K .......... .......... .......... .......... .......... 38%  199M 0s\n",
            " 21750K .......... .......... .......... .......... .......... 38%  164M 0s\n",
            " 21800K .......... .......... .......... .......... .......... 38%  173M 0s\n",
            " 21850K .......... .......... .......... .......... .......... 38%  187M 0s\n",
            " 21900K .......... .......... .......... .......... .......... 38%  193M 0s\n",
            " 21950K .......... .......... .......... .......... .......... 38%  169M 0s\n",
            " 22000K .......... .......... .......... .......... .......... 38%  162M 0s\n",
            " 22050K .......... .......... .......... .......... .......... 38%  185M 0s\n",
            " 22100K .......... .......... .......... .......... .......... 38%  196M 0s\n",
            " 22150K .......... .......... .......... .......... .......... 38%  151M 0s\n",
            " 22200K .......... .......... .......... .......... .......... 38%  198M 0s\n",
            " 22250K .......... .......... .......... .......... .......... 39%  189M 0s\n",
            " 22300K .......... .......... .......... .......... .......... 39%  200M 0s\n",
            " 22350K .......... .......... .......... .......... .......... 39%  163M 0s\n",
            " 22400K .......... .......... .......... .......... .......... 39%  202M 0s\n",
            " 22450K .......... .......... .......... .......... .......... 39%  206M 0s\n",
            " 22500K .......... .......... .......... .......... .......... 39%  186M 0s\n",
            " 22550K .......... .......... .......... .......... .......... 39%  130M 0s\n",
            " 22600K .......... .......... .......... .......... .......... 39%  179M 0s\n",
            " 22650K .......... .......... .......... .......... .......... 39%  198M 0s\n",
            " 22700K .......... .......... .......... .......... .......... 39%  178M 0s\n",
            " 22750K .......... .......... .......... .......... .......... 39%  173M 0s\n",
            " 22800K .......... .......... .......... .......... .......... 40%  194M 0s\n",
            " 22850K .......... .......... .......... .......... .......... 40%  191M 0s\n",
            " 22900K .......... .......... .......... .......... .......... 40%  158M 0s\n",
            " 22950K .......... .......... .......... .......... .......... 40%  145M 0s\n",
            " 23000K .......... .......... .......... .......... .......... 40%  191M 0s\n",
            " 23050K .......... .......... .......... .......... .......... 40%  180M 0s\n",
            " 23100K .......... .......... .......... .......... .......... 40%  195M 0s\n",
            " 23150K .......... .......... .......... .......... .......... 40%  179M 0s\n",
            " 23200K .......... .......... .......... .......... .......... 40%  197M 0s\n",
            " 23250K .......... .......... .......... .......... .......... 40%  177M 0s\n",
            " 23300K .......... .......... .......... .......... .......... 40%  193M 0s\n",
            " 23350K .......... .......... .......... .......... .......... 40%  137M 0s\n",
            " 23400K .......... .......... .......... .......... .......... 41%  183M 0s\n",
            " 23450K .......... .......... .......... .......... .......... 41%  187M 0s\n",
            " 23500K .......... .......... .......... .......... .......... 41%  194M 0s\n",
            " 23550K .......... .......... .......... .......... .......... 41% 37.1M 0s\n",
            " 23600K .......... .......... .......... .......... .......... 41%  148M 0s\n",
            " 23650K .......... .......... .......... .......... .......... 41%  169M 0s\n",
            " 23700K .......... .......... .......... .......... .......... 41%  159M 0s\n",
            " 23750K .......... .......... .......... .......... .......... 41%  133M 0s\n",
            " 23800K .......... .......... .......... .......... .......... 41%  176M 0s\n",
            " 23850K .......... .......... .......... .......... .......... 41%  194M 0s\n",
            " 23900K .......... .......... .......... .......... .......... 41%  197M 0s\n",
            " 23950K .......... .......... .......... .......... .......... 42%  146M 0s\n",
            " 24000K .......... .......... .......... .......... .......... 42%  196M 0s\n",
            " 24050K .......... .......... .......... .......... .......... 42%  195M 0s\n",
            " 24100K .......... .......... .......... .......... .......... 42%  183M 0s\n",
            " 24150K .......... .......... .......... .......... .......... 42%  140M 0s\n",
            " 24200K .......... .......... .......... .......... .......... 42%  196M 0s\n",
            " 24250K .......... .......... .......... .......... .......... 42%  198M 0s\n",
            " 24300K .......... .......... .......... .......... .......... 42%  179M 0s\n",
            " 24350K .......... .......... .......... .......... .......... 42%  177M 0s\n",
            " 24400K .......... .......... .......... .......... .......... 42%  195M 0s\n",
            " 24450K .......... .......... .......... .......... .......... 42%  199M 0s\n",
            " 24500K .......... .......... .......... .......... .......... 42%  161M 0s\n",
            " 24550K .......... .......... .......... .......... .......... 43%  159M 0s\n",
            " 24600K .......... .......... .......... .......... .......... 43%  196M 0s\n",
            " 24650K .......... .......... .......... .......... .......... 43%  189M 0s\n",
            " 24700K .......... .......... .......... .......... .......... 43%  190M 0s\n",
            " 24750K .......... .......... .......... .......... .......... 43%  175M 0s\n",
            " 24800K .......... .......... .......... .......... .......... 43%  197M 0s\n",
            " 24850K .......... .......... .......... .......... .......... 43%  176M 0s\n",
            " 24900K .......... .......... .......... .......... .......... 43%  196M 0s\n",
            " 24950K .......... .......... .......... .......... .......... 43%  163M 0s\n",
            " 25000K .......... .......... .......... .......... .......... 43%  193M 0s\n",
            " 25050K .......... .......... .......... .......... .......... 43%  152M 0s\n",
            " 25100K .......... .......... .......... .......... .......... 44%  177M 0s\n",
            " 25150K .......... .......... .......... .......... .......... 44%  161M 0s\n",
            " 25200K .......... .......... .......... .......... .......... 44%  179M 0s\n",
            " 25250K .......... .......... .......... .......... .......... 44%  193M 0s\n",
            " 25300K .......... .......... .......... .......... .......... 44%  175M 0s\n",
            " 25350K .......... .......... .......... .......... .......... 44%  147M 0s\n",
            " 25400K .......... .......... .......... .......... .......... 44%  163M 0s\n",
            " 25450K .......... .......... .......... .......... .......... 44%  191M 0s\n",
            " 25500K .......... .......... .......... .......... .......... 44%  193M 0s\n",
            " 25550K .......... .......... .......... .......... .......... 44%  165M 0s\n",
            " 25600K .......... .......... .......... .......... .......... 44% 36.6M 0s\n",
            " 25650K .......... .......... .......... .......... .......... 45%  182M 0s\n",
            " 25700K .......... .......... .......... .......... .......... 45%  195M 0s\n",
            " 25750K .......... .......... .......... .......... .......... 45%  152M 0s\n",
            " 25800K .......... .......... .......... .......... .......... 45%  192M 0s\n",
            " 25850K .......... .......... .......... .......... .......... 45%  195M 0s\n",
            " 25900K .......... .......... .......... .......... .......... 45%  197M 0s\n",
            " 25950K .......... .......... .......... .......... .......... 45%  154M 0s\n",
            " 26000K .......... .......... .......... .......... .......... 45%  194M 0s\n",
            " 26050K .......... .......... .......... .......... .......... 45%  195M 0s\n",
            " 26100K .......... .......... .......... .......... .......... 45%  173M 0s\n",
            " 26150K .......... .......... .......... .......... .......... 45%  150M 0s\n",
            " 26200K .......... .......... .......... .......... .......... 45%  225M 0s\n",
            " 26250K .......... .......... .......... .......... .......... 46%  180M 0s\n",
            " 26300K .......... .......... .......... .......... .......... 46%  176M 0s\n",
            " 26350K .......... .......... .......... .......... .......... 46%  176M 0s\n",
            " 26400K .......... .......... .......... .......... .......... 46%  191M 0s\n",
            " 26450K .......... .......... .......... .......... .......... 46%  195M 0s\n",
            " 26500K .......... .......... .......... .......... .......... 46%  163M 0s\n",
            " 26550K .......... .......... .......... .......... .......... 46%  154M 0s\n",
            " 26600K .......... .......... .......... .......... .......... 46%  198M 0s\n",
            " 26650K .......... .......... .......... .......... .......... 46%  186M 0s\n",
            " 26700K .......... .......... .......... .......... .......... 46%  238M 0s\n",
            " 26750K .......... .......... .......... .......... .......... 46%  205M 0s\n",
            " 26800K .......... .......... .......... .......... .......... 47% 59.8M 0s\n",
            " 26850K .......... .......... .......... .......... .......... 47%  197M 0s\n",
            " 26900K .......... .......... .......... .......... .......... 47%  222M 0s\n",
            " 26950K .......... .......... .......... .......... .......... 47%  140M 0s\n",
            " 27000K .......... .......... .......... .......... .......... 47%  194M 0s\n",
            " 27050K .......... .......... .......... .......... .......... 47%  182M 0s\n",
            " 27100K .......... .......... .......... .......... .......... 47%  172M 0s\n",
            " 27150K .......... .......... .......... .......... .......... 47%  167M 0s\n",
            " 27200K .......... .......... .......... .......... .......... 47%  208M 0s\n",
            " 27250K .......... .......... .......... .......... .......... 47%  238M 0s\n",
            " 27300K .......... .......... .......... .......... .......... 47%  232M 0s\n",
            " 27350K .......... .......... .......... .......... .......... 47%  148M 0s\n",
            " 27400K .......... .......... .......... .......... .......... 48%  197M 0s\n",
            " 27450K .......... .......... .......... .......... .......... 48%  201M 0s\n",
            " 27500K .......... .......... .......... .......... .......... 48%  213M 0s\n",
            " 27550K .......... .......... .......... .......... .......... 48% 36.9M 0s\n",
            " 27600K .......... .......... .......... .......... .......... 48%  222M 0s\n",
            " 27650K .......... .......... .......... .......... .......... 48%  175M 0s\n",
            " 27700K .......... .......... .......... .......... .......... 48%  179M 0s\n",
            " 27750K .......... .......... .......... .......... .......... 48%  177M 0s\n",
            " 27800K .......... .......... .......... .......... .......... 48%  229M 0s\n",
            " 27850K .......... .......... .......... .......... .......... 48%  238M 0s\n",
            " 27900K .......... .......... .......... .......... .......... 48%  222M 0s\n",
            " 27950K .......... .......... .......... .......... .......... 49%  135M 0s\n",
            " 28000K .......... .......... .......... .......... .......... 49%  235M 0s\n",
            " 28050K .......... .......... .......... .......... .......... 49%  201M 0s\n",
            " 28100K .......... .......... .......... .......... .......... 49%  181M 0s\n",
            " 28150K .......... .......... .......... .......... .......... 49%  176M 0s\n",
            " 28200K .......... .......... .......... .......... .......... 49%  184M 0s\n",
            " 28250K .......... .......... .......... .......... .......... 49%  215M 0s\n",
            " 28300K .......... .......... .......... .......... .......... 49%  205M 0s\n",
            " 28350K .......... .......... .......... .......... .......... 49%  192M 0s\n",
            " 28400K .......... .......... .......... .......... .......... 49%  219M 0s\n",
            " 28450K .......... .......... .......... .......... .......... 49%  217M 0s\n",
            " 28500K .......... .......... .......... .......... .......... 50%  222M 0s\n",
            " 28550K .......... .......... .......... .......... .......... 50%  136M 0s\n",
            " 28600K .......... .......... .......... .......... .......... 50%  217M 0s\n",
            " 28650K .......... .......... .......... .......... .......... 50%  197M 0s\n",
            " 28700K .......... .......... .......... .......... .......... 50%  200M 0s\n",
            " 28750K .......... .......... .......... .......... .......... 50%  185M 0s\n",
            " 28800K .......... .......... .......... .......... .......... 50%  215M 0s\n",
            " 28850K .......... .......... .......... .......... .......... 50%  218M 0s\n",
            " 28900K .......... .......... .......... .......... .......... 50%  219M 0s\n",
            " 28950K .......... .......... .......... .......... .......... 50%  162M 0s\n",
            " 29000K .......... .......... .......... .......... .......... 50%  236M 0s\n",
            " 29050K .......... .......... .......... .......... .......... 50%  223M 0s\n",
            " 29100K .......... .......... .......... .......... .......... 51%  199M 0s\n",
            " 29150K .......... .......... .......... .......... .......... 51% 43.2M 0s\n",
            " 29200K .......... .......... .......... .......... .......... 51%  106M 0s\n",
            " 29250K .......... .......... .......... .......... .......... 51%  192M 0s\n",
            " 29300K .......... .......... .......... .......... .......... 51%  149M 0s\n",
            " 29350K .......... .......... .......... .......... .......... 51%  147M 0s\n",
            " 29400K .......... .......... .......... .......... .......... 51%  111M 0s\n",
            " 29450K .......... .......... .......... .......... .......... 51%  127M 0s\n",
            " 29500K .......... .......... .......... .......... .......... 51%  158M 0s\n",
            " 29550K .......... .......... .......... .......... .......... 51%  112M 0s\n",
            " 29600K .......... .......... .......... .......... .......... 51% 38.3M 0s\n",
            " 29650K .......... .......... .......... .......... .......... 52%  166M 0s\n",
            " 29700K .......... .......... .......... .......... .......... 52%  159M 0s\n",
            " 29750K .......... .......... .......... .......... .......... 52%  105M 0s\n",
            " 29800K .......... .......... .......... .......... .......... 52%  166M 0s\n",
            " 29850K .......... .......... .......... .......... .......... 52%  153M 0s\n",
            " 29900K .......... .......... .......... .......... .......... 52%  141M 0s\n",
            " 29950K .......... .......... .......... .......... .......... 52%  137M 0s\n",
            " 30000K .......... .......... .......... .......... .......... 52%  166M 0s\n",
            " 30050K .......... .......... .......... .......... .......... 52%  153M 0s\n",
            " 30100K .......... .......... .......... .......... .......... 52%  164M 0s\n",
            " 30150K .......... .......... .......... .......... .......... 52%  110M 0s\n",
            " 30200K .......... .......... .......... .......... .......... 52%  138M 0s\n",
            " 30250K .......... .......... .......... .......... .......... 53%  141M 0s\n",
            " 30300K .......... .......... .......... .......... .......... 53%  150M 0s\n",
            " 30350K .......... .......... .......... .......... .......... 53%  143M 0s\n",
            " 30400K .......... .......... .......... .......... .......... 53%  166M 0s\n",
            " 30450K .......... .......... .......... .......... .......... 53%  154M 0s\n",
            " 30500K .......... .......... .......... .......... .......... 53%  159M 0s\n",
            " 30550K .......... .......... .......... .......... .......... 53%  138M 0s\n",
            " 30600K .......... .......... .......... .......... .......... 53%  159M 0s\n",
            " 30650K .......... .......... .......... .......... .......... 53%  133M 0s\n",
            " 30700K .......... .......... .......... .......... .......... 53%  141M 0s\n",
            " 30750K .......... .......... .......... .......... .......... 53%  126M 0s\n",
            " 30800K .......... .......... .......... .......... .......... 54%  166M 0s\n",
            " 30850K .......... .......... .......... .......... .......... 54%  150M 0s\n",
            " 30900K .......... .......... .......... .......... .......... 54%  135M 0s\n",
            " 30950K .......... .......... .......... .......... .......... 54%  126M 0s\n",
            " 31000K .......... .......... .......... .......... .......... 54%  145M 0s\n",
            " 31050K .......... .......... .......... .......... .......... 54%  109M 0s\n",
            " 31100K .......... .......... .......... .......... .......... 54%  160M 0s\n",
            " 31150K .......... .......... .......... .......... .......... 54%  123M 0s\n",
            " 31200K .......... .......... .......... .......... .......... 54%  140M 0s\n",
            " 31250K .......... .......... .......... .......... .......... 54%  162M 0s\n",
            " 31300K .......... .......... .......... .......... .......... 54%  162M 0s\n",
            " 31350K .......... .......... .......... .......... .......... 54%  119M 0s\n",
            " 31400K .......... .......... .......... .......... .......... 55%  164M 0s\n",
            " 31450K .......... .......... .......... .......... .......... 55%  164M 0s\n",
            " 31500K .......... .......... .......... .......... .......... 55%  137M 0s\n",
            " 31550K .......... .......... .......... .......... .......... 55%  130M 0s\n",
            " 31600K .......... .......... .......... .......... .......... 55%  163M 0s\n",
            " 31650K .......... .......... .......... .......... .......... 55% 38.3M 0s\n",
            " 31700K .......... .......... .......... .......... .......... 55%  159M 0s\n",
            " 31750K .......... .......... .......... .......... .......... 55%  106M 0s\n",
            " 31800K .......... .......... .......... .......... .......... 55%  126M 0s\n",
            " 31850K .......... .......... .......... .......... .......... 55%  172M 0s\n",
            " 31900K .......... .......... .......... .......... .......... 55%  187M 0s\n",
            " 31950K .......... .......... .......... .......... .......... 56%  151M 0s\n",
            " 32000K .......... .......... .......... .......... .......... 56%  182M 0s\n",
            " 32050K .......... .......... .......... .......... .......... 56%  188M 0s\n",
            " 32100K .......... .......... .......... .......... .......... 56%  174M 0s\n",
            " 32150K .......... .......... .......... .......... .......... 56% 98.0M 0s\n",
            " 32200K .......... .......... .......... .......... .......... 56%  174M 0s\n",
            " 32250K .......... .......... .......... .......... .......... 56%  173M 0s\n",
            " 32300K .......... .......... .......... .......... .......... 56%  157M 0s\n",
            " 32350K .......... .......... .......... .......... .......... 56%  156M 0s\n",
            " 32400K .......... .......... .......... .......... .......... 56%  153M 0s\n",
            " 32450K .......... .......... .......... .......... .......... 56%  183M 0s\n",
            " 32500K .......... .......... .......... .......... .......... 57%  178M 0s\n",
            " 32550K .......... .......... .......... .......... .......... 57%  137M 0s\n",
            " 32600K .......... .......... .......... .......... .......... 57%  150M 0s\n",
            " 32650K .......... .......... .......... .......... .......... 57%  186M 0s\n",
            " 32700K .......... .......... .......... .......... .......... 57%  184M 0s\n",
            " 32750K .......... .......... .......... .......... .......... 57%  158M 0s\n",
            " 32800K .......... .......... .......... .......... .......... 57%  186M 0s\n",
            " 32850K .......... .......... .......... .......... .......... 57%  190M 0s\n",
            " 32900K .......... .......... .......... .......... .......... 57%  173M 0s\n",
            " 32950K .......... .......... .......... .......... .......... 57%  120M 0s\n",
            " 33000K .......... .......... .......... .......... .......... 57% 39.5M 0s\n",
            " 33050K .......... .......... .......... .......... .......... 57%  182M 0s\n",
            " 33100K .......... .......... .......... .......... .......... 58%  180M 0s\n",
            " 33150K .......... .......... .......... .......... .......... 58%  149M 0s\n",
            " 33200K .......... .......... .......... .......... .......... 58%  176M 0s\n",
            " 33250K .......... .......... .......... .......... .......... 58%  218M 0s\n",
            " 33300K .......... .......... .......... .......... .......... 58%  181M 0s\n",
            " 33350K .......... .......... .......... .......... .......... 58%  143M 0s\n",
            " 33400K .......... .......... .......... .......... .......... 58%  193M 0s\n",
            " 33450K .......... .......... .......... .......... .......... 58%  183M 0s\n",
            " 33500K .......... .......... .......... .......... .......... 58%  176M 0s\n",
            " 33550K .......... .......... .......... .......... .......... 58%  170M 0s\n",
            " 33600K .......... .......... .......... .......... .......... 58%  198M 0s\n",
            " 33650K .......... .......... .......... .......... .......... 59%  183M 0s\n",
            " 33700K .......... .......... .......... .......... .......... 59% 41.0M 0s\n",
            " 33750K .......... .......... .......... .......... .......... 59%  157M 0s\n",
            " 33800K .......... .......... .......... .......... .......... 59%  198M 0s\n",
            " 33850K .......... .......... .......... .......... .......... 59%  190M 0s\n",
            " 33900K .......... .......... .......... .......... .......... 59%  191M 0s\n",
            " 33950K .......... .......... .......... .......... .......... 59%  176M 0s\n",
            " 34000K .......... .......... .......... .......... .......... 59%  197M 0s\n",
            " 34050K .......... .......... .......... .......... .......... 59%  161M 0s\n",
            " 34100K .......... .......... .......... .......... .......... 59%  160M 0s\n",
            " 34150K .......... .......... .......... .......... .......... 59%  141M 0s\n",
            " 34200K .......... .......... .......... .......... .......... 59%  181M 0s\n",
            " 34250K .......... .......... .......... .......... .......... 60%  190M 0s\n",
            " 34300K .......... .......... .......... .......... .......... 60%  198M 0s\n",
            " 34350K .......... .......... .......... .......... .......... 60%  176M 0s\n",
            " 34400K .......... .......... .......... .......... .......... 60%  184M 0s\n",
            " 34450K .......... .......... .......... .......... .......... 60%  177M 0s\n",
            " 34500K .......... .......... .......... .......... .......... 60%  194M 0s\n",
            " 34550K .......... .......... .......... .......... .......... 60%  156M 0s\n",
            " 34600K .......... .......... .......... .......... .......... 60%  136M 0s\n",
            " 34650K .......... .......... .......... .......... .......... 60%  132M 0s\n",
            " 34700K .......... .......... .......... .......... .......... 60%  124M 0s\n",
            " 34750K .......... .......... .......... .......... .......... 60%  109M 0s\n",
            " 34800K .......... .......... .......... .......... .......... 61%  151M 0s\n",
            " 34850K .......... .......... .......... .......... .......... 61%  127M 0s\n",
            " 34900K .......... .......... .......... .......... .......... 61%  142M 0s\n",
            " 34950K .......... .......... .......... .......... .......... 61%  123M 0s\n",
            " 35000K .......... .......... .......... .......... .......... 61%  129M 0s\n",
            " 35050K .......... .......... .......... .......... .......... 61%  147M 0s\n",
            " 35100K .......... .......... .......... .......... .......... 61%  132M 0s\n",
            " 35150K .......... .......... .......... .......... .......... 61%  136M 0s\n",
            " 35200K .......... .......... .......... .......... .......... 61%  162M 0s\n",
            " 35250K .......... .......... .......... .......... .......... 61%  148M 0s\n",
            " 35300K .......... .......... .......... .......... .......... 61%  159M 0s\n",
            " 35350K .......... .......... .......... .......... .......... 61%  133M 0s\n",
            " 35400K .......... .......... .......... .......... .......... 62%  148M 0s\n",
            " 35450K .......... .......... .......... .......... .......... 62%  138M 0s\n",
            " 35500K .......... .......... .......... .......... .......... 62%  143M 0s\n",
            " 35550K .......... .......... .......... .......... .......... 62%  133M 0s\n",
            " 35600K .......... .......... .......... .......... .......... 62%  159M 0s\n",
            " 35650K .......... .......... .......... .......... .......... 62%  163M 0s\n",
            " 35700K .......... .......... .......... .......... .......... 62%  155M 0s\n",
            " 35750K .......... .......... .......... .......... .......... 62% 36.4M 0s\n",
            " 35800K .......... .......... .......... .......... .......... 62%  131M 0s\n",
            " 35850K .......... .......... .......... .......... .......... 62%  149M 0s\n",
            " 35900K .......... .......... .......... .......... .......... 62%  162M 0s\n",
            " 35950K .......... .......... .......... .......... .......... 63%  146M 0s\n",
            " 36000K .......... .......... .......... .......... .......... 63%  149M 0s\n",
            " 36050K .......... .......... .......... .......... .......... 63%  163M 0s\n",
            " 36100K .......... .......... .......... .......... .......... 63%  146M 0s\n",
            " 36150K .......... .......... .......... .......... .......... 63%  127M 0s\n",
            " 36200K .......... .......... .......... .......... .......... 63%  163M 0s\n",
            " 36250K .......... .......... .......... .......... .......... 63%  163M 0s\n",
            " 36300K .......... .......... .......... .......... .......... 63%  134M 0s\n",
            " 36350K .......... .......... .......... .......... .......... 63%  132M 0s\n",
            " 36400K .......... .......... .......... .......... .......... 63%  145M 0s\n",
            " 36450K .......... .......... .......... .......... .......... 63%  137M 0s\n",
            " 36500K .......... .......... .......... .......... .......... 64%  147M 0s\n",
            " 36550K .......... .......... .......... .......... .......... 64%  126M 0s\n",
            " 36600K .......... .......... .......... .......... .......... 64%  134M 0s\n",
            " 36650K .......... .......... .......... .......... .......... 64%  153M 0s\n",
            " 36700K .......... .......... .......... .......... .......... 64%  161M 0s\n",
            " 36750K .......... .......... .......... .......... .......... 64%  125M 0s\n",
            " 36800K .......... .......... .......... .......... .......... 64%  164M 0s\n",
            " 36850K .......... .......... .......... .......... .......... 64%  159M 0s\n",
            " 36900K .......... .......... .......... .......... .......... 64%  137M 0s\n",
            " 36950K .......... .......... .......... .......... .......... 64%  135M 0s\n",
            " 37000K .......... .......... .......... .......... .......... 64%  161M 0s\n",
            " 37050K .......... .......... .......... .......... .......... 64%  137M 0s\n",
            " 37100K .......... .......... .......... .......... .......... 65%  146M 0s\n",
            " 37150K .......... .......... .......... .......... .......... 65%  135M 0s\n",
            " 37200K .......... .......... .......... .......... .......... 65%  150M 0s\n",
            " 37250K .......... .......... .......... .......... .......... 65%  154M 0s\n",
            " 37300K .......... .......... .......... .......... .......... 65%  163M 0s\n",
            " 37350K .......... .......... .......... .......... .......... 65%  127M 0s\n",
            " 37400K .......... .......... .......... .......... .......... 65%  158M 0s\n",
            " 37450K .......... .......... .......... .......... .......... 65%  162M 0s\n",
            " 37500K .......... .......... .......... .......... .......... 65%  137M 0s\n",
            " 37550K .......... .......... .......... .......... .......... 65%  121M 0s\n",
            " 37600K .......... .......... .......... .......... .......... 65%  160M 0s\n",
            " 37650K .......... .......... .......... .......... .......... 66%  152M 0s\n",
            " 37700K .......... .......... .......... .......... .......... 66%  199M 0s\n",
            " 37750K .......... .......... .......... .......... .......... 66%  177M 0s\n",
            " 37800K .......... .......... .......... .......... .......... 66% 37.3M 0s\n",
            " 37850K .......... .......... .......... .......... .......... 66%  160M 0s\n",
            " 37900K .......... .......... .......... .......... .......... 66%  166M 0s\n",
            " 37950K .......... .......... .......... .......... .......... 66%  120M 0s\n",
            " 38000K .......... .......... .......... .......... .......... 66%  132M 0s\n",
            " 38050K .......... .......... .......... .......... .......... 66%  166M 0s\n",
            " 38100K .......... .......... .......... .......... .......... 66%  144M 0s\n",
            " 38150K .......... .......... .......... .......... .......... 66%  132M 0s\n",
            " 38200K .......... .......... .......... .......... .......... 66%  163M 0s\n",
            " 38250K .......... .......... .......... .......... .......... 67%  140M 0s\n",
            " 38300K .......... .......... .......... .......... .......... 67%  148M 0s\n",
            " 38350K .......... .......... .......... .......... .......... 67%  146M 0s\n",
            " 38400K .......... .......... .......... .......... .......... 67%  153M 0s\n",
            " 38450K .......... .......... .......... .......... .......... 67%  145M 0s\n",
            " 38500K .......... .......... .......... .......... .......... 67%  163M 0s\n",
            " 38550K .......... .......... .......... .......... .......... 67%  116M 0s\n",
            " 38600K .......... .......... .......... .......... .......... 67%  183M 0s\n",
            " 38650K .......... .......... .......... .......... .......... 67%  179M 0s\n",
            " 38700K .......... .......... .......... .......... .......... 67%  167M 0s\n",
            " 38750K .......... .......... .......... .......... .......... 67%  125M 0s\n",
            " 38800K .......... .......... .......... .......... .......... 68%  177M 0s\n",
            " 38850K .......... .......... .......... .......... .......... 68%  187M 0s\n",
            " 38900K .......... .......... .......... .......... .......... 68%  152M 0s\n",
            " 38950K .......... .......... .......... .......... .......... 68%  224M 0s\n",
            " 39000K .......... .......... .......... .......... .......... 68%  268M 0s\n",
            " 39050K .......... .......... .......... .......... .......... 68%  271M 0s\n",
            " 39100K .......... .......... .......... .......... .......... 68%  236M 0s\n",
            " 39150K .......... .......... .......... .......... .......... 68%  131M 0s\n",
            " 39200K .......... .......... .......... .......... .......... 68%  162M 0s\n",
            " 39250K .......... .......... .......... .......... .......... 68%  150M 0s\n",
            " 39300K .......... .......... .......... .......... .......... 68%  155M 0s\n",
            " 39350K .......... .......... .......... .......... .......... 69%  115M 0s\n",
            " 39400K .......... .......... .......... .......... .......... 69%  154M 0s\n",
            " 39450K .......... .......... .......... .......... .......... 69%  165M 0s\n",
            " 39500K .......... .......... .......... .......... .......... 69%  161M 0s\n",
            " 39550K .......... .......... .......... .......... .......... 69%  137M 0s\n",
            " 39600K .......... .......... .......... .......... .......... 69%  164M 0s\n",
            " 39650K .......... .......... .......... .......... .......... 69%  155M 0s\n",
            " 39700K .......... .......... .......... .......... .......... 69%  164M 0s\n",
            " 39750K .......... .......... .......... .......... .......... 69%  118M 0s\n",
            " 39800K .......... .......... .......... .......... .......... 69%  164M 0s\n",
            " 39850K .......... .......... .......... .......... .......... 69% 37.8M 0s\n",
            " 39900K .......... .......... .......... .......... .......... 69%  150M 0s\n",
            " 39950K .......... .......... .......... .......... .......... 70%  148M 0s\n",
            " 40000K .......... .......... .......... .......... .......... 70%  143M 0s\n",
            " 40050K .......... .......... .......... .......... .......... 70%  153M 0s\n",
            " 40100K .......... .......... .......... .......... .......... 70%  147M 0s\n",
            " 40150K .......... .......... .......... .......... .......... 70%  130M 0s\n",
            " 40200K .......... .......... .......... .......... .......... 70%  128M 0s\n",
            " 40250K .......... .......... .......... .......... .......... 70%  165M 0s\n",
            " 40300K .......... .......... .......... .......... .......... 70%  153M 0s\n",
            " 40350K .......... .......... .......... .......... .......... 70%  136M 0s\n",
            " 40400K .......... .......... .......... .......... .......... 70%  168M 0s\n",
            " 40450K .......... .......... .......... .......... .......... 70%  164M 0s\n",
            " 40500K .......... .......... .......... .......... .......... 71%  154M 0s\n",
            " 40550K .......... .......... .......... .......... .......... 71%  138M 0s\n",
            " 40600K .......... .......... .......... .......... .......... 71%  164M 0s\n",
            " 40650K .......... .......... .......... .......... .......... 71%  144M 0s\n",
            " 40700K .......... .......... .......... .......... .......... 71%  137M 0s\n",
            " 40750K .......... .......... .......... .......... .......... 71%  141M 0s\n",
            " 40800K .......... .......... .......... .......... .......... 71%  143M 0s\n",
            " 40850K .......... .......... .......... .......... .......... 71%  162M 0s\n",
            " 40900K .......... .......... .......... .......... .......... 71%  167M 0s\n",
            " 40950K .......... .......... .......... .......... .......... 71%  124M 0s\n",
            " 41000K .......... .......... .......... .......... .......... 71%  171M 0s\n",
            " 41050K .......... .......... .......... .......... .......... 71%  158M 0s\n",
            " 41100K .......... .......... .......... .......... .......... 72%  154M 0s\n",
            " 41150K .......... .......... .......... .......... .......... 72%  149M 0s\n",
            " 41200K .......... .......... .......... .......... .......... 72%  167M 0s\n",
            " 41250K .......... .......... .......... .......... .......... 72%  141M 0s\n",
            " 41300K .......... .......... .......... .......... .......... 72%  150M 0s\n",
            " 41350K .......... .......... .......... .......... .......... 72%  138M 0s\n",
            " 41400K .......... .......... .......... .......... .......... 72%  132M 0s\n",
            " 41450K .......... .......... .......... .......... .......... 72%  165M 0s\n",
            " 41500K .......... .......... .......... .......... .......... 72%  169M 0s\n",
            " 41550K .......... .......... .......... .......... .......... 72%  139M 0s\n",
            " 41600K .......... .......... .......... .......... .......... 72%  149M 0s\n",
            " 41650K .......... .......... .......... .......... .......... 73%  162M 0s\n",
            " 41700K .......... .......... .......... .......... .......... 73%  131M 0s\n",
            " 41750K .......... .......... .......... .......... .......... 73%  135M 0s\n",
            " 41800K .......... .......... .......... .......... .......... 73%  155M 0s\n",
            " 41850K .......... .......... .......... .......... .......... 73%  148M 0s\n",
            " 41900K .......... .......... .......... .......... .......... 73% 38.7M 0s\n",
            " 41950K .......... .......... .......... .......... .......... 73%  140M 0s\n",
            " 42000K .......... .......... .......... .......... .......... 73%  151M 0s\n",
            " 42050K .......... .......... .......... .......... .......... 73%  166M 0s\n",
            " 42100K .......... .......... .......... .......... .......... 73%  168M 0s\n",
            " 42150K .......... .......... .......... .......... .......... 73%  129M 0s\n",
            " 42200K .......... .......... .......... .......... .......... 73%  157M 0s\n",
            " 42250K .......... .......... .......... .......... .......... 74%  166M 0s\n",
            " 42300K .......... .......... .......... .......... .......... 74%  162M 0s\n",
            " 42350K .......... .......... .......... .......... .......... 74%  126M 0s\n",
            " 42400K .......... .......... .......... .......... .......... 74%  147M 0s\n",
            " 42450K .......... .......... .......... .......... .......... 74%  164M 0s\n",
            " 42500K .......... .......... .......... .......... .......... 74%  136M 0s\n",
            " 42550K .......... .......... .......... .......... .......... 74%  139M 0s\n",
            " 42600K .......... .......... .......... .......... .......... 74%  163M 0s\n",
            " 42650K .......... .......... .......... .......... .......... 74%  149M 0s\n",
            " 42700K .......... .......... .......... .......... .......... 74%  164M 0s\n",
            " 42750K .......... .......... .......... .......... .......... 74%  140M 0s\n",
            " 42800K .......... .......... .......... .......... .......... 75%  122M 0s\n",
            " 42850K .......... .......... .......... .......... .......... 75%  152M 0s\n",
            " 42900K .......... .......... .......... .......... .......... 75%  165M 0s\n",
            " 42950K .......... .......... .......... .......... .......... 75%  116M 0s\n",
            " 43000K .......... .......... .......... .......... .......... 75%  158M 0s\n",
            " 43050K .......... .......... .......... .......... .......... 75%  143M 0s\n",
            " 43100K .......... .......... .......... .......... .......... 75%  158M 0s\n",
            " 43150K .......... .......... .......... .......... .......... 75%  146M 0s\n",
            " 43200K .......... .......... .......... .......... .......... 75%  165M 0s\n",
            " 43250K .......... .......... .......... .......... .......... 75%  135M 0s\n",
            " 43300K .......... .......... .......... .......... .......... 75%  161M 0s\n",
            " 43350K .......... .......... .......... .......... .......... 76%  139M 0s\n",
            " 43400K .......... .......... .......... .......... .......... 76%  115M 0s\n",
            " 43450K .......... .......... .......... .......... .......... 76%  151M 0s\n",
            " 43500K .......... .......... .......... .......... .......... 76%  149M 0s\n",
            " 43550K .......... .......... .......... .......... .......... 76%  140M 0s\n",
            " 43600K .......... .......... .......... .......... .......... 76%  164M 0s\n",
            " 43650K .......... .......... .......... .......... .......... 76%  154M 0s\n",
            " 43700K .......... .......... .......... .......... .......... 76%  148M 0s\n",
            " 43750K .......... .......... .......... .......... .......... 76%  127M 0s\n",
            " 43800K .......... .......... .......... .......... .......... 76%  139M 0s\n",
            " 43850K .......... .......... .......... .......... .......... 76%  161M 0s\n",
            " 43900K .......... .......... .......... .......... .......... 76%  163M 0s\n",
            " 43950K .......... .......... .......... .......... .......... 77% 40.3M 0s\n",
            " 44000K .......... .......... .......... .......... .......... 77%  151M 0s\n",
            " 44050K .......... .......... .......... .......... .......... 77%  138M 0s\n",
            " 44100K .......... .......... .......... .......... .......... 77%  142M 0s\n",
            " 44150K .......... .......... .......... .......... .......... 77%  117M 0s\n",
            " 44200K .......... .......... .......... .......... .......... 77%  165M 0s\n",
            " 44250K .......... .......... .......... .......... .......... 77%  157M 0s\n",
            " 44300K .......... .......... .......... .......... .......... 77%  150M 0s\n",
            " 44350K .......... .......... .......... .......... .......... 77%  139M 0s\n",
            " 44400K .......... .......... .......... .......... .......... 77%  140M 0s\n",
            " 44450K .......... .......... .......... .......... .......... 77%  151M 0s\n",
            " 44500K .......... .......... .......... .......... .......... 78%  153M 0s\n",
            " 44550K .......... .......... .......... .......... .......... 78%  136M 0s\n",
            " 44600K .......... .......... .......... .......... .......... 78%  153M 0s\n",
            " 44650K .......... .......... .......... .......... .......... 78%  163M 0s\n",
            " 44700K .......... .......... .......... .......... .......... 78%  166M 0s\n",
            " 44750K .......... .......... .......... .......... .......... 78%  135M 0s\n",
            " 44800K .......... .......... .......... .......... .......... 78%  158M 0s\n",
            " 44850K .......... .......... .......... .......... .......... 78%  150M 0s\n",
            " 44900K .......... .......... .......... .......... .......... 78%  145M 0s\n",
            " 44950K .......... .......... .......... .......... .......... 78%  125M 0s\n",
            " 45000K .......... .......... .......... .......... .......... 78%  142M 0s\n",
            " 45050K .......... .......... .......... .......... .......... 78%  144M 0s\n",
            " 45100K .......... .......... .......... .......... .......... 79%  154M 0s\n",
            " 45150K .......... .......... .......... .......... .......... 79%  142M 0s\n",
            " 45200K .......... .......... .......... .......... .......... 79%  152M 0s\n",
            " 45250K .......... .......... .......... .......... .......... 79%  133M 0s\n",
            " 45300K .......... .......... .......... .......... .......... 79%  157M 0s\n",
            " 45350K .......... .......... .......... .......... .......... 79%  113M 0s\n",
            " 45400K .......... .......... .......... .......... .......... 79%  162M 0s\n",
            " 45450K .......... .......... .......... .......... .......... 79%  161M 0s\n",
            " 45500K .......... .......... .......... .......... .......... 79%  146M 0s\n",
            " 45550K .......... .......... .......... .......... .......... 79%  145M 0s\n",
            " 45600K .......... .......... .......... .......... .......... 79%  163M 0s\n",
            " 45650K .......... .......... .......... .......... .......... 80%  147M 0s\n",
            " 45700K .......... .......... .......... .......... .......... 80%  165M 0s\n",
            " 45750K .......... .......... .......... .......... .......... 80%  136M 0s\n",
            " 45800K .......... .......... .......... .......... .......... 80%  152M 0s\n",
            " 45850K .......... .......... .......... .......... .......... 80%  158M 0s\n",
            " 45900K .......... .......... .......... .......... .......... 80%  162M 0s\n",
            " 45950K .......... .......... .......... .......... .......... 80%  135M 0s\n",
            " 46000K .......... .......... .......... .......... .......... 80% 36.4M 0s\n",
            " 46050K .......... .......... .......... .......... .......... 80%  153M 0s\n",
            " 46100K .......... .......... .......... .......... .......... 80%  151M 0s\n",
            " 46150K .......... .......... .......... .......... .......... 80%  147M 0s\n",
            " 46200K .......... .......... .......... .......... .......... 81%  163M 0s\n",
            " 46250K .......... .......... .......... .......... .......... 81%  150M 0s\n",
            " 46300K .......... .......... .......... .......... .......... 81%  164M 0s\n",
            " 46350K .......... .......... .......... .......... .......... 81%  134M 0s\n",
            " 46400K .......... .......... .......... .......... .......... 81%  154M 0s\n",
            " 46450K .......... .......... .......... .......... .......... 81%  165M 0s\n",
            " 46500K .......... .......... .......... .......... .......... 81%  143M 0s\n",
            " 46550K .......... .......... .......... .......... .......... 81%  112M 0s\n",
            " 46600K .......... .......... .......... .......... .......... 81%  138M 0s\n",
            " 46650K .......... .......... .......... .......... .......... 81%  151M 0s\n",
            " 46700K .......... .......... .......... .......... .......... 81%  145M 0s\n",
            " 46750K .......... .......... .......... .......... .......... 81%  127M 0s\n",
            " 46800K .......... .......... .......... .......... .......... 82%  143M 0s\n",
            " 46850K .......... .......... .......... .......... .......... 82%  133M 0s\n",
            " 46900K .......... .......... .......... .......... .......... 82%  141M 0s\n",
            " 46950K .......... .......... .......... .......... .......... 82%  128M 0s\n",
            " 47000K .......... .......... .......... .......... .......... 82%  156M 0s\n",
            " 47050K .......... .......... .......... .......... .......... 82%  141M 0s\n",
            " 47100K .......... .......... .......... .......... .......... 82%  148M 0s\n",
            " 47150K .......... .......... .......... .......... .......... 82%  142M 0s\n",
            " 47200K .......... .......... .......... .......... .......... 82%  158M 0s\n",
            " 47250K .......... .......... .......... .......... .......... 82%  185M 0s\n",
            " 47300K .......... .......... .......... .......... .......... 82%  132M 0s\n",
            " 47350K .......... .......... .......... .......... .......... 83%  141M 0s\n",
            " 47400K .......... .......... .......... .......... .......... 83%  164M 0s\n",
            " 47450K .......... .......... .......... .......... .......... 83%  149M 0s\n",
            " 47500K .......... .......... .......... .......... .......... 83%  166M 0s\n",
            " 47550K .......... .......... .......... .......... .......... 83%  136M 0s\n",
            " 47600K .......... .......... .......... .......... .......... 83%  145M 0s\n",
            " 47650K .......... .......... .......... .......... .......... 83%  164M 0s\n",
            " 47700K .......... .......... .......... .......... .......... 83%  162M 0s\n",
            " 47750K .......... .......... .......... .......... .......... 83%  116M 0s\n",
            " 47800K .......... .......... .......... .......... .......... 83%  152M 0s\n",
            " 47850K .......... .......... .......... .......... .......... 83%  146M 0s\n",
            " 47900K .......... .......... .......... .......... .......... 83%  150M 0s\n",
            " 47950K .......... .......... .......... .......... .......... 84%  140M 0s\n",
            " 48000K .......... .......... .......... .......... .......... 84%  162M 0s\n",
            " 48050K .......... .......... .......... .......... .......... 84% 40.4M 0s\n",
            " 48100K .......... .......... .......... .......... .......... 84%  151M 0s\n",
            " 48150K .......... .......... .......... .......... .......... 84%  144M 0s\n",
            " 48200K .......... .......... .......... .......... .......... 84%  124M 0s\n",
            " 48250K .......... .......... .......... .......... .......... 84%  151M 0s\n",
            " 48300K .......... .......... .......... .......... .......... 84%  164M 0s\n",
            " 48350K .......... .......... .......... .......... .......... 84%  130M 0s\n",
            " 48400K .......... .......... .......... .......... .......... 84%  161M 0s\n",
            " 48450K .......... .......... .......... .......... .......... 84%  143M 0s\n",
            " 48500K .......... .......... .......... .......... .......... 85%  137M 0s\n",
            " 48550K .......... .......... .......... .......... .......... 85%  141M 0s\n",
            " 48600K .......... .......... .......... .......... .......... 85%  138M 0s\n",
            " 48650K .......... .......... .......... .......... .......... 85%  152M 0s\n",
            " 48700K .......... .......... .......... .......... .......... 85%  167M 0s\n",
            " 48750K .......... .......... .......... .......... .......... 85%  136M 0s\n",
            " 48800K .......... .......... .......... .......... .......... 85%  149M 0s\n",
            " 48850K .......... .......... .......... .......... .......... 85%  161M 0s\n",
            " 48900K .......... .......... .......... .......... .......... 85%  156M 0s\n",
            " 48950K .......... .......... .......... .......... .......... 85%  127M 0s\n",
            " 49000K .......... .......... .......... .......... .......... 85%  167M 0s\n",
            " 49050K .......... .......... .......... .......... .......... 85%  157M 0s\n",
            " 49100K .......... .......... .......... .......... .......... 86%  147M 0s\n",
            " 49150K .......... .......... .......... .......... .......... 86% 29.2M 0s\n",
            " 49200K .......... .......... .......... .......... .......... 86%  128M 0s\n",
            " 49250K .......... .......... .......... .......... .......... 86%  159M 0s\n",
            " 49300K .......... .......... .......... .......... .......... 86%  155M 0s\n",
            " 49350K .......... .......... .......... .......... .......... 86%  123M 0s\n",
            " 49400K .......... .......... .......... .......... .......... 86%  137M 0s\n",
            " 49450K .......... .......... .......... .......... .......... 86%  157M 0s\n",
            " 49500K .......... .......... .......... .......... .......... 86%  149M 0s\n",
            " 49550K .......... .......... .......... .......... .......... 86% 18.7M 0s\n",
            " 49600K .......... .......... .......... .......... .......... 86%  126M 0s\n",
            " 49650K .......... .......... .......... .......... .......... 87%  153M 0s\n",
            " 49700K .......... .......... .......... .......... .......... 87%  168M 0s\n",
            " 49750K .......... .......... .......... .......... .......... 87%  155M 0s\n",
            " 49800K .......... .......... .......... .......... .......... 87%  187M 0s\n",
            " 49850K .......... .......... .......... .......... .......... 87%  184M 0s\n",
            " 49900K .......... .......... .......... .......... .......... 87%  189M 0s\n",
            " 49950K .......... .......... .......... .......... .......... 87%  137M 0s\n",
            " 50000K .......... .......... .......... .......... .......... 87%  196M 0s\n",
            " 50050K .......... .......... .......... .......... .......... 87%  184M 0s\n",
            " 50100K .......... .......... .......... .......... .......... 87% 34.3M 0s\n",
            " 50150K .......... .......... .......... .......... .......... 87%  157M 0s\n",
            " 50200K .......... .......... .......... .......... .......... 88%  189M 0s\n",
            " 50250K .......... .......... .......... .......... .......... 88%  150M 0s\n",
            " 50300K .......... .......... .......... .......... .......... 88%  171M 0s\n",
            " 50350K .......... .......... .......... .......... .......... 88%  152M 0s\n",
            " 50400K .......... .......... .......... .......... .......... 88%  152M 0s\n",
            " 50450K .......... .......... .......... .......... .......... 88%  187M 0s\n",
            " 50500K .......... .......... .......... .......... .......... 88%  173M 0s\n",
            " 50550K .......... .......... .......... .......... .......... 88%  162M 0s\n",
            " 50600K .......... .......... .......... .......... .......... 88%  174M 0s\n",
            " 50650K .......... .......... .......... .......... .......... 88%  195M 0s\n",
            " 50700K .......... .......... .......... .......... .......... 88%  187M 0s\n",
            " 50750K .......... .......... .......... .......... .......... 88%  148M 0s\n",
            " 50800K .......... .......... .......... .......... .......... 89%  183M 0s\n",
            " 50850K .......... .......... .......... .......... .......... 89%  191M 0s\n",
            " 50900K .......... .......... .......... .......... .......... 89% 51.3M 0s\n",
            " 50950K .......... .......... .......... .......... .......... 89%  138M 0s\n",
            " 51000K .......... .......... .......... .......... .......... 89%  169M 0s\n",
            " 51050K .......... .......... .......... .......... .......... 89%  189M 0s\n",
            " 51100K .......... .......... .......... .......... .......... 89%  191M 0s\n",
            " 51150K .......... .......... .......... .......... .......... 89% 73.2M 0s\n",
            " 51200K .......... .......... .......... .......... .......... 89%  168M 0s\n",
            " 51250K .......... .......... .......... .......... .......... 89%  190M 0s\n",
            " 51300K .......... .......... .......... .......... .......... 89%  200M 0s\n",
            " 51350K .......... .......... .......... .......... .......... 90%  202M 0s\n",
            " 51400K .......... .......... .......... .......... .......... 90% 50.8M 0s\n",
            " 51450K .......... .......... .......... .......... .......... 90%  158M 0s\n",
            " 51500K .......... .......... .......... .......... .......... 90%  154M 0s\n",
            " 51550K .......... .......... .......... .......... .......... 90%  142M 0s\n",
            " 51600K .......... .......... .......... .......... .......... 90% 18.6M 0s\n",
            " 51650K .......... .......... .......... .......... .......... 90%  115M 0s\n",
            " 51700K .......... .......... .......... .......... .......... 90%  170M 0s\n",
            " 51750K .......... .......... .......... .......... .......... 90%  177M 0s\n",
            " 51800K .......... .......... .......... .......... .......... 90%  180M 0s\n",
            " 51850K .......... .......... .......... .......... .......... 90%  179M 0s\n",
            " 51900K .......... .......... .......... .......... .......... 90%  197M 0s\n",
            " 51950K .......... .......... .......... .......... .......... 91%  143M 0s\n",
            " 52000K .......... .......... .......... .......... .......... 91%  177M 0s\n",
            " 52050K .......... .......... .......... .......... .......... 91%  206M 0s\n",
            " 52100K .......... .......... .......... .......... .......... 91%  303M 0s\n",
            " 52150K .......... .......... .......... .......... .......... 91% 39.0M 0s\n",
            " 52200K .......... .......... .......... .......... .......... 91%  171M 0s\n",
            " 52250K .......... .......... .......... .......... .......... 91%  188M 0s\n",
            " 52300K .......... .......... .......... .......... .......... 91%  201M 0s\n",
            " 52350K .......... .......... .......... .......... .......... 91%  180M 0s\n",
            " 52400K .......... .......... .......... .......... .......... 91%  215M 0s\n",
            " 52450K .......... .......... .......... .......... .......... 91%  217M 0s\n",
            " 52500K .......... .......... .......... .......... .......... 92%  192M 0s\n",
            " 52550K .......... .......... .......... .......... .......... 92%  185M 0s\n",
            " 52600K .......... .......... .......... .......... .......... 92%  142M 0s\n",
            " 52650K .......... .......... .......... .......... .......... 92%  200M 0s\n",
            " 52700K .......... .......... .......... .......... .......... 92%  188M 0s\n",
            " 52750K .......... .......... .......... .......... .......... 92%  115M 0s\n",
            " 52800K .......... .......... .......... .......... .......... 92%  168M 0s\n",
            " 52850K .......... .......... .......... .......... .......... 92%  167M 0s\n",
            " 52900K .......... .......... .......... .......... .......... 92%  173M 0s\n",
            " 52950K .......... .......... .......... .......... .......... 92%  172M 0s\n",
            " 53000K .......... .......... .......... .......... .......... 92%  190M 0s\n",
            " 53050K .......... .......... .......... .......... .......... 92%  176M 0s\n",
            " 53100K .......... .......... .......... .......... .......... 93%  157M 0s\n",
            " 53150K .......... .......... .......... .......... .......... 93%  143M 0s\n",
            " 53200K .......... .......... .......... .......... .......... 93%  195M 0s\n",
            " 53250K .......... .......... .......... .......... .......... 93%  165M 0s\n",
            " 53300K .......... .......... .......... .......... .......... 93%  165M 0s\n",
            " 53350K .......... .......... .......... .......... .......... 93%  102M 0s\n",
            " 53400K .......... .......... .......... .......... .......... 93%  165M 0s\n",
            " 53450K .......... .......... .......... .......... .......... 93%  161M 0s\n",
            " 53500K .......... .......... .......... .......... .......... 93%  180M 0s\n",
            " 53550K .......... .......... .......... .......... .......... 93%  122M 0s\n",
            " 53600K .......... .......... .......... .......... .......... 93%  161M 0s\n",
            " 53650K .......... .......... .......... .......... .......... 94%  165M 0s\n",
            " 53700K .......... .......... .......... .......... .......... 94% 27.9M 0s\n",
            " 53750K .......... .......... .......... .......... .......... 94%  149M 0s\n",
            " 53800K .......... .......... .......... .......... .......... 94%  168M 0s\n",
            " 53850K .......... .......... .......... .......... .......... 94%  184M 0s\n",
            " 53900K .......... .......... .......... .......... .......... 94% 83.0M 0s\n",
            " 53950K .......... .......... .......... .......... .......... 94%  133M 0s\n",
            " 54000K .......... .......... .......... .......... .......... 94%  185M 0s\n",
            " 54050K .......... .......... .......... .......... .......... 94%  157M 0s\n",
            " 54100K .......... .......... .......... .......... .......... 94%  172M 0s\n",
            " 54150K .......... .......... .......... .......... .......... 94% 40.8M 0s\n",
            " 54200K .......... .......... .......... .......... .......... 95%  179M 0s\n",
            " 54250K .......... .......... .......... .......... .......... 95%  191M 0s\n",
            " 54300K .......... .......... .......... .......... .......... 95%  181M 0s\n",
            " 54350K .......... .......... .......... .......... .......... 95% 90.9M 0s\n",
            " 54400K .......... .......... .......... .......... .......... 95%  160M 0s\n",
            " 54450K .......... .......... .......... .......... .......... 95%  159M 0s\n",
            " 54500K .......... .......... .......... .......... .......... 95%  181M 0s\n",
            " 54550K .......... .......... .......... .......... .......... 95%  174M 0s\n",
            " 54600K .......... .......... .......... .......... .......... 95%  179M 0s\n",
            " 54650K .......... .......... .......... .......... .......... 95%  178M 0s\n",
            " 54700K .......... .......... .......... .......... .......... 95%  192M 0s\n",
            " 54750K .......... .......... .......... .......... .......... 95%  143M 0s\n",
            " 54800K .......... .......... .......... .......... .......... 96%  169M 0s\n",
            " 54850K .......... .......... .......... .......... .......... 96%  176M 0s\n",
            " 54900K .......... .......... .......... .......... .......... 96%  182M 0s\n",
            " 54950K .......... .......... .......... .......... .......... 96%  148M 0s\n",
            " 55000K .......... .......... .......... .......... .......... 96% 13.5M 0s\n",
            " 55050K .......... .......... .......... .......... .......... 96%  152M 0s\n",
            " 55100K .......... .......... .......... .......... .......... 96%  187M 0s\n",
            " 55150K .......... .......... .......... .......... .......... 96%  138M 0s\n",
            " 55200K .......... .......... .......... .......... .......... 96%  149M 0s\n",
            " 55250K .......... .......... .......... .......... .......... 96%  179M 0s\n",
            " 55300K .......... .......... .......... .......... .......... 96%  165M 0s\n",
            " 55350K .......... .......... .......... .......... .......... 97%  180M 0s\n",
            " 55400K .......... .......... .......... .......... .......... 97%  164M 0s\n",
            " 55450K .......... .......... .......... .......... .......... 97%  175M 0s\n",
            " 55500K .......... .......... .......... .......... .......... 97%  175M 0s\n",
            " 55550K .......... .......... .......... .......... .......... 97%  134M 0s\n",
            " 55600K .......... .......... .......... .......... .......... 97%  194M 0s\n",
            " 55650K .......... .......... .......... .......... .......... 97%  167M 0s\n",
            " 55700K .......... .......... .......... .......... .......... 97%  190M 0s\n",
            " 55750K .......... .......... .......... .......... .......... 97%  131M 0s\n",
            " 55800K .......... .......... .......... .......... .......... 97%  190M 0s\n",
            " 55850K .......... .......... .......... .......... .......... 97%  182M 0s\n",
            " 55900K .......... .......... .......... .......... .......... 97%  150M 0s\n",
            " 55950K .......... .......... .......... .......... .......... 98%  141M 0s\n",
            " 56000K .......... .......... .......... .......... .......... 98%  188M 0s\n",
            " 56050K .......... .......... .......... .......... .......... 98%  160M 0s\n",
            " 56100K .......... .......... .......... .......... .......... 98%  181M 0s\n",
            " 56150K .......... .......... .......... .......... .......... 98%  150M 0s\n",
            " 56200K .......... .......... .......... .......... .......... 98% 43.7M 0s\n",
            " 56250K .......... .......... .......... .......... .......... 98%  180M 0s\n",
            " 56300K .......... .......... .......... .......... .......... 98%  194M 0s\n",
            " 56350K .......... .......... .......... .......... .......... 98%  166M 0s\n",
            " 56400K .......... .......... .......... .......... .......... 98% 44.7M 0s\n",
            " 56450K .......... .......... .......... .......... .......... 98%  148M 0s\n",
            " 56500K .......... .......... .......... .......... .......... 99%  159M 0s\n",
            " 56550K .......... .......... .......... .......... .......... 99%  155M 0s\n",
            " 56600K .......... .......... .......... .......... .......... 99%  183M 0s\n",
            " 56650K .......... .......... .......... .......... .......... 99%  169M 0s\n",
            " 56700K .......... .......... .......... .......... .......... 99%  200M 0s\n",
            " 56750K .......... .......... .......... .......... .......... 99%  171M 0s\n",
            " 56800K .......... .......... .......... .......... .......... 99%  181M 0s\n",
            " 56850K .......... .......... .......... .......... .......... 99%  184M 0s\n",
            " 56900K .......... .......... .......... .......... .......... 99%  188M 0s\n",
            " 56950K .......... .......... .......... .......... .......... 99%  154M 0s\n",
            " 57000K .......... .......... .......... .......... .......... 99%  176M 0s\n",
            " 57050K .......... .......... .......... .......... ........  100%  192M=0.4s\n",
            "\n",
            "2020-11-28 17:39:14 (124 MB/s) - ‘Miniconda3-4.5.4-Linux-x86_64.sh’ saved [58468498/58468498]\n",
            "\n",
            "Python 3.6.5 :: Anaconda, Inc.\n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "tz3ng7NeaSil"
      },
      "source": [
        "Verificando se o conda foi instalado, qual versão e se modificou a versão do python."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "R3kOHLgJZ3hL",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "a7353111-6570-49b6-aefd-495d9e64fdf0"
      },
      "source": [
        "!which conda\n",
        "!conda --version\n",
        "!python --version"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/usr/local/bin/conda\n",
            "conda 4.5.4\n",
            "Python 3.6.5 :: Anaconda, Inc.\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "T5J0GyNMaJSX"
      },
      "source": [
        "Atualizando o conda.\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "iZU6D7rEaMg_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9eafddfc-bf07-46a5-e929-71f2611f4f37"
      },
      "source": [
        "%%bash\n",
        "conda install --channel defaults conda python=3.6 --yes\n",
        "conda update --channel defaults --all --yes"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Solving environment: ...working... done\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs: \n",
            "    - conda\n",
            "    - python=3.6\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    pycosat-0.6.3              |   py36h27cfd23_0         107 KB\n",
            "    openssl-1.1.1h             |       h7b6447c_0         3.8 MB\n",
            "    chardet-3.0.4              |py36h06a4308_1003         197 KB\n",
            "    cryptography-3.2.1         |   py36h3c74f83_1         631 KB\n",
            "    pycparser-2.20             |             py_2          94 KB\n",
            "    cffi-1.14.3                |   py36h261ae71_2         229 KB\n",
            "    ruamel_yaml-0.15.87        |   py36h7b6447c_1         256 KB\n",
            "    conda-4.9.2                |   py36h06a4308_0         3.1 MB\n",
            "    brotlipy-0.7.0             |py36h27cfd23_1003         349 KB\n",
            "    yaml-0.2.5                 |       h7b6447c_0          87 KB\n",
            "    ncurses-6.2                |       he6710b0_1         1.1 MB\n",
            "    tqdm-4.51.0                |     pyhd3eb1b0_0          66 KB\n",
            "    sqlite-3.33.0              |       h62c20be_0         2.0 MB\n",
            "    libedit-3.1.20191231       |       h14c3975_1         121 KB\n",
            "    setuptools-50.3.1          |   py36h06a4308_1         900 KB\n",
            "    pip-20.2.4                 |   py36h06a4308_0         2.1 MB\n",
            "    pyopenssl-19.1.0           |     pyhd3eb1b0_1          52 KB\n",
            "    libstdcxx-ng-9.1.0         |       hdf63c60_0         4.0 MB\n",
            "    certifi-2020.11.8          |   py36h06a4308_0         150 KB\n",
            "    pysocks-1.7.1              |   py36h06a4308_0          30 KB\n",
            "    xz-5.2.5                   |       h7b6447c_0         438 KB\n",
            "    ld_impl_linux-64-2.33.1    |       h53a641e_7         645 KB\n",
            "    requests-2.24.0            |             py_0          54 KB\n",
            "    zlib-1.2.11                |       h7b6447c_3         120 KB\n",
            "    _libgcc_mutex-0.1          |             main           3 KB\n",
            "    libffi-3.3                 |       he6710b0_2          54 KB\n",
            "    tk-8.6.10                  |       hbc83047_0         3.2 MB\n",
            "    six-1.15.0                 |   py36h06a4308_0          27 KB\n",
            "    wheel-0.35.1               |     pyhd3eb1b0_0          37 KB\n",
            "    conda-package-handling-1.7.2|   py36h03888b9_0         967 KB\n",
            "    python-3.6.12              |       hcff3b4d_2        34.0 MB\n",
            "    idna-2.10                  |             py_0          56 KB\n",
            "    readline-8.0               |       h7b6447c_0         428 KB\n",
            "    urllib3-1.25.11            |             py_0          93 KB\n",
            "    ca-certificates-2020.10.14 |                0         128 KB\n",
            "    libgcc-ng-9.1.0            |       hdf63c60_0         8.1 MB\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        67.6 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "    _libgcc_mutex:          0.1-main               \n",
            "    brotlipy:               0.7.0-py36h27cfd23_1003\n",
            "    conda-package-handling: 1.7.2-py36h03888b9_0   \n",
            "    ld_impl_linux-64:       2.33.1-h53a641e_7      \n",
            "    tqdm:                   4.51.0-pyhd3eb1b0_0    \n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "    ca-certificates:        2018.03.07-0            --> 2020.10.14-0            \n",
            "    certifi:                2018.4.16-py36_0        --> 2020.11.8-py36h06a4308_0\n",
            "    cffi:                   1.11.5-py36h9745a5d_0   --> 1.14.3-py36h261ae71_2   \n",
            "    chardet:                3.0.4-py36h0f667ec_1    --> 3.0.4-py36h06a4308_1003 \n",
            "    conda:                  4.5.4-py36_0            --> 4.9.2-py36h06a4308_0    \n",
            "    cryptography:           2.2.2-py36h14c3975_0    --> 3.2.1-py36h3c74f83_1    \n",
            "    idna:                   2.6-py36h82fb2a8_1      --> 2.10-py_0               \n",
            "    libedit:                3.1.20170329-h6b74fdf_2 --> 3.1.20191231-h14c3975_1 \n",
            "    libffi:                 3.2.1-hd88cf55_4        --> 3.3-he6710b0_2          \n",
            "    libgcc-ng:              7.2.0-hdf63c60_3        --> 9.1.0-hdf63c60_0        \n",
            "    libstdcxx-ng:           7.2.0-hdf63c60_3        --> 9.1.0-hdf63c60_0        \n",
            "    ncurses:                6.1-hf484d3e_0          --> 6.2-he6710b0_1          \n",
            "    openssl:                1.0.2o-h20670df_0       --> 1.1.1h-h7b6447c_0       \n",
            "    pip:                    10.0.1-py36_0           --> 20.2.4-py36h06a4308_0   \n",
            "    pycosat:                0.6.3-py36h0a5515d_0    --> 0.6.3-py36h27cfd23_0    \n",
            "    pycparser:              2.18-py36hf9f622e_1     --> 2.20-py_2               \n",
            "    pyopenssl:              18.0.0-py36_0           --> 19.1.0-pyhd3eb1b0_1     \n",
            "    pysocks:                1.6.8-py36_0            --> 1.7.1-py36h06a4308_0    \n",
            "    python:                 3.6.5-hc3d631a_2        --> 3.6.12-hcff3b4d_2       \n",
            "    readline:               7.0-ha6073c6_4          --> 8.0-h7b6447c_0          \n",
            "    requests:               2.18.4-py36he2e5f8d_1   --> 2.24.0-py_0             \n",
            "    ruamel_yaml:            0.15.37-py36h14c3975_2  --> 0.15.87-py36h7b6447c_1  \n",
            "    setuptools:             39.2.0-py36_0           --> 50.3.1-py36h06a4308_1   \n",
            "    six:                    1.11.0-py36h372c433_1   --> 1.15.0-py36h06a4308_0   \n",
            "    sqlite:                 3.23.1-he433501_0       --> 3.33.0-h62c20be_0       \n",
            "    tk:                     8.6.7-hc745277_3        --> 8.6.10-hbc83047_0       \n",
            "    urllib3:                1.22-py36hbe7ace6_0     --> 1.25.11-py_0            \n",
            "    wheel:                  0.31.1-py36_0           --> 0.35.1-pyhd3eb1b0_0     \n",
            "    xz:                     5.2.4-h14c3975_4        --> 5.2.5-h7b6447c_0        \n",
            "    yaml:                   0.1.7-had09818_2        --> 0.2.5-h7b6447c_0        \n",
            "    zlib:                   1.2.11-ha838bed_2       --> 1.2.11-h7b6447c_3       \n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "Preparing transaction: ...working... done\n",
            "Verifying transaction: ...working... done\n",
            "Executing transaction: ...working... done\n",
            "Collecting package metadata (current_repodata.json): ...working... done\n",
            "Solving environment: ...working... done\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "\n",
            "The following packages will be REMOVED:\n",
            "\n",
            "  asn1crypto-0.24.0-py36_0\n",
            "  conda-env-2.6.0-h36134e3_1\n",
            "\n",
            "\n",
            "Preparing transaction: ...working... done\n",
            "Verifying transaction: ...working... done\n",
            "Executing transaction: ...working... done\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "stream",
          "text": [
            "\rpycosat-0.6.3        |  107 KB |            |   0% \rpycosat-0.6.3        |  107 KB | ########## | 100% \n",
            "\ropenssl-1.1.1h       |  3.8 MB |            |   0% \ropenssl-1.1.1h       |  3.8 MB | #######7   |  77% \ropenssl-1.1.1h       |  3.8 MB | #########7 |  97% \ropenssl-1.1.1h       |  3.8 MB | ########## | 100% \n",
            "\rchardet-3.0.4        |  197 KB |            |   0% \rchardet-3.0.4        |  197 KB | #########9 |  99% \rchardet-3.0.4        |  197 KB | ########## | 100% \n",
            "\rcryptography-3.2.1   |  631 KB |            |   0% \rcryptography-3.2.1   |  631 KB | #########  |  90% \rcryptography-3.2.1   |  631 KB | ########## | 100% \n",
            "\rpycparser-2.20       |   94 KB |            |   0% \rpycparser-2.20       |   94 KB | ########## | 100% \n",
            "\rcffi-1.14.3          |  229 KB |            |   0% \rcffi-1.14.3          |  229 KB | ########## | 100% \n",
            "\rruamel_yaml-0.15.87  |  256 KB |            |   0% \rruamel_yaml-0.15.87  |  256 KB | #########6 |  97% \rruamel_yaml-0.15.87  |  256 KB | ########## | 100% \n",
            "\rconda-4.9.2          |  3.1 MB |            |   0% \rconda-4.9.2          |  3.1 MB | #######9   |  79% \rconda-4.9.2          |  3.1 MB | #########2 |  93% \rconda-4.9.2          |  3.1 MB | ########## | 100% \n",
            "\rbrotlipy-0.7.0       |  349 KB |            |   0% \rbrotlipy-0.7.0       |  349 KB | ########## | 100% \n",
            "\ryaml-0.2.5           |   87 KB |            |   0% \ryaml-0.2.5           |   87 KB | ########## | 100% \n",
            "\rncurses-6.2          |  1.1 MB |            |   0% \rncurses-6.2          |  1.1 MB | #######8   |  78% \rncurses-6.2          |  1.1 MB | #########9 | 100% \rncurses-6.2          |  1.1 MB | ########## | 100% \n",
            "\rtqdm-4.51.0          |   66 KB |            |   0% \rtqdm-4.51.0          |   66 KB | ########## | 100% \n",
            "\rsqlite-3.33.0        |  2.0 MB |            |   0% \rsqlite-3.33.0        |  2.0 MB | #######9   |  79% \rsqlite-3.33.0        |  2.0 MB | ########## | 100% \n",
            "\rlibedit-3.1.20191231 |  121 KB |            |   0% \rlibedit-3.1.20191231 |  121 KB | ########## | 100% \n",
            "\rsetuptools-50.3.1    |  900 KB |            |   0% \rsetuptools-50.3.1    |  900 KB | ########1  |  82% \rsetuptools-50.3.1    |  900 KB | #########6 |  96% \rsetuptools-50.3.1    |  900 KB | ########## | 100% \n",
            "\rpip-20.2.4           |  2.1 MB |            |   0% \rpip-20.2.4           |  2.1 MB | #######7   |  77% \rpip-20.2.4           |  2.1 MB | #########1 |  91% \rpip-20.2.4           |  2.1 MB | ########## | 100% \n",
            "\rpyopenssl-19.1.0     |   52 KB |            |   0% \rpyopenssl-19.1.0     |   52 KB | ########## | 100% \n",
            "\rlibstdcxx-ng-9.1.0   |  4.0 MB |            |   0% \rlibstdcxx-ng-9.1.0   |  4.0 MB | #######6   |  77% \rlibstdcxx-ng-9.1.0   |  4.0 MB | #########9 | 100% \rlibstdcxx-ng-9.1.0   |  4.0 MB | ########## | 100% \n",
            "\rcertifi-2020.11.8    |  150 KB |            |   0% \rcertifi-2020.11.8    |  150 KB | ########## | 100% \n",
            "\rpysocks-1.7.1        |   30 KB |            |   0% \rpysocks-1.7.1        |   30 KB | ########## | 100% \n",
            "\rxz-5.2.5             |  438 KB |            |   0% \rxz-5.2.5             |  438 KB | #########  |  90% \rxz-5.2.5             |  438 KB | ########## | 100% \n",
            "\rld_impl_linux-64-2.3 |  645 KB |            |   0% \rld_impl_linux-64-2.3 |  645 KB | #########4 |  95% \rld_impl_linux-64-2.3 |  645 KB | ########## | 100% \n",
            "\rrequests-2.24.0      |   54 KB |            |   0% \rrequests-2.24.0      |   54 KB | ########## | 100% \n",
            "\rzlib-1.2.11          |  120 KB |            |   0% \rzlib-1.2.11          |  120 KB | ########## | 100% \n",
            "\r_libgcc_mutex-0.1    |    3 KB |            |   0% \r_libgcc_mutex-0.1    |    3 KB | ########## | 100% \n",
            "\rlibffi-3.3           |   54 KB |            |   0% \rlibffi-3.3           |   54 KB | ########## | 100% \n",
            "\rtk-8.6.10            |  3.2 MB |            |   0% \rtk-8.6.10            |  3.2 MB | #######7   |  78% \rtk-8.6.10            |  3.2 MB | #########9 | 100% \rtk-8.6.10            |  3.2 MB | ########## | 100% \n",
            "\rsix-1.15.0           |   27 KB |            |   0% \rsix-1.15.0           |   27 KB | ########## | 100% \n",
            "\rwheel-0.35.1         |   37 KB |            |   0% \rwheel-0.35.1         |   37 KB | ########## | 100% \n",
            "\rconda-package-handli |  967 KB |            |   0% \rconda-package-handli |  967 KB | ########7  |  88% \rconda-package-handli |  967 KB | ########## | 100% \n",
            "\rpython-3.6.12        | 34.0 MB |            |   0% \rpython-3.6.12        | 34.0 MB | ##6        |  27% \rpython-3.6.12        | 34.0 MB | ######1    |  61% \rpython-3.6.12        | 34.0 MB | #######7   |  78% \rpython-3.6.12        | 34.0 MB | ########9  |  89% \rpython-3.6.12        | 34.0 MB | #########7 |  98% \rpython-3.6.12        | 34.0 MB | ########## | 100% \n",
            "\ridna-2.10            |   56 KB |            |   0% \ridna-2.10            |   56 KB | ########## | 100% \n",
            "\rreadline-8.0         |  428 KB |            |   0% \rreadline-8.0         |  428 KB | ########## | 100% \n",
            "\rurllib3-1.25.11      |   93 KB |            |   0% \rurllib3-1.25.11      |   93 KB | ########## | 100% \n",
            "\rca-certificates-2020 |  128 KB |            |   0% \rca-certificates-2020 |  128 KB | ########## | 100% \n",
            "\rlibgcc-ng-9.1.0      |  8.1 MB |            |   0% \rlibgcc-ng-9.1.0      |  8.1 MB | #######5   |  76% \rlibgcc-ng-9.1.0      |  8.1 MB | #########2 |  93% \rlibgcc-ng-9.1.0      |  8.1 MB | ########## | 100% \n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "k7kkzU0BazDs"
      },
      "source": [
        "Versao do conda e do python mudaram novamente."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "DhJh4M4daiG_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "aa3f11f2-2d4e-4fb4-a7cb-1992790c091c"
      },
      "source": [
        "!conda --version\n",
        "!python --version"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "conda 4.9.2\n",
            "Python 3.6.12 :: Anaconda, Inc.\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "UluNE7TgbE4n"
      },
      "source": [
        "Enviando o conda para o sistema"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "R7cxFwH-axxJ",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "603430d2-1140-46bb-ced9-abd27f032f06"
      },
      "source": [
        "import sys\n",
        "\n",
        "sys.path"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['',\n",
              " '/env/python',\n",
              " '/usr/lib/python36.zip',\n",
              " '/usr/lib/python3.6',\n",
              " '/usr/lib/python3.6/lib-dynload',\n",
              " '/usr/local/lib/python3.6/dist-packages',\n",
              " '/usr/lib/python3/dist-packages',\n",
              " '/usr/local/lib/python3.6/dist-packages/IPython/extensions',\n",
              " '/root/.ipython',\n",
              " '/usr/local/lib/python3.6/site-packages',\n",
              " '/usr/local/lib/python3.6/site-packages',\n",
              " '/usr/local/lib/python3.6/site-packages']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "t1youLXXbOqN"
      },
      "source": [
        "Verificar os pacotes pre-instalados do conda"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "uQ6qPBvmbSU_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "5fd10170-2fe0-4fa6-a04f-8d7d0a2255e8"
      },
      "source": [
        "!ls /usr/local/lib/python3.6/dist-packages\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "absl\n",
            "absl_py-0.10.0.dist-info\n",
            "alabaster\n",
            "alabaster-0.7.12.dist-info\n",
            "albumentations\n",
            "albumentations-0.1.12.dist-info\n",
            "altair\n",
            "altair-4.1.0.dist-info\n",
            "apiclient\n",
            "argon2\n",
            "argon2_cffi-20.1.0.dist-info\n",
            "asgiref\n",
            "asgiref-3.3.1.dist-info\n",
            "astor\n",
            "astor-0.8.1.dist-info\n",
            "astropy\n",
            "astropy-4.1.dist-info\n",
            "astunparse\n",
            "astunparse-1.6.3.dist-info\n",
            "async_generator\n",
            "async_generator-1.10.dist-info\n",
            "atari_py\n",
            "atari_py-0.2.6.dist-info\n",
            "atomicwrites\n",
            "atomicwrites-1.4.0.dist-info\n",
            "attr\n",
            "attrs-20.3.0.dist-info\n",
            "audioread\n",
            "audioread-2.1.9.dist-info\n",
            "autograd\n",
            "autograd-1.3.dist-info\n",
            "babel\n",
            "Babel-2.9.0.dist-info\n",
            "backcall\n",
            "backcall-0.2.0.dist-info\n",
            "beautifulsoup4-4.6.3.dist-info\n",
            "bin\n",
            "bleach\n",
            "bleach-3.2.1.dist-info\n",
            "blis\n",
            "blis-0.4.1.dist-info\n",
            "bokeh\n",
            "bokeh-2.1.1.dist-info\n",
            "bottleneck\n",
            "Bottleneck-1.3.2.dist-info\n",
            "branca\n",
            "branca-0.4.1.dist-info\n",
            "bs4\n",
            "bs4-0.0.1.dist-info\n",
            "bson\n",
            "cachecontrol\n",
            "CacheControl-0.12.6.dist-info\n",
            "cachetools\n",
            "cachetools-4.1.1.dist-info\n",
            "caffe2\n",
            "catalogue-1.0.0.dist-info\n",
            "catalogue.py\n",
            "certifi\n",
            "certifi-2020.11.8.dist-info\n",
            "cffi\n",
            "cffi-1.14.3.dist-info\n",
            "_cffi_backend.cpython-36m-x86_64-linux-gnu.so\n",
            "cffi.libs\n",
            "chainer\n",
            "chainer-7.4.0.dist-info\n",
            "chainermn\n",
            "chainerx\n",
            "chardet\n",
            "chardet-3.0.4.dist-info\n",
            "chess\n",
            "click\n",
            "click-7.1.2.dist-info\n",
            "client\n",
            "cloudpickle\n",
            "cloudpickle-1.3.0.dist-info\n",
            "cmake\n",
            "cmake-3.12.0.dist-info\n",
            "cmdstanpy\n",
            "cmdstanpy-0.9.5.dist-info\n",
            "colab\n",
            "colorlover\n",
            "colorlover-0.3.0.dist-info\n",
            "community\n",
            "community-1.0.0b1.dist-info\n",
            "contextlib2-0.5.5.dist-info\n",
            "contextlib2.py\n",
            "convertdate\n",
            "convertdate-2.3.0.dist-info\n",
            "coverage\n",
            "coverage-3.7.1.dist-info\n",
            "coveralls\n",
            "coveralls-0.5.dist-info\n",
            "crcmod\n",
            "crcmod-1.7.dist-info\n",
            "cufflinks\n",
            "cufflinks-0.17.3.dist-info\n",
            "cv2\n",
            "_cvxcore.cpython-36m-x86_64-linux-gnu.so\n",
            "cvxopt\n",
            "cvxopt-1.2.5.dist-info\n",
            "cvxopt.libs\n",
            "cvxpy\n",
            "cvxpy-1.0.31.dist-info\n",
            "cycler-0.10.0.dist-info\n",
            "cycler.py\n",
            "cymem\n",
            "cymem-2.0.4.dist-info\n",
            "Cython\n",
            "Cython-0.29.21.dist-info\n",
            "cython.py\n",
            "daft-0.0.4.dist-info\n",
            "daft.py\n",
            "dask\n",
            "dask-2.12.0.dist-info\n",
            "dataclasses-0.8.dist-info\n",
            "dataclasses.py\n",
            "datascience\n",
            "datascience-0.10.6.dist-info\n",
            "dateutil\n",
            "debugpy\n",
            "debugpy-1.0.0.dist-info\n",
            "decorator-4.4.2.dist-info\n",
            "decorator.py\n",
            "defusedxml\n",
            "defusedxml-0.6.0.dist-info\n",
            "descartes\n",
            "descartes-1.1.0.dist-info\n",
            "dill\n",
            "dill-0.3.3.dist-info\n",
            "distributed\n",
            "distributed-1.25.3.dist-info\n",
            "_distutils_hack\n",
            "distutils-precedence.pth\n",
            "django\n",
            "Django-3.1.3.dist-info\n",
            "dlib-19.18.0.dist-info\n",
            "dlib.cpython-36m-x86_64-linux-gnu.so\n",
            "dm_tree-0.1.5.dist-info\n",
            "docopt-0.6.2.dist-info\n",
            "docopt.py\n",
            "docs\n",
            "docutils\n",
            "docutils-0.16.dist-info\n",
            "dopamine\n",
            "dopamine_rl-1.0.5.dist-info\n",
            "dot_parser.py\n",
            "earthengine_api-0.1.238.dist-info\n",
            "easydict\n",
            "easydict-1.9.dist-info\n",
            "easy_install.py\n",
            "ecos\n",
            "ecos-2.0.7.post1.dist-info\n",
            "_ecos.cpython-36m-x86_64-linux-gnu.so\n",
            "editdistance\n",
            "editdistance-0.5.3.dist-info\n",
            "ee\n",
            "en_core_web_sm\n",
            "en_core_web_sm-2.2.5.dist-info\n",
            "entrypoints-0.3.dist-info\n",
            "entrypoints.py\n",
            "ephem\n",
            "ephem-3.7.7.1.dist-info\n",
            "et_xmlfile\n",
            "et_xmlfile-1.0.1.dist-info\n",
            "examples\n",
            "fa2\n",
            "fa2-0.3.5.dist-info\n",
            "fancyimpute\n",
            "fancyimpute-0.4.3.dist-info\n",
            "fastai\n",
            "fastai-1.0.61.dist-info\n",
            "fastdtw\n",
            "fastdtw-0.3.4.dist-info\n",
            "fastprogress\n",
            "fastprogress-1.0.0.dist-info\n",
            "fastrlock\n",
            "fastrlock-0.5.dist-info\n",
            "fbprophet\n",
            "fbprophet-0.7.1-py3.6.egg-info\n",
            "feather\n",
            "feather_format-0.4.1.dist-info\n",
            "filelock-3.0.12.dist-info\n",
            "filelock.py\n",
            "firebase_admin\n",
            "firebase_admin-4.4.0.dist-info\n",
            "fix_yahoo_finance\n",
            "fix_yahoo_finance-0.0.22.dist-info\n",
            "flask\n",
            "Flask-1.1.2.dist-info\n",
            "flatbuffers\n",
            "flatbuffers-1.12.dist-info\n",
            "folium\n",
            "folium-0.8.3.dist-info\n",
            "_foo.cpython-36m-x86_64-linux-gnu.so\n",
            "future\n",
            "future-0.16.0.dist-info\n",
            "gast\n",
            "gast-0.3.3.dist-info\n",
            "gdown\n",
            "gdown-3.6.4.dist-info\n",
            "gensim\n",
            "gensim-3.6.0.dist-info\n",
            "geographiclib\n",
            "geographiclib-1.50.dist-info\n",
            "geopy\n",
            "geopy-1.17.0.dist-info\n",
            "gin\n",
            "gin_config-0.3.0.dist-info\n",
            "github2pypi\n",
            "glob2\n",
            "glob2-0.7.dist-info\n",
            "google\n",
            "google-2.0.3.dist-info\n",
            "googleapiclient\n",
            "google_api_core-1.16.0.dist-info\n",
            "google_api_core-1.16.0-py3.8-nspkg.pth\n",
            "google_api_python_client-1.7.12.dist-info\n",
            "googleapis_common_protos-1.52.0.dist-info\n",
            "googleapis_common_protos-1.52.0-py3.8-nspkg.pth\n",
            "google_auth-1.17.2.dist-info\n",
            "google_auth-1.17.2-py3.8-nspkg.pth\n",
            "google_auth_httplib2-0.0.4.dist-info\n",
            "google_auth_httplib2.py\n",
            "google_auth_oauthlib\n",
            "google_auth_oauthlib-0.4.2.dist-info\n",
            "google_cloud_bigquery-1.21.0.dist-info\n",
            "google_cloud_bigquery-1.21.0-py3.6-nspkg.pth\n",
            "google_cloud_bigquery_storage-1.1.0.dist-info\n",
            "google_cloud_bigquery_storage-1.1.0-py3.8-nspkg.pth\n",
            "google_cloud_core-1.0.3.dist-info\n",
            "google_cloud_core-1.0.3-py3.6-nspkg.pth\n",
            "google_cloud_datastore-1.8.0.dist-info\n",
            "google_cloud_datastore-1.8.0-py3.6-nspkg.pth\n",
            "google_cloud_firestore-1.7.0.dist-info\n",
            "google_cloud_firestore-1.7.0-py3.8-nspkg.pth\n",
            "google_cloud_language-1.2.0.dist-info\n",
            "google_cloud_language-1.2.0-py3.6-nspkg.pth\n",
            "google_cloud_storage-1.18.1.dist-info\n",
            "google_cloud_storage-1.18.1-py3.7-nspkg.pth\n",
            "google_cloud_translate-1.5.0.dist-info\n",
            "google_cloud_translate-1.5.0-py3.6-nspkg.pth\n",
            "google_colab-1.0.0.dist-info\n",
            "google_colab-1.0.0-py3.6-nspkg.pth\n",
            "google_drive_downloader\n",
            "googledrivedownloader-0.4.dist-info\n",
            "google_pasta-0.2.0.dist-info\n",
            "google_resumable_media-0.4.1.dist-info\n",
            "google_resumable_media-0.4.1-py3.6-nspkg.pth\n",
            "googlesearch\n",
            "graphviz\n",
            "graphviz-0.10.1.dist-info\n",
            "gridfs\n",
            "grpc\n",
            "grpcio-1.33.2.dist-info\n",
            "gspread\n",
            "gspread-3.0.1.dist-info\n",
            "gspread_dataframe-3.0.8.dist-info\n",
            "gspread_dataframe.py\n",
            "gym\n",
            "gym-0.17.3.dist-info\n",
            "h5py\n",
            "h5py-2.10.0.dist-info\n",
            "HeapDict-1.0.1.dist-info\n",
            "heapdict.py\n",
            "helper\n",
            "holidays\n",
            "holidays-0.10.3.dist-info\n",
            "holoviews\n",
            "holoviews-1.13.5.dist-info\n",
            "html5lib\n",
            "html5lib-1.0.1.dist-info\n",
            "httpimport-0.5.18.dist-info\n",
            "httpimport.py\n",
            "httplib2\n",
            "httplib2-0.17.4.dist-info\n",
            "httplib2shim\n",
            "httplib2shim-0.0.3.dist-info\n",
            "humanize\n",
            "humanize-0.5.1.dist-info\n",
            "hyperopt\n",
            "hyperopt-0.1.2.dist-info\n",
            "ideep4py\n",
            "ideep4py-2.0.0.post3.dist-info\n",
            "idna\n",
            "idna-2.10.dist-info\n",
            "image\n",
            "image-1.5.33.dist-info\n",
            "imageio\n",
            "imageio-2.4.1.dist-info\n",
            "imagesize-1.2.0.dist-info\n",
            "imagesize.py\n",
            "imbalanced_learn-0.4.3.dist-info\n",
            "imblearn\n",
            "imblearn-0.0.dist-info\n",
            "imgaug\n",
            "imgaug-0.2.9.dist-info\n",
            "importlib_metadata\n",
            "importlib_metadata-2.0.0.dist-info\n",
            "importlib_resources\n",
            "importlib_resources-3.3.0.dist-info\n",
            "imutils\n",
            "imutils-0.5.3.dist-info\n",
            "inflect-2.1.0.dist-info\n",
            "inflect.py\n",
            "iniconfig\n",
            "iniconfig-1.1.1.dist-info\n",
            "intel_openmp-2020.0.133.dist-info\n",
            "intervaltree\n",
            "intervaltree-2.1.0.dist-info\n",
            "ipykernel\n",
            "ipykernel-4.10.1.dist-info\n",
            "ipykernel_launcher.py\n",
            "IPython\n",
            "ipython-5.5.0.dist-info\n",
            "ipython_genutils\n",
            "ipython_genutils-0.2.0.dist-info\n",
            "ipython_sql-0.3.9.dist-info\n",
            "ipywidgets\n",
            "ipywidgets-7.5.1.dist-info\n",
            "itsdangerous\n",
            "itsdangerous-1.1.0.dist-info\n",
            "jax\n",
            "jax-0.2.6.dist-info\n",
            "jaxlib\n",
            "jaxlib-0.1.57+cuda101.dist-info\n",
            "jdcal-1.4.1.dist-info\n",
            "jdcal.py\n",
            "jedi\n",
            "jedi-0.17.2.dist-info\n",
            "jieba\n",
            "jieba-0.42.1.dist-info\n",
            "jinja2\n",
            "Jinja2-2.11.2.dist-info\n",
            "joblib\n",
            "joblib-0.17.0.dist-info\n",
            "jpeg4py\n",
            "jpeg4py-0.1.4.dist-info\n",
            "jsonschema\n",
            "jsonschema-2.6.0.dist-info\n",
            "jupyter-1.0.0.dist-info\n",
            "jupyter_client\n",
            "jupyter_client-5.3.5.dist-info\n",
            "jupyter_console\n",
            "jupyter_console-5.2.0.dist-info\n",
            "jupyter_core\n",
            "jupyter_core-4.7.0.dist-info\n",
            "jupyterlab_pygments\n",
            "jupyterlab_pygments-0.1.2.dist-info\n",
            "jupyter.py\n",
            "kaggle\n",
            "kaggle-1.5.9.dist-info\n",
            "kapre\n",
            "kapre-0.1.3.1.dist-info\n",
            "keras\n",
            "Keras-2.4.3.dist-info\n",
            "keras_preprocessing\n",
            "Keras_Preprocessing-1.1.2.dist-info\n",
            "keras_vis-0.4.1.dist-info\n",
            "kiwisolver-1.3.1.dist-info\n",
            "kiwisolver.cpython-36m-x86_64-linux-gnu.so\n",
            "knnimpute\n",
            "knnimpute-0.1.0.dist-info\n",
            "korean_lunar_calendar\n",
            "korean_lunar_calendar-0.2.1.dist-info\n",
            "libfuturize\n",
            "libpasteurize\n",
            "librosa\n",
            "librosa-0.6.3.dist-info\n",
            "lightgbm\n",
            "lightgbm-2.2.3.dist-info\n",
            "llvmlite\n",
            "llvmlite-0.31.0.dist-info\n",
            "lmdb\n",
            "lmdb-0.99.dist-info\n",
            "lucid\n",
            "lucid-0.3.8.dist-info\n",
            "lunarcalendar\n",
            "LunarCalendar-0.0.9.dist-info\n",
            "lxml\n",
            "lxml-4.2.6.dist-info\n",
            "markdown\n",
            "Markdown-3.3.3.dist-info\n",
            "markupsafe\n",
            "MarkupSafe-1.1.1.dist-info\n",
            "matplotlib\n",
            "matplotlib-3.2.2.dist-info\n",
            "matplotlib-3.2.2-py3.6-nspkg.pth\n",
            "matplotlib.libs\n",
            "matplotlib_venn\n",
            "matplotlib_venn-0.11.6.dist-info\n",
            "missingno\n",
            "missingno-0.4.2.dist-info\n",
            "mistune-0.8.4.dist-info\n",
            "mistune.py\n",
            "mizani\n",
            "mizani-0.6.0.dist-info\n",
            "mkl-2019.0.dist-info\n",
            "mlxtend\n",
            "mlxtend-0.14.0.dist-info\n",
            "more_itertools\n",
            "more_itertools-8.6.0.dist-info\n",
            "moviepy\n",
            "moviepy-0.2.3.5.dist-info\n",
            "mpl_toolkits\n",
            "mpmath\n",
            "mpmath-1.1.0.dist-info\n",
            "msgpack\n",
            "msgpack-1.0.0.dist-info\n",
            "multiprocess\n",
            "_multiprocess\n",
            "multiprocess-0.70.11.1.dist-info\n",
            "multitasking\n",
            "multitasking-0.0.9.dist-info\n",
            "murmurhash\n",
            "murmurhash-1.0.4.dist-info\n",
            "music21\n",
            "music21-5.5.0.dist-info\n",
            "natsort\n",
            "natsort-5.5.0.dist-info\n",
            "nbclient\n",
            "nbclient-0.5.1.dist-info\n",
            "nbconvert\n",
            "nbconvert-5.6.1.dist-info\n",
            "nbformat\n",
            "nbformat-5.0.8.dist-info\n",
            "nest_asyncio-1.4.3.dist-info\n",
            "nest_asyncio.py\n",
            "networkx\n",
            "networkx-2.5.dist-info\n",
            "nibabel\n",
            "nibabel-3.0.2.dist-info\n",
            "nisext\n",
            "nltk\n",
            "nltk-3.2.5.dist-info\n",
            "notebook\n",
            "notebook-5.3.1.dist-info\n",
            "np_utils\n",
            "np_utils-0.5.12.1.dist-info\n",
            "numba\n",
            "numba-0.48.0.dist-info\n",
            "numbergen\n",
            "numexpr\n",
            "numexpr-2.7.1.dist-info\n",
            "numpy\n",
            "numpy-1.18.5.dist-info\n",
            "numpy.libs\n",
            "nvidia_ml_py3-7.352.0.dist-info\n",
            "nvidia_smi.py\n",
            "oauth2client\n",
            "oauth2client-4.1.3.dist-info\n",
            "oauthlib\n",
            "oauthlib-3.1.0.dist-info\n",
            "okgrade\n",
            "okgrade-0.4.3.dist-info\n",
            "onnx_chainer\n",
            "opencv_contrib_python-4.1.2.30.dist-info\n",
            "opencv_python-4.1.2.30.dist-info\n",
            "OpenGL\n",
            "openpyxl\n",
            "openpyxl-2.5.9.dist-info\n",
            "opt_einsum\n",
            "opt_einsum-3.3.0.dist-info\n",
            "osqp\n",
            "osqp-0.6.1.dist-info\n",
            "osqppurepy\n",
            "packaging\n",
            "packaging-20.4.dist-info\n",
            "palettable\n",
            "palettable-3.3.0.dist-info\n",
            "pandas\n",
            "pandas-1.1.4.dist-info\n",
            "pandas_datareader\n",
            "pandas_datareader-0.9.0.dist-info\n",
            "pandas_gbq\n",
            "pandas_gbq-0.13.3.dist-info\n",
            "pandas_profiling\n",
            "pandas_profiling-1.4.1.dist-info\n",
            "pandocfilters-1.4.3.dist-info\n",
            "pandocfilters.py\n",
            "panel\n",
            "panel-0.9.7.dist-info\n",
            "param\n",
            "param-1.10.0.dist-info\n",
            "parso\n",
            "parso-0.7.1.dist-info\n",
            "past\n",
            "pasta\n",
            "pathlib-1.0.1.dist-info\n",
            "pathlib.py\n",
            "patsy\n",
            "patsy-0.5.1.dist-info\n",
            "pexpect\n",
            "pexpect-4.8.0.dist-info\n",
            "pickleshare-0.7.5.dist-info\n",
            "pickleshare.py\n",
            "PIL\n",
            "Pillow-7.0.0.dist-info\n",
            "pip\n",
            "pip-19.3.1.dist-info\n",
            "piptools\n",
            "pip_tools-4.5.1.dist-info\n",
            "pkg_resources\n",
            "plac-1.1.3.dist-info\n",
            "plac_core.py\n",
            "plac_ext.py\n",
            "plac.py\n",
            "plac_tk.py\n",
            "plotly\n",
            "plotly-4.4.1.dist-info\n",
            "_plotly_future_\n",
            "_plotly_utils\n",
            "plotlywidget\n",
            "plotnine\n",
            "plotnine-0.6.0.dist-info\n",
            "pluggy\n",
            "pluggy-0.7.1.dist-info\n",
            "portpicker-1.3.1.dist-info\n",
            "portpicker.py\n",
            "prefetch_generator\n",
            "prefetch_generator-1.0.1.dist-info\n",
            "preshed\n",
            "preshed-3.0.4.dist-info\n",
            "prettytable\n",
            "prettytable-2.0.0.dist-info\n",
            "progressbar\n",
            "progressbar2-3.38.0.dist-info\n",
            "prometheus_client\n",
            "prometheus_client-0.9.0.dist-info\n",
            "promise\n",
            "promise-2.3.dist-info\n",
            "prompt_toolkit\n",
            "prompt_toolkit-1.0.18.dist-info\n",
            "protobuf-3.12.4.dist-info\n",
            "protobuf-3.12.4-py3.6-nspkg.pth\n",
            "psutil\n",
            "psutil-5.4.8.dist-info\n",
            "psycopg2\n",
            "psycopg2-2.7.6.1.dist-info\n",
            "ptyprocess\n",
            "ptyprocess-0.6.0.dist-info\n",
            "pvectorc.cpython-36m-x86_64-linux-gnu.so\n",
            "py\n",
            "py-1.9.0.dist-info\n",
            "pyarrow\n",
            "pyarrow-0.14.1.dist-info\n",
            "pyasn1\n",
            "pyasn1-0.4.8.dist-info\n",
            "pyasn1_modules\n",
            "pyasn1_modules-0.2.8.dist-info\n",
            "__pycache__\n",
            "pycocotools\n",
            "pycocotools-2.0.2.dist-info\n",
            "pycparser\n",
            "pycparser-2.20.dist-info\n",
            "pyct\n",
            "pyct-0.4.8.dist-info\n",
            "pydata_google_auth\n",
            "pydata_google_auth-1.1.0.dist-info\n",
            "pydot-1.3.0.dist-info\n",
            "pydot_ng\n",
            "pydot_ng-2.0.0.dist-info\n",
            "pydotplus\n",
            "pydotplus-2.0.2.dist-info\n",
            "pydot.py\n",
            "pydrive\n",
            "PyDrive-1.3.1.dist-info\n",
            "pyemd\n",
            "pyemd-0.5.1.dist-info\n",
            "pyglet\n",
            "pyglet-1.5.0.dist-info\n",
            "pygments\n",
            "Pygments-2.6.1.dist-info\n",
            "pylab.py\n",
            "pymc3\n",
            "pymc3-3.7.dist-info\n",
            "pymeeus\n",
            "PyMeeus-0.3.7.dist-info\n",
            "pymongo\n",
            "pymongo-3.11.1.dist-info\n",
            "pymystem3\n",
            "pymystem3-0.2.0.dist-info\n",
            "pynvml.py\n",
            "PyOpenGL-3.1.5.dist-info\n",
            "pyparsing-2.4.7.dist-info\n",
            "pyparsing.py\n",
            "pyrsistent\n",
            "pyrsistent-0.17.3.dist-info\n",
            "_pyrsistent_version.py\n",
            "pysndfile\n",
            "pysndfile-1.3.8.dist-info\n",
            "PySocks-1.7.1.dist-info\n",
            "pystan\n",
            "pystan-2.19.1.1.dist-info\n",
            "_pytest\n",
            "pytest-3.6.4.dist-info\n",
            "pytest.py\n",
            "python_chess-0.23.11.dist-info\n",
            "python_dateutil-2.8.1.dist-info\n",
            "python_louvain-0.14.dist-info\n",
            "python_slugify-4.0.1.dist-info\n",
            "python_utils\n",
            "python_utils-2.4.0.dist-info\n",
            "pytz\n",
            "pytz-2018.9.dist-info\n",
            "pyviz_comms\n",
            "pyviz_comms-0.7.6.dist-info\n",
            "PyWavelets-1.1.1.dist-info\n",
            "pywt\n",
            "pyximport\n",
            "PyYAML-3.13.dist-info\n",
            "pyzmq-20.0.0.dist-info\n",
            "pyzmq.libs\n",
            "qtconsole\n",
            "qtconsole-4.7.7.dist-info\n",
            "qtpy\n",
            "QtPy-1.9.0.dist-info\n",
            "regex\n",
            "regex-2019.12.20.dist-info\n",
            "requests\n",
            "requests-2.23.0.dist-info\n",
            "requests_oauthlib\n",
            "requests_oauthlib-1.3.0.dist-info\n",
            "resampy\n",
            "resampy-0.2.2.dist-info\n",
            "retrying-1.3.3.dist-info\n",
            "retrying.py\n",
            "_rinterface_cffi_abi.py\n",
            "_rinterface_cffi_api.abi3.so\n",
            "rpy2\n",
            "rpy2-3.2.7.dist-info\n",
            "rsa\n",
            "rsa-4.6.dist-info\n",
            "samples\n",
            "scikit_image-0.16.2.dist-info\n",
            "scikit_learn-0.22.2.post1.dist-info\n",
            "scipy\n",
            "scipy-1.4.1.dist-info\n",
            "scs\n",
            "scs-2.1.2.dist-info\n",
            "_scs_direct.cpython-36m-x86_64-linux-gnu.so\n",
            "_scs_indirect.cpython-36m-x86_64-linux-gnu.so\n",
            "_scs_python.cpython-36m-x86_64-linux-gnu.so\n",
            "seaborn\n",
            "seaborn-0.11.0.dist-info\n",
            "send2trash\n",
            "Send2Trash-1.5.0.dist-info\n",
            "setuptools\n",
            "setuptools-50.3.2.dist-info\n",
            "setuptools_git\n",
            "setuptools_git-1.2.dist-info\n",
            "shapely\n",
            "Shapely-1.7.1.dist-info\n",
            "simplegeneric-0.8.1.dist-info\n",
            "simplegeneric.py\n",
            "six-1.15.0.dist-info\n",
            "six.py\n",
            "skimage\n",
            "sklearn\n",
            "sklearn-0.0.dist-info\n",
            "sklearn_pandas\n",
            "sklearn_pandas-1.8.0.dist-info\n",
            "slugify\n",
            "slugify-0.0.1.dist-info\n",
            "slugify.py\n",
            "smart_open\n",
            "smart_open-3.0.0.dist-info\n",
            "snowballstemmer\n",
            "snowballstemmer-2.0.0.dist-info\n",
            "sockshandler.py\n",
            "socks.py\n",
            "sortedcontainers\n",
            "sortedcontainers-2.3.0.dist-info\n",
            "spacy\n",
            "spacy-2.2.4.dist-info\n",
            "sphinx\n",
            "Sphinx-1.8.5.dist-info\n",
            "sphinxcontrib\n",
            "sphinxcontrib_serializinghtml-1.1.4.dist-info\n",
            "sphinxcontrib_serializinghtml-1.1.4-py3.8-nspkg.pth\n",
            "sphinxcontrib_websupport-1.2.4.dist-info\n",
            "sphinxcontrib_websupport-1.2.4-py3.8-nspkg.pth\n",
            "sql\n",
            "sqlalchemy\n",
            "SQLAlchemy-1.3.20.dist-info\n",
            "sqlparse\n",
            "sqlparse-0.4.1.dist-info\n",
            "srsly\n",
            "srsly-1.0.4.dist-info\n",
            "statsmodels\n",
            "statsmodels-0.10.2.dist-info\n",
            "sympy\n",
            "sympy-1.1.1.dist-info\n",
            "tables\n",
            "tables-3.4.4.dist-info\n",
            "tabulate-0.8.7.dist-info\n",
            "tabulate.py\n",
            "tblib\n",
            "tblib-1.7.0.dist-info\n",
            "tensorboard\n",
            "tensorboard-2.3.0.dist-info\n",
            "tensorboardcolab\n",
            "tensorboardcolab-0.0.22.dist-info\n",
            "tensorboard_plugin_wit\n",
            "tensorboard_plugin_wit-1.7.0.dist-info\n",
            "tensorflow\n",
            "tensorflow-2.3.0.dist-info\n",
            "tensorflow_addons\n",
            "tensorflow_addons-0.8.3.dist-info\n",
            "tensorflow_datasets\n",
            "tensorflow_datasets-4.0.1.dist-info\n",
            "tensorflow_estimator\n",
            "tensorflow_estimator-2.3.0.dist-info\n",
            "tensorflow_gcs_config\n",
            "tensorflow_gcs_config-2.3.0.dist-info\n",
            "tensorflow_hub\n",
            "tensorflow_hub-0.10.0.dist-info\n",
            "tensorflow_metadata\n",
            "tensorflow_metadata-0.25.0.dist-info\n",
            "tensorflow_privacy\n",
            "tensorflow_privacy-0.2.2.dist-info\n",
            "tensorflow_probability\n",
            "tensorflow_probability-0.11.0.dist-info\n",
            "termcolor-1.1.0.dist-info\n",
            "termcolor.py\n",
            "terminado\n",
            "terminado-0.9.1.dist-info\n",
            "test\n",
            "testpath\n",
            "testpath-0.4.4.dist-info\n",
            "tests\n",
            "textblob\n",
            "textblob-0.15.3.dist-info\n",
            "textgenrnn\n",
            "textgenrnn-1.4.1.dist-info\n",
            "text_unidecode\n",
            "text_unidecode-1.3.dist-info\n",
            "theano\n",
            "Theano-1.0.5.dist-info\n",
            "thinc\n",
            "thinc-7.4.0.dist-info\n",
            "tifffile\n",
            "tifffile-2020.9.3.dist-info\n",
            "tlz\n",
            "toml\n",
            "toml-0.10.2.dist-info\n",
            "toolz\n",
            "toolz-0.11.1.dist-info\n",
            "torch\n",
            "torch-1.7.0+cu101.dist-info\n",
            "torchsummary\n",
            "torchsummary-1.5.1.dist-info\n",
            "torchtext\n",
            "torchtext-0.3.1.dist-info\n",
            "torchvision\n",
            "torchvision-0.8.1+cu101.dist-info\n",
            "torchvision.libs\n",
            "tornado\n",
            "tornado-5.1.1.dist-info\n",
            "tqdm\n",
            "tqdm-4.41.1.dist-info\n",
            "traitlets\n",
            "traitlets-4.3.3.dist-info\n",
            "tree\n",
            "tweepy\n",
            "tweepy-3.6.0.dist-info\n",
            "typeguard\n",
            "typeguard-2.7.1.dist-info\n",
            "typing_extensions-3.7.4.3.dist-info\n",
            "typing_extensions.py\n",
            "tzlocal\n",
            "tzlocal-1.5.1.dist-info\n",
            "umap\n",
            "umap_learn-0.4.6.dist-info\n",
            "uritemplate\n",
            "uritemplate-3.0.1.dist-info\n",
            "urllib3\n",
            "urllib3-1.24.3.dist-info\n",
            "vega_datasets\n",
            "vega_datasets-0.8.0.dist-info\n",
            "vis\n",
            "wasabi\n",
            "wasabi-0.8.0.dist-info\n",
            "wcwidth\n",
            "wcwidth-0.2.5.dist-info\n",
            "webencodings\n",
            "webencodings-0.5.1.dist-info\n",
            "werkzeug\n",
            "Werkzeug-1.0.1.dist-info\n",
            "wheel\n",
            "wheel-0.35.1.dist-info\n",
            "widgetsnbextension\n",
            "widgetsnbextension-3.5.1.dist-info\n",
            "wordcloud\n",
            "wordcloud-1.5.0.dist-info\n",
            "wrapt\n",
            "wrapt-1.12.1.dist-info\n",
            "xarray\n",
            "xarray-0.15.1.dist-info\n",
            "xgboost\n",
            "xgboost-0.90.dist-info\n",
            "xlrd\n",
            "xlrd-1.1.0.dist-info\n",
            "xlwt\n",
            "xlwt-1.3.0.dist-info\n",
            "yaml\n",
            "yellowbrick\n",
            "yellowbrick-0.9.1.dist-info\n",
            "zict\n",
            "zict-2.0.0.dist-info\n",
            "zipp-3.4.0.dist-info\n",
            "zipp.py\n",
            "zmq\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "uNt7sxpAbmAn"
      },
      "source": [
        "Atualizar o local para instalar os pacotes do conda"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "whDhkl2-blWp"
      },
      "source": [
        "import sys\n",
        "\n",
        "_ = (sys.path\n",
        "        .append(\"/usr/local/lib/python3.6/site-packages\"))"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "LWjg34tTby6e"
      },
      "source": [
        "Instalado o plink usando o conda."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gPzITOsNb22O",
        "outputId": "11700817-6b07-4c91-fa5f-59d50bc933dd"
      },
      "source": [
        "!conda install -c bioconda plink"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - plink\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    plink-1.90b6.18            |       h516909a_0         6.8 MB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         6.8 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  plink              bioconda/linux-64::plink-1.90b6.18-h516909a_0\n",
            "\n",
            "\n",
            "Proceed ([y]/n)? y\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "plink-1.90b6.18      | 6.8 MB    | : 100% 1.0/1 [00:01<00:00,  1.86s/it]\n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\bdone\n",
            "Executing transaction: \\ \b\bdone\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "QCMNAufGcVZz"
      },
      "source": [
        "Testando o plink."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "iLqMNtVWcYN7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "8c6bd7eb-c6f6-4250-8d7c-30dbdb7c2e64"
      },
      "source": [
        "!plink --help\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "PLINK v1.90b6.18 64-bit (16 Jun 2020)          www.cog-genomics.org/plink/1.9/\n",
            "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
            "\n",
            "In the command line flag definitions that follow,\n",
            "  * <angle brackets> denote a required parameter, where the text between the\n",
            "    angle brackets describes its nature.\n",
            "  * ['square brackets + single-quotes'] denotes an optional modifier.  Use the\n",
            "    EXACT text in the quotes.\n",
            "  * [{bar|separated|braced|bracketed|values}] denotes a collection of mutually\n",
            "    exclusive optional modifiers (again, the exact text must be used).  When\n",
            "    there are no outer square brackets, one of the choices must be selected.\n",
            "  * ['quoted_text='<description of value>] denotes an optional modifier that\n",
            "    must begin with the quoted text, and be followed by a value with no\n",
            "    whitespace in between.  '|' may also be used here to indicate mutually\n",
            "    exclusive options.\n",
            "  * [square brackets without quotes or braces] denote an optional parameter,\n",
            "    where the text between the brackets describes its nature.\n",
            "  * An ellipsis (...) indicates that you may enter multiple parameters of the\n",
            "    specified type.\n",
            "\n",
            "  plink <input flag(s)...> [command flag(s)...] [other flag(s)...]\n",
            "  plink --help [flag name(s)...]\n",
            "\n",
            "Most PLINK runs require exactly one main input fileset.  The following flags\n",
            "are available for defining its form and location:\n",
            "\n",
            "  --bfile [prefix] : Specify .bed + .bim + .fam prefix (default 'plink').\n",
            "  --bed <filename> : Specify full name of .bed file.\n",
            "  --bim <filename> : Specify full name of .bim file.\n",
            "  --fam <filename> : Specify full name of .fam file.\n",
            "\n",
            "  --keep-autoconv  : With --file/--tfile/--lfile/--vcf/--bcf/--data/--23file,\n",
            "                     don't delete autogenerated binary fileset at end of run.\n",
            "\n",
            "  --file [prefix]  : Specify .ped + .map filename prefix (default 'plink').\n",
            "  --ped <filename> : Specify full name of .ped file.\n",
            "  --map <filename> : Specify full name of .map file.\n",
            "\n",
            "  --no-fid         : .fam/.ped file does not contain column 1 (family ID).\n",
            "  --no-parents     : .fam/.ped file does not contain columns 3-4 (parents).\n",
            "  --no-sex         : .fam/.ped file does not contain column 5 (sex).\n",
            "  --no-pheno       : .fam/.ped file does not contain column 6 (phenotype).\n",
            "\n",
            "  --tfile [prefix] : Specify .tped + .tfam filename prefix (default 'plink').\n",
            "  --tped <fname>   : Specify full name of .tped file.\n",
            "  --tfam <fname>   : Specify full name of .tfam file.\n",
            "\n",
            "  --lfile [prefix] : Specify .lgen + .map + .fam (long-format fileset) prefix.\n",
            "  --lgen <fname>   : Specify full name of .lgen file.\n",
            "  --reference <fn> : Specify default allele file accompanying .lgen input.\n",
            "  --allele-count   : When used with --lfile/--lgen + --reference, specifies\n",
            "                     that the .lgen file contains reference allele counts.\n",
            "\n",
            "  --vcf <filename> : Specify full name of .vcf or .vcf.gz file.\n",
            "  --bcf <filename> : Specify full name of BCF2 file.\n",
            "\n",
            "  --data [prefix]  : Specify Oxford .gen + .sample prefix (default 'plink').\n",
            "  --gen <filename> : Specify full name of .gen or .gen.gz file.\n",
            "  --bgen <f> ['snpid-chr'] : Specify full name of .bgen file.\n",
            "  --sample <fname> : Specify full name of .sample file.\n",
            "\n",
            "  --23file <fname> [FID] [IID] [sex] [pheno] [pat. ID] [mat. ID] :\n",
            "    Specify 23andMe input file.\n",
            "\n",
            "  --grm-gz [prfx]  : Specify .grm.gz + .grm.id (GCTA rel. matrix) prefix.\n",
            "  --grm-bin [prfx] : Specify .grm.bin + .grm.N.bin + .grm.id (GCTA triangular\n",
            "                     binary relationship matrix) filename prefix.\n",
            "\n",
            "  --dummy <sample ct> <SNP ct> [missing geno freq] [missing pheno freq]\n",
            "          [{acgt | 1234 | 12}] ['scalar-pheno']\n",
            "    This generates a fake input dataset with the specified number of samples\n",
            "    and SNPs.  By default, the missing genotype and phenotype frequencies are\n",
            "    zero, and genotypes are As and Bs (change the latter with\n",
            "    'acgt'/'1234'/'12').  The 'scalar-pheno' modifier causes a normally\n",
            "    distributed scalar phenotype to be generated instead of a binary one.\n",
            "\n",
            "  --simulate <simulation parameter file> [{tags | haps}] [{acgt | 1234 | 12}]\n",
            "  --simulate-qt <sim. parameter file> [{tags | haps}] [{acgt | 1234 | 12}]\n",
            "    --simulate generates a fake input dataset with disease-associated SNPs,\n",
            "    while --simulate-qt generates a dataset with quantitative trait loci.\n",
            "\n",
            "Output files have names of the form 'plink.<extension>' by default.  You can\n",
            "change the 'plink' prefix with\n",
            "\n",
            "  --out <prefix>   : Specify prefix for output files.\n",
            "\n",
            "Most runs also require at least one of the following commands:\n",
            "\n",
            "  --make-bed\n",
            "    Create a new binary fileset.  Unlike the automatic text-to-binary\n",
            "    converters (which only heed chromosome filters), this supports all of\n",
            "    PLINK's filtering flags.\n",
            "  --make-just-bim\n",
            "  --make-just-fam\n",
            "    Variants of --make-bed which only write a new .bim or .fam file.  Can be\n",
            "    used with only .bim/.fam input.\n",
            "    USE THESE CAUTIOUSLY.  It is very easy to desynchronize your binary\n",
            "    genotype data and your .bim/.fam indexes if you use these commands\n",
            "    improperly.  If you have any doubt, stick with --make-bed.\n",
            "\n",
            "  --recode <output format> [{01 | 12}] [{tab | tabx | spacex | bgz | gen-gz}]\n",
            "           ['include-alt'] ['omit-nonmale-y']\n",
            "    Create a new text fileset with all filters applied.  The following output\n",
            "    formats are supported:\n",
            "    * '23': 23andMe 4-column format.  This can only be used on a single\n",
            "      sample's data (--keep may be handy), and does not support multicharacter\n",
            "      allele codes.\n",
            "    * 'A': Sample-major additive (0/1/2) coding, suitable for loading from R.\n",
            "      If you need uncounted alleles to be named in the header line, add the\n",
            "      'include-alt' modifier.\n",
            "    * 'AD': Sample-major additive (0/1/2) + dominant (het=1/hom=0) coding.\n",
            "      Also supports 'include-alt'.\n",
            "    * 'A-transpose': Variant-major 0/1/2.\n",
            "    * 'beagle': Unphased per-autosome .dat and .map files, readable by early\n",
            "      BEAGLE versions.\n",
            "    * 'beagle-nomap': Single .beagle.dat file.\n",
            "    * 'bimbam': Regular BIMBAM format.\n",
            "    * 'bimbam-1chr': BIMBAM format, with a two-column .pos.txt file.  Does not\n",
            "      support multiple chromosomes.\n",
            "    * 'fastphase': Per-chromosome fastPHASE files, with\n",
            "      .chr-<chr #>.recode.phase.inp filename extensions.\n",
            "    * 'fastphase-1chr': Single .recode.phase.inp file.  Does not support\n",
            "      multiple chromosomes.\n",
            "    * 'HV': Per-chromosome Haploview files, with .chr-<chr #>{.ped,.info}\n",
            "      filename extensions.\n",
            "    * 'HV-1chr': Single Haploview .ped + .info file pair.  Does not support\n",
            "      multiple chromosomes.\n",
            "    * 'lgen': PLINK 1 long-format (.lgen + .fam + .map), loadable with --lfile.\n",
            "    * 'lgen-ref': .lgen + .fam + .map + .ref, loadable with --lfile +\n",
            "       --reference.\n",
            "    * 'list': Single genotype-based list, up to 4 lines per variant.  To omit\n",
            "      nonmale genotypes on the Y chromosome, add the 'omit-nonmale-y' modifier.\n",
            "    * 'rlist': .rlist + .fam + .map fileset, where the .rlist file is a\n",
            "      genotype-based list which omits the most common genotype for each\n",
            "      variant.  Also supports 'omit-nonmale-y'.\n",
            "    * 'oxford': Oxford-format .gen + .sample.  With the 'gen-gz' modifier, the\n",
            "      .gen file is gzipped.\n",
            "    * 'ped': PLINK 1 sample-major (.ped + .map), loadable with --file.\n",
            "    * 'compound-genotypes': Same as 'ped', except that the space between each\n",
            "      pair of same-variant allele codes is removed.\n",
            "    * 'structure': Structure-format.\n",
            "    * 'transpose': PLINK 1 variant-major (.tped + .tfam), loadable with\n",
            "      --tfile.\n",
            "    * 'vcf', 'vcf-fid', 'vcf-iid': VCFv4.2.  'vcf-fid' and 'vcf-iid' cause\n",
            "      family IDs or within-family IDs respectively to be used for the sample\n",
            "      IDs in the last header row, while 'vcf' merges both IDs and puts an\n",
            "      underscore between them.  If the 'bgz' modifier is added, the VCF file is\n",
            "      block-gzipped.\n",
            "      The A2 allele is saved as the reference and normally flagged as not based\n",
            "      on a real reference genome (INFO:PR).  When it is important for reference\n",
            "      alleles to be correct, you'll also want to include --a2-allele and\n",
            "      --real-ref-alleles in your command.\n",
            "    In addition,\n",
            "    * The '12' modifier causes A1 (usually minor) alleles to be coded as '1'\n",
            "      and A2 alleles to be coded as '2', while '01' maps A1 -> 0 and A2 -> 1.\n",
            "    * The 'tab' modifier makes the output mostly tab-delimited instead of\n",
            "      mostly space-delimited.  'tabx' and 'spacex' force all tabs and all\n",
            "      spaces, respectively.\n",
            "\n",
            "  --flip-scan ['verbose']\n",
            "    (alias: --flipscan)\n",
            "    LD-based scan for case/control strand inconsistency.\n",
            "\n",
            "  --write-covar\n",
            "    If a --covar file is loaded, --make-bed/--make-just-fam and --recode\n",
            "    automatically generate an updated version (with all filters applied).\n",
            "    However, if you do not wish to simultaneously generate a new genotype file,\n",
            "    you can use --write-covar to just produce a pruned covariate file.\n",
            "\n",
            "  --write-cluster ['omit-unassigned']\n",
            "    If clusters are specified with --within/--family, this generates a new\n",
            "    cluster file (with all filters applied).  The 'omit-unassigned' modifier\n",
            "    causes unclustered samples to be omitted from the file; otherwise their\n",
            "    cluster is 'NA'.\n",
            "\n",
            "  --write-set\n",
            "  --set-table\n",
            "    If sets have been defined, --write-set dumps 'END'-terminated set\n",
            "    membership lists to <output prefix>.set, while --set-table writes a\n",
            "    variant-by-set membership table to <output prefix>.set.table.\n",
            "\n",
            "  --merge <.ped filename> <.map filename>\n",
            "  --merge <text fileset prefix>\n",
            "  --bmerge <.bed filename> <.bim filename> <.fam filename>\n",
            "  --bmerge <binary fileset prefix>\n",
            "    Merge the given fileset with the initially loaded fileset, writing the\n",
            "    result to <output prefix>.bed + .bim + .fam.  (It is no longer necessary to\n",
            "    simultaneously specify --make-bed.)\n",
            "  --merge-list <filename>\n",
            "    Merge all filesets named in the text file with the reference fileset, if\n",
            "    one was specified.  (However, this can also be used *without* a reference;\n",
            "    in that case, the newly created fileset is then treated as the reference by\n",
            "    most other PLINK operations.)  The text file is interpreted as follows:\n",
            "    * If a line contains only one name, it is assumed to be the prefix for a\n",
            "      binary fileset.\n",
            "    * If a line contains exactly two names, they are assumed to be the full\n",
            "      filenames for a text fileset (.ped first, then .map).\n",
            "    * If a line contains exactly three names, they are assumed to be the full\n",
            "      filenames for a binary fileset (.bed, then .bim, then .fam).\n",
            "\n",
            "  --write-snplist\n",
            "  --list-23-indels\n",
            "    --write-snplist writes a .snplist file listing the names of all variants\n",
            "    which pass the filters and inclusion thresholds you've specified, while\n",
            "    --list-23-indels writes the subset with 23andMe-style indel calls (D/I\n",
            "    allele codes).\n",
            "\n",
            "  --list-duplicate-vars ['require-same-ref'] ['ids-only'] ['suppress-first']\n",
            "    --list-duplicate-vars writes a .dupvar file describing all groups of\n",
            "    variants with matching positions and allele codes.\n",
            "    * By default, A1/A2 allele assignments are ignored; use 'require-same-ref'\n",
            "      to override this.\n",
            "    * Normally, the report contains position and allele codes.  To remove them\n",
            "      (and produce a file directly usable with e.g. --extract/--exclude), use\n",
            "      'ids-only'.  Note that this command will fail in 'ids-only' mode if any\n",
            "      of the reported IDs are not unique.\n",
            "    * 'suppress-first' causes the first variant ID in each group to be omitted\n",
            "      from the report.\n",
            "\n",
            "  --freq [{counts | case-control}] ['gz']\n",
            "  --freqx ['gz']\n",
            "    --freq generates a basic allele frequency (or count, if the 'counts'\n",
            "    modifier is present) report.  This can be combined with --within/--family\n",
            "    to produce a cluster-stratified allele frequency/count report instead, or\n",
            "    the 'case-control' modifier to report case and control allele frequencies\n",
            "    separately.\n",
            "    --freqx generates a more detailed genotype count report, designed for use\n",
            "    with --read-freq.\n",
            "\n",
            "  --missing ['gz']\n",
            "    Generate sample- and variant-based missing data reports.  If clusters are\n",
            "    defined, the variant-based report is cluster-stratified.  'gz' causes the\n",
            "    output files to be gzipped.\n",
            "    Unlike most other commands, this doesn't treat het. haploids as missing.\n",
            "\n",
            "  --test-mishap\n",
            "    Check for association between missing calls and flanking haplotypes.\n",
            "\n",
            "  --hardy ['midp'] ['gz']\n",
            "    Generate a Hardy-Weinberg exact test p-value report.  (This does NOT\n",
            "    simultaneously filter on the p-value any more; use --hwe for that.)  With\n",
            "    the 'midp' modifier, the test applies the mid-p adjustment described in\n",
            "    Graffelman J, Moreno V (2013) The mid p-value in exact tests for\n",
            "    Hardy-Weinberg Equilibrium.\n",
            "\n",
            "  --mendel ['summaries-only']\n",
            "    Generate a Mendel error report.  The 'summaries-only' modifier causes the\n",
            "    .mendel file (listing every single error) to be skipped.\n",
            "\n",
            "  --het ['small-sample'] ['gz']\n",
            "  --ibc\n",
            "    Estimate inbreeding coefficients.  --het reports method-of-moments\n",
            "    estimates, while --ibc calculates all three values described in Yang J, Lee\n",
            "    SH, Goddard ME and Visscher PM (2011) GCTA: A Tool for Genome-wide Complex\n",
            "    Trait Analysis.  (That paper also describes the relationship matrix\n",
            "    computation we reimplement.)\n",
            "    * These functions require decent MAF estimates.  If there are very few\n",
            "      samples in your immediate fileset, --read-freq is practically mandatory\n",
            "      since imputed MAFs are wildly inaccurate in that case.\n",
            "    * They also assume the marker set is in approximate linkage equilibrium.\n",
            "    * By default, --het omits the n/(n-1) multiplier in Nei's expected\n",
            "      homozygosity formula.  The 'small-sample' modifier causes it to be\n",
            "      included, while forcing --het to use MAFs imputed from founders in the\n",
            "      immediate dataset.\n",
            "\n",
            "  --check-sex [female max F] [male min F]\n",
            "  --check-sex ycount [female max F] [male min F] [female max Y obs]\n",
            "                     [male min Y obs]\n",
            "  --check-sex y-only [female max Y obs] [male min Y obs]\n",
            "  --impute-sex [female max F] [male min F]\n",
            "  --impute-sex ycount [female max F] [male min F] [female max Y obs]\n",
            "                      [male min Y obs]\n",
            "  --impute-sex y-only [female max Y obs] [male min Y obs]\n",
            "    --check-sex normally compares sex assignments in the input dataset with\n",
            "    those imputed from X chromosome inbreeding coefficients.\n",
            "    * Make sure that the X chromosome pseudo-autosomal region has been split\n",
            "      off (with e.g. --split-x) before using this.\n",
            "    * You also need decent MAF estimates (so, with very few samples in your\n",
            "      immediate fileset, use --read-freq), and your marker set should be in\n",
            "      approximate linkage equilibrium.\n",
            "    * By default, F estimates smaller than 0.2 yield female calls, and values\n",
            "      larger than 0.8 yield male calls.  If you pass numeric parameter(s) to\n",
            "      --check-sex, the first two control these thresholds.\n",
            "    There are now two modes which consider Y chromosome data.\n",
            "    * In 'ycount' mode, gender is still imputed from the X chromosome, but\n",
            "      female calls are downgraded to ambiguous whenever more than 0 nonmissing\n",
            "      Y genotypes are present, and male calls are downgraded when fewer than 0\n",
            "      are present.  (Note that these are counts, not rates.)  These thresholds\n",
            "      are controllable with --check-sex ycount's optional 3rd and 4th numeric\n",
            "      parameters.\n",
            "    * In 'y-only' mode, gender is imputed from nonmissing Y genotype counts.\n",
            "      The male minimum threshold defaults to 1 instead of zero in this case.\n",
            "    --impute-sex changes sex assignments to the imputed values, and is\n",
            "    otherwise identical to --check-sex.  It must be used with\n",
            "    --make-bed/--recode/--write-covar.\n",
            "\n",
            "  --fst ['case-control']\n",
            "    (alias: --Fst)\n",
            "    Estimate Wright's Fst for each autosomal diploid variant using the method\n",
            "    introduced in Weir BS, Cockerham CC (1984) Estimating F-statistics for the\n",
            "    analysis of population structure, given a set of subpopulations defined via\n",
            "    --within.  Raw and weighted global means are also reported.\n",
            "    * If you're interested in the global means, it is usually best to perform\n",
            "      this calculation on a marker set in approximate linkage equilibrium.\n",
            "    * If you have only two subpopulations, you can represent them with\n",
            "      case/control status and use the 'case-control' modifier.\n",
            "\n",
            "  --indep <window size>['kb'] <step size (variant ct)> <VIF threshold>\n",
            "  --indep-pairwise <window size>['kb'] <step size (variant ct)> <r^2 threshold>\n",
            "  --indep-pairphase <window size>['kb'] <step size (variant ct)> <r^2 thresh>\n",
            "    Generate a list of markers in approximate linkage equilibrium.  With the\n",
            "    'kb' modifier, the window size is in kilobase instead of variant count\n",
            "    units.  (Pre-'kb' space is optional, i.e. \"--indep-pairwise 500 kb 5 0.5\"\n",
            "    and \"--indep-pairwise 500kb 5 0.5\" have the same effect.)\n",
            "    Note that you need to rerun PLINK using --extract or --exclude on the\n",
            "    .prune.in/.prune.out file to apply the list to another computation.\n",
            "\n",
            "  --r [{square | square0 | triangle | inter-chr}] [{gz | bin | bin4}]\n",
            "      ['spaces'] ['in-phase'] [{d | dprime | dprime-signed}] ['with-freqs']\n",
            "      ['yes-really']\n",
            "  --r2 [{square | square0 | triangle | inter-chr}] [{gz | bin | bin4}]\n",
            "       ['spaces'] ['in-phase'] [{d | dprime | dprime-signed}] ['with-freqs']\n",
            "       ['yes-really']\n",
            "    LD statistic reports.  --r yields raw inter-variant correlations, while\n",
            "    --r2 reports their squares.  You can request results for all pairs in\n",
            "    matrix format (if you specify 'bin' or one of the shape modifiers), all\n",
            "    pairs in table format ('inter-chr'), or a limited window in table format\n",
            "    (default).\n",
            "    * The 'gz' modifier causes the output text file to be gzipped.\n",
            "    * 'bin' causes the output matrix to be written in double-precision binary\n",
            "      format, while 'bin4' specifics single-precision binary.  The matrix is\n",
            "      square if no shape is explicitly specified.\n",
            "    * By default, text matrices are tab-delimited; 'spaces' switches this.\n",
            "    * 'in-phase' adds a column with in-phase allele pairs to table-formatted\n",
            "      reports.  (This cannot be used with very long allele codes.)\n",
            "    * 'dprime' adds the absolute value of Lewontin's D-prime statistic to\n",
            "      table-formatted reports, and forces both r/r^2 and D-prime to be based on\n",
            "      the maximum likelihood solution to the cubic equation discussed in Gaunt\n",
            "      T, Rodriguez S, Day I (2007) Cubic exact solutions for the estimation of\n",
            "      pairwise haplotype frequencies.\n",
            "      'dprime-signed' keeps the sign, while 'd' skips division by D_{max}.\n",
            "    * 'with-freqs' adds MAF columns to table-formatted reports.\n",
            "    * Since the resulting file can easily be huge, you're required to add the\n",
            "      'yes-really' modifier when requesting an unfiltered, non-distributed all\n",
            "      pairs computation on more than 400k variants.\n",
            "    * These computations can be subdivided with --parallel (even when the\n",
            "      'square' modifier is active).\n",
            "  --ld <variant ID> <variant ID> ['hwe-midp']\n",
            "    This displays haplotype frequencies, r^2, and D' for a single pair of\n",
            "    variants.  When there are multiple biologically possible solutions to the\n",
            "    haplotype frequency cubic equation, all are displayed (instead of just the\n",
            "    maximum likelihood solution identified by --r/--r2), along with HWE exact\n",
            "    test statistics.\n",
            "\n",
            "  --show-tags <filename>\n",
            "  --show-tags all\n",
            "    * If a file is specified, list all variants which tag at least one variant\n",
            "      named in the file.  (This will normally be a superset of the original\n",
            "      list, since a variant is considered to tag itself here.)\n",
            "    * If 'all' mode is specified, for each variant, each *other* variant which\n",
            "      tags it is reported.\n",
            "\n",
            "  --blocks ['no-pheno-req'] ['no-small-max-span']\n",
            "    Estimate haplotype blocks, via Haploview's interpretation of the block\n",
            "    definition suggested by Gabriel S et al. (2002) The Structure of Haplotype\n",
            "    Blocks in the Human Genome.\n",
            "    * Normally, samples with missing phenotypes are not considered by this\n",
            "      computation; the 'no-pheno-req' modifier lifts this restriction.\n",
            "    * Normally, size-2 blocks may not span more than 20kb, and size-3 blocks\n",
            "      are limited to 30kb.  The 'no-small-max-span' modifier removes these\n",
            "      limits.\n",
            "    The .blocks file is valid input for PLINK 1.07's --hap command.  However,\n",
            "    the --hap... family of flags has not been reimplemented in PLINK 1.9 due to\n",
            "    poor phasing accuracy relative to other software; for now, we recommend\n",
            "    using BEAGLE instead of PLINK for case/control haplotype association\n",
            "    analysis.  (You can use \"--recode beagle\" to export data to BEAGLE 3.3.)\n",
            "    We apologize for the inconvenience, and plan to develop variants of the\n",
            "    --hap... flags which handle pre-phased data effectively.\n",
            "\n",
            "  --distance [{square | square0 | triangle}] [{gz | bin | bin4}] ['ibs']\n",
            "             ['1-ibs'] ['allele-ct'] ['flat-missing']\n",
            "    Write a lower-triangular tab-delimited table of (weighted) genomic\n",
            "    distances in allele count units to <output prefix>.dist, and a list of the\n",
            "    corresponding sample IDs to <output prefix>.dist.id.  The first row of the\n",
            "    .dist file contains a single <genome 1-genome 2> distance, the second row\n",
            "    has the <genome 1-genome 3> and <genome 2-genome 3> distances in that\n",
            "    order, etc.\n",
            "    * It is usually best to perform this calculation on a marker set in\n",
            "      approximate linkage equilibrium.\n",
            "    * If the 'square' or 'square0' modifier is present, a square matrix is\n",
            "      written instead; 'square0' fills the upper right triangle with zeroes.\n",
            "    * If the 'gz' modifier is present, a compressed .dist.gz file is written\n",
            "      instead of a plain text file.\n",
            "    * If the 'bin' modifier is present, a binary (square) matrix of\n",
            "      double-precision floating point values, suitable for loading from R, is\n",
            "      instead written to <output prefix>.dist.bin.  ('bin4' specifies\n",
            "      single-precision numbers instead.)  This can be combined with 'square0'\n",
            "      if you still want the upper right zeroed out, or 'triangle' if you don't\n",
            "      want to pad the upper right at all.\n",
            "    * If the 'ibs' modifier is present, an identity-by-state matrix is written\n",
            "      to <output prefix>.mibs.  '1-ibs' causes distances expressed as genomic\n",
            "      proportions (i.e. 1 - IBS) to be written to <output prefix>.mdist.\n",
            "      Combine with 'allele-ct' if you want to generate the usual .dist file as\n",
            "      well.\n",
            "    * By default, distance rescaling in the presence of missing genotype calls\n",
            "      is sensitive to allele count distributions: if variant A contributes, on\n",
            "      average, twice as much to other pairwise distances as variant B, a\n",
            "      missing call at variant A will result in twice as large of a missingness\n",
            "      correction.  To turn this off (because e.g. your missing calls are highly\n",
            "      nonrandom), use the 'flat-missing' modifier.\n",
            "    * The computation can be subdivided with --parallel.\n",
            "  --distance-matrix\n",
            "  --ibs-matrix\n",
            "    These deprecated commands are equivalent to \"--distance 1-ibs flat-missing\n",
            "    square\" and \"--distance ibs flat-missing square\", respectively, except that\n",
            "    they generate space- instead of tab-delimited text matrices.\n",
            "\n",
            "  --make-rel [{square | square0 | triangle}] [{gz | bin | bin4}]\n",
            "             [{cov | ibc2 | ibc3}]\n",
            "    Write a lower-triangular variance-standardized realized relationship matrix\n",
            "    to <output prefix>.rel, and corresponding IDs to <output prefix>.rel.id.\n",
            "    * It is usually best to perform this calculation on a marker set in\n",
            "      approximate linkage equilibrium.\n",
            "    * 'square', 'square0', 'triangle', 'gz', 'bin', and 'bin4' act as they do\n",
            "      on --distance.\n",
            "    * The 'cov' modifier removes the variance standardization step, causing a\n",
            "      covariance matrix to be calculated instead.\n",
            "    * By default, the diagonal elements in the relationship matrix are based on\n",
            "      --ibc's Fhat1; use the 'ibc2' or 'ibc3' modifiers to base them on Fhat2\n",
            "      or Fhat3 instead.\n",
            "    * The computation can be subdivided with --parallel.\n",
            "  --make-grm-gz ['no-gz'] [{cov | ibc2 | ibc3}]\n",
            "  --make-grm-bin [{cov | ibc2 | ibc3}]\n",
            "    --make-grm-gz writes the relationships in GCTA's original gzipped list\n",
            "    format, which describes one pair per line, while --make-grm-bin writes them\n",
            "    in GCTA 1.1+'s single-precision triangular binary format.  Note that these\n",
            "    formats explicitly report the number of valid observations (where neither\n",
            "    sample has a missing call) for each pair, which is useful input for some\n",
            "    scripts.\n",
            "    These computations can be subdivided with --parallel.\n",
            "\n",
            "  --rel-cutoff [val]\n",
            "    (alias: --grm-cutoff)\n",
            "    Exclude one member of each pair of samples with relatedness greater than\n",
            "    the given cutoff value (default 0.025).  If no later operation will cause\n",
            "    the list of remaining samples to be written to disk, this will save it to\n",
            "    <output prefix>.rel.id.\n",
            "    Note that maximizing the remaining sample size is equivalent to the NP-hard\n",
            "    maximum independent set problem, so we use a greedy algorithm instead of\n",
            "    guaranteeing optimality.  (Use the --make-rel and --keep/--remove flags if\n",
            "    you want to try to do better.)\n",
            "\n",
            "  --ibs-test [permutation count]\n",
            "  --groupdist [iters] [d]\n",
            "    Given case/control phenotype data, these commands consider three subsets of\n",
            "    the distance matrix: pairs of affected samples, affected-unaffected pairs,\n",
            "    and pairs of unaffected samples.  Each of these subsets has a distribution\n",
            "    of pairwise genomic distances; --ibs-test uses permutation to estimate\n",
            "    p-values re: which types of pairs are most similar, while --groupdist\n",
            "    focuses on the differences between the centers of these distributions and\n",
            "    estimates standard errors via delete-d jackknife.\n",
            "\n",
            "  --regress-distance [iters] [d]\n",
            "    Linear regression of pairwise genomic distances on pairwise average\n",
            "    phenotypes and vice versa, using delete-d jackknife for standard errors.  A\n",
            "    scalar phenotype is required.\n",
            "    * With less than two parameters, d is set to <number of people>^0.6 rounded\n",
            "      down.  With no parameters, 100k iterations are run.\n",
            "  --regress-rel [iters] [d]\n",
            "    Linear regression of pairwise genomic relationships on pairwise average\n",
            "    phenotypes, and vice versa.  Defaults for iters and d are the same as for\n",
            "    --regress-distance.\n",
            "\n",
            "  --genome ['gz'] ['rel-check'] ['full'] ['unbounded'] ['nudge']\n",
            "    Generate an identity-by-descent report.\n",
            "    * It is usually best to perform this calculation on a marker set in\n",
            "      approximate linkage equilibrium.\n",
            "    * The 'rel-check' modifier excludes pairs of samples with different FIDs\n",
            "      from the final report.\n",
            "    * 'full' adds raw pairwise comparison data to the report.\n",
            "    * The P(IBD=0/1/2) estimator employed by this command sometimes yields\n",
            "      numbers outside the range [0,1]; by default, these are clipped.  The\n",
            "      'unbounded' modifier turns off this clipping.\n",
            "    * Then, when PI_HAT^2 < P(IBD=2), 'nudge' adjusts the final P(IBD=0/1/2)\n",
            "      estimates to a theoretically possible configuration.\n",
            "    * The computation can be subdivided with --parallel.\n",
            "\n",
            "  --homozyg [{group | group-verbose}] ['consensus-match'] ['extend']\n",
            "            ['subtract-1-from-lengths']\n",
            "  --homozyg-snp <min var count>\n",
            "  --homozyg-kb <min length>\n",
            "  --homozyg-density <max inverse density (kb/var)>\n",
            "  --homozyg-gap <max internal gap kb length>\n",
            "  --homozyg-het <max hets>\n",
            "  --homozyg-window-snp <scanning window size>\n",
            "  --homozyg-window-het <max hets in scanning window hit>\n",
            "  --homozyg-window-missing <max missing calls in scanning window hit>\n",
            "  --homozyg-window-threshold <min scanning window hit rate>\n",
            "    These commands request a set of run-of-homozygosity reports, and allow you\n",
            "    to customize how they are generated.\n",
            "    * If you're satisfied with all the default settings described below, just\n",
            "      use --homozyg with no modifiers.  Otherwise, --homozyg lets you change a\n",
            "      few binary settings:\n",
            "      * 'group[-verbose]' adds a report on pools of overlapping runs of\n",
            "        homozygosity.  (Automatically set when --homozyg-match is present.)\n",
            "      * With 'group[-verbose]', 'consensus-match' causes pairwise segmental\n",
            "        matches to be called based on the variants in the pool's consensus\n",
            "        segment, rather than the variants in the pairwise intersection.\n",
            "      * Due to how the scanning window algorithm works, it is possible for a\n",
            "        reported ROH to be adjacent to a few homozygous variants.  The 'extend'\n",
            "        modifier causes them to be included in the reported ROH if that\n",
            "        wouldn't cause a violation of the --homozyg-density bound.\n",
            "      * By default, segment bp lengths are calculated as <end bp position> -\n",
            "        <start bp position> + 1.  Therefore, reports normally differ slightly\n",
            "        from PLINK 1.07, which does not add 1 at the end.  For testing\n",
            "        purposes, you can use the 'subtract-1-from-lengths' modifier to apply\n",
            "        the old formula.\n",
            "    * By default, only runs of homozygosity containing at least 100 variants,\n",
            "      and of total length >= 1000 kilobases, are noted.  You can change these\n",
            "      minimums with --homozyg-snp and --homozyg-kb, respectively.\n",
            "    * By default, a ROH must have at least one variant per 50 kb on average;\n",
            "      change this bound with --homozyg-density.\n",
            "    * By default, if two consecutive variants are more than 1000 kb apart, they\n",
            "      cannot be in the same ROH; change this bound with --homozyg-gap.\n",
            "    * By default, a ROH can contain an unlimited number of heterozygous calls;\n",
            "      you can impose a limit with --homozyg-het.\n",
            "    * By default, the scanning window contains 50 variants; change this with\n",
            "      --homozyg-window-snp.\n",
            "    * By default, a scanning window hit can contain at most 1 heterozygous\n",
            "      call and 5 missing calls; change these limits with --homozyg-window-het\n",
            "      and --homozyg-window-missing, respectively.\n",
            "    * By default, for a variant to be eligible for inclusion in a ROH, the hit\n",
            "      rate of all scanning windows containing the variant must be at least\n",
            "      0.05; change this threshold with --homozyg-window-threshold.\n",
            "\n",
            "  --cluster ['cc'] [{group-avg | old-tiebreaks}] ['missing'] ['only2']\n",
            "    Cluster samples using a pairwise similarity statistic (normally IBS).\n",
            "    * The 'cc' modifier forces every cluster to have at least one case and one\n",
            "      control.\n",
            "    * The 'group-avg' modifier causes clusters to be joined based on average\n",
            "      instead of minimum pairwise similarity.\n",
            "    * The 'missing' modifier causes clustering to be based on\n",
            "      identity-by-missingness instead of identity-by-state, and writes a\n",
            "      space-delimited identity-by-missingness matrix to disk.\n",
            "    * The 'only2' modifier causes only a .cluster2 file (which is valid input\n",
            "      for --within) to be written; otherwise 2 other files will be produced.\n",
            "    * By default, IBS ties are not broken in the same manner as PLINK 1.07, so\n",
            "      final cluster solutions tend to differ.  This is generally harmless.\n",
            "      However, to simplify testing, you can use the 'old-tiebreaks' modifier to\n",
            "      force emulation of the old algorithm.\n",
            "\n",
            "  --pca [count] ['header'] ['tabs'] ['var-wts']\n",
            "    Calculates a variance-standardized relationship matrix (use\n",
            "    --make-rel/--make-grm-gz/--make-grm-bin to dump it), and extracts the top\n",
            "    20 principal components.\n",
            "    * It is usually best to perform this calculation on a marker set in\n",
            "      approximate linkage equilibrium.\n",
            "    * You can change the number of PCs by passing a numeric parameter.\n",
            "    * The 'header' modifier adds a header line to the .eigenvec output file.\n",
            "      (For compatibility with the GCTA flag of the same name, the default is no\n",
            "      header line.)\n",
            "    * The 'tabs' modifier causes the .eigenvec file(s) to be tab-delimited.\n",
            "    * The 'var-wts' modifier requests an additional .eigenvec.var file with PCs\n",
            "      expressed as variant weights instead of sample weights.\n",
            "\n",
            "  --neighbour <n1> <n2>\n",
            "    (alias: --neighbor)\n",
            "    Report IBS distances from each sample to their n1th- to n2th-nearest\n",
            "    neighbors, associated Z-scores, and the identities of those neighbors.\n",
            "    Useful for outlier detection.\n",
            "\n",
            "  --assoc ['perm' | 'mperm='<value>] ['perm-count'] [{fisher | fisher-midp}]\n",
            "          ['counts'] ['set-test']\n",
            "  --assoc ['perm' | 'mperm='<value>] ['perm-count'] ['qt-means'] ['lin']\n",
            "          ['set-test']\n",
            "  --model ['perm' | 'mperm='<value>] ['perm-count']\n",
            "          [{fisher | fisher-midp | trend-only}] ['set-test']\n",
            "          [{dom | rec | gen | trend}]\n",
            "    Basic association analysis report.\n",
            "    Given a case/control phenotype, --assoc performs a 1df chi-square allelic\n",
            "    test, while --model performs 4 other tests as well (1df dominant gene\n",
            "    action, 1df recessive gene action, 2df genotypic, Cochran-Armitage trend).\n",
            "    * With 'fisher'/'fisher-midp', Fisher's exact test is used to generate\n",
            "      p-values.  'fisher-midp' also applies Lancaster's mid-p adjustment.\n",
            "    * 'perm' causes an adaptive permutation test to be performed.\n",
            "    * 'mperm='<value> causes a max(T) permutation test with the specified\n",
            "      number of replications to be performed.\n",
            "    * 'perm-count' causes the permutation test report to include counts instead\n",
            "      of frequencies.\n",
            "    * 'counts' causes --assoc to report allele counts instead of frequencies.\n",
            "    * 'set-test' tests the significance of variant sets.  Requires permutation;\n",
            "      can be customized with --set-p/--set-r2/--set-max.\n",
            "    * 'dom', 'rec', 'gen', and 'trend' force the corresponding test to be used\n",
            "      as the basis for --model permutation.  (By default, the most significant\n",
            "      result among the allelic, dominant, and recessive tests is used.)\n",
            "    * 'trend-only' causes only the trend test to be performed.\n",
            "    Given a quantitative phenotype, --assoc normally performs a Wald test.\n",
            "    * In this case, the 'qt-means' modifier causes trait means and standard\n",
            "      deviations stratified by genotype to be reported as well.\n",
            "    * 'lin' causes the Lin statistic to be computed, and makes it the basis for\n",
            "      multiple-testing corrections and permutation tests.\n",
            "    Several other flags (most notably, --aperm) can be used to customize the\n",
            "    permutation test.\n",
            "\n",
            "  --mh ['perm' | 'mperm='<value>] ['perm-count'] ['set-test']\n",
            "    (alias: --cmh)\n",
            "  --bd ['perm' | 'perm-bd' | 'mperm='<value>] ['perm-count'] ['set-test']\n",
            "  --mh2\n",
            "  --homog\n",
            "    Given a case/control phenotype and a set of clusters, --mh computes 2x2xK\n",
            "    Cochran-Mantel-Haenszel statistics for each variant, while --bd also\n",
            "    performs the Breslow-Day test for odds ratio homogeneity.  Permutation and\n",
            "    variant set testing based on the CMH (default) or Breslow-Day (when\n",
            "    'perm-bd' is present) statistic are supported.\n",
            "    The following similar analyses are also available:\n",
            "    * --mh2 swaps the roles of case/control status and cluster membership,\n",
            "      performing a phenotype-stratified IxJxK Cochran-Mantel-Haenszel test on\n",
            "      association between cluster assignments and genotypes.\n",
            "    * --homog executes an alternative to the Breslow-Day test, based on\n",
            "      partitioning of the chi-square statistic.\n",
            "\n",
            "  --gxe [covariate index]\n",
            "    Given both a quantitative phenotype and a case/control covariate loaded\n",
            "    with --covar defining two groups, --gxe compares the regression coefficient\n",
            "    derived from considering only members of one group to the regression\n",
            "    coefficient derived from considering only members of the other.  By\n",
            "    default, the first covariate in the --covar file defines the groups; use\n",
            "    e.g. \"--gxe 3\" to base them on the third covariate instead.\n",
            "\n",
            "  --linear ['perm' | 'mperm='<value>] ['perm-count'] ['set-test']\n",
            "           [{genotypic | hethom | dominant | recessive | no-snp}]\n",
            "           ['hide-covar'] [{sex | no-x-sex}] ['interaction'] ['beta']\n",
            "           ['standard-beta'] ['intercept']\n",
            "  --logistic ['perm' | 'mperm='<value>] ['perm-count'] ['set-test']\n",
            "             [{genotypic | hethom | dominant | recessive | no-snp}]\n",
            "             ['hide-covar'] [{sex | no-x-sex}] ['interaction'] ['beta']\n",
            "             ['intercept']\n",
            "    Multi-covariate association analysis on a quantitative (--linear) or\n",
            "    case/control (--logistic) phenotype.  Normally used with --covar.\n",
            "    * 'perm' normally causes an adaptive permutation test to be performed on\n",
            "      the main effect, while 'mperm='<value> starts a max(T) permutation test.\n",
            "    * 'perm-count' causes the permutation test report to include counts instead\n",
            "      of frequencies.\n",
            "    * 'set-test' tests the significance of variant sets.  Requires permutation;\n",
            "      can be customized with --set-p/--set-r2/--set-max.\n",
            "    * The 'genotypic' modifier adds an additive effect/dominance deviation 2df\n",
            "      joint test (0/1/2 and 0/1/0 coding), while 'hethom' uses 0/0/1 and 0/1/0\n",
            "      coding instead.  If permutation is also requested, these modifiers cause\n",
            "      permutation to be based on the joint test.\n",
            "    * 'dominant' and 'recessive' specify a model assuming full dominance or\n",
            "      recessiveness, respectively, for the A1 allele.\n",
            "    * 'no-snp' causes regression to be performed only on the phenotype and the\n",
            "      covariates, without reference to genomic data.  If permutation is also\n",
            "      requested, results are reported for all covariates.\n",
            "    * 'hide-covar' removes covariate-specific lines from the report.\n",
            "    * By default, sex (male = 1, female = 0) is automatically added as a\n",
            "      covariate on X chromosome variants, and nowhere else.  The 'sex' modifier\n",
            "      causes it to be added everywhere, while 'no-x-sex' excludes it.\n",
            "    * 'interaction' adds genotype x covariate interactions to the model.  This\n",
            "      cannot be used with the usual permutation tests; use --tests to define\n",
            "      the permutation test statistic instead.\n",
            "    * 'intercept' causes intercepts to be included in the main report.\n",
            "    * For logistic regressions, the 'beta' modifier causes regression\n",
            "      coefficients instead of odds ratios to be reported.\n",
            "    * With --linear, the 'standard-beta' modifier standardizes the phenotype\n",
            "      and all predictors to zero mean and unit variance before regression.\n",
            "\n",
            "  --dosage <allele dosage file> ['noheader'] ['skip0='<i>] ['skip1='<j>]\n",
            "           ['skip2='<k>] ['dose1'] ['format='<m>] ['Zout']\n",
            "           [{occur | standard-beta}] ['sex'] ['case-control-freqs']\n",
            "  --dosage <list file> list [{sepheader | noheader}] ['skip0='<i>]\n",
            "           ['skip1='<j>] ['skip2='<k>] ['dose1'] ['format='<m>] ['Zout']\n",
            "           [{occur | standard-beta}] ['sex'] ['case-control-freqs']\n",
            "  --write-dosage\n",
            "    Process (possibly gzipped) text files with variant-major allelic dosage\n",
            "    data.  This cannot be used with a regular input fileset; instead, you must\n",
            "    *only* specify a .fam and possibly a .map file, and you can't specify any\n",
            "    other commands.\n",
            "    * PLINK 2.0 will have first-class support for genotype probabilities.  An\n",
            "      equivalent data import flag will be provided then, and --dosage will be\n",
            "      retired.\n",
            "    * By default, --dosage assumes that only one allelic dosage file should be\n",
            "      loaded.  To specify multiple files,\n",
            "      1. create a master list with one entry per line.  There are normally two\n",
            "         supported formats for this list: just a filename per line, or variant\n",
            "         batch numbers in the first column and filenames in the second.\n",
            "      2. Provide the name of that list as the first --dosage parameter.\n",
            "      3. Add the 'list' modifier.\n",
            "    * By default, --dosage assumes the allelic dosage file(s) contain a header\n",
            "      line, which has 'SNP' in column i+1, 'A1' in column i+j+2, 'A2' in column\n",
            "      i+j+3, and sample FID/IIDs starting from column i+j+k+4.  (i/j/k are\n",
            "      normally zero, but can be changed with 'skip0', 'skip1', and 'skip2'\n",
            "      respectively.)  If such a header line is not present,\n",
            "      * when all samples appear in the same order as they do in the .fam file,\n",
            "        you can use the 'noheader' modifier.\n",
            "      * Otherwise, use the 'sepheader' modifier, and append sample ID filenames\n",
            "        to your 'list' file entries.\n",
            "    * The 'format=' modifier lets you specify the number of values used to\n",
            "      represent each dosage.  'format=1' normally indicates a single 0..2 A1\n",
            "      expected count; 'dose1' modifies this to a 0..1 frequency.  'format=2'\n",
            "      (the default) indicates a 0..1 homozygous A1 likelihood followed by a\n",
            "      0..1 het likelihood, while 'format=3' indicates 0..1 hom A1, 0..1 het,\n",
            "      0..1 hom A2.\n",
            "    * 'Zout' causes the output file to be gzipped.\n",
            "    * Normally, an association analysis is performed.  'standard-beta' and\n",
            "      'sex' behave as they are supposed to with --linear/--logistic.\n",
            "      'case-control-freqs' causes case and control allele frequencies to be\n",
            "      reported separately.\n",
            "    * There are three alternate modes which cause the association analysis to\n",
            "      be skipped.\n",
            "      * 'occur' requests a simple variant occurrence report.\n",
            "      * --write-dosage causes a simple merged file matching the 'format'\n",
            "        specification (not including 'dose1') to be generated.\n",
            "      * --score applies a linear scoring system to the dosages.\n",
            "\n",
            "  --lasso <h2 estimate> [min lambda] ['report-zeroes']\n",
            "    Estimate variant effect sizes via LASSO regression.  You must provide an\n",
            "    additive heritability estimate to calibrate the regression.\n",
            "    Note that this method may require a very large sample size (e.g. hundreds\n",
            "    of thousands) to be effective on complex polygenic traits.\n",
            "\n",
            "  --test-missing ['perm' | 'mperm='<value>] ['perm-count'] ['midp']\n",
            "    Check for association between missingness and case/control status, using\n",
            "    Fisher's exact test.  (Het. haploids are treated as missing.)\n",
            "    The 'midp' modifier causes Lancaster's mid-p adjustment to be applied.\n",
            "\n",
            "  --make-perm-pheno <ct>\n",
            "    Generate phenotype permutations and write them to disk, without invoking an\n",
            "    association test.\n",
            "\n",
            "  --tdt [{exact | exact-midp | poo}] ['perm' | 'mperm='<value>] ['perm-count']\n",
            "        [{parentdt1 | parentdt2 | pat | mat}] ['set-test']\n",
            "    Report transmission disequilibrium test statistics, given case/control\n",
            "    phenotypes and pedigree information.\n",
            "    * A Mendel error check is performed before the main tests; offending\n",
            "      genotypes are treated as missing by this analysis.\n",
            "    * By default, the basic TDT p-value is based on a chi-square test unless\n",
            "      you request the exact binomial test with 'exact' or 'exact-midp'.\n",
            "    * 'perm'/'mperm=' requests a family-based adaptive or max(T) permutation\n",
            "      test.  By default, the permutation test statistic is the basic TDT\n",
            "      p-value; 'parentdt1'/'parentdt2' cause parenTDT or combined test\n",
            "      p-values, respectively, to be considered instead.\n",
            "    * 'set-test' tests the significance of variant sets.  This cannot be used\n",
            "      with exact tests for now.\n",
            "    The 'poo' modifier causes a parent-of-origin analysis to be performed\n",
            "    instead, with transmissions from heterozygous fathers and heterozygous\n",
            "    mothers considered separately.\n",
            "    * The parent-of-origin analysis does not currently support exact tests.\n",
            "    * By default, the permutation test statistic is the absolute\n",
            "      parent-of-origin test Z score; 'pat'/'mat' cause paternal or maternal TDT\n",
            "      chi-square statistics, respectively, to be considered instead.\n",
            "\n",
            "  --qfam ['perm' | 'mperm='<value>] ['perm-count'] ['emp-se']\n",
            "  --qfam-parents ['perm' | 'mperm='<value>] ['perm-count'] ['emp-se']\n",
            "  --qfam-between ['perm' | 'mperm='<value>] ['perm-count'] ['emp-se']\n",
            "  --qfam-total ['perm' | 'mperm='<value>] ['perm-count'] ['emp-se']\n",
            "    QFAM family-based association test for quantitative traits.\n",
            "    * A Mendel error check is performed before the main tests; offending\n",
            "      genotypes are treated as missing by this analysis.\n",
            "    * This procedure requires permutation.  'perm' and 'perm-count' have the\n",
            "      usual meanings.  However, 'mperm='<value> just specifies a fixed number\n",
            "      of permutations; the method does not support a proper max(T) test.\n",
            "    * The 'emp-se' modifier adds BETA and EMP_SE (empirical standard error for\n",
            "      beta) fields to the .perm output file.\n",
            "\n",
            "  --annotate <PLINK report> ['attrib='<file>] ['ranges='<file>]\n",
            "             ['filter='<file>] ['snps='<file>] [{NA | prune}] ['block']\n",
            "             ['subset='<file>] ['minimal'] ['distance']\n",
            "    Add annotations to a variant-based PLINK report.  This requires an\n",
            "    annotation source:\n",
            "    * 'attrib='<file> specifies a (possibly gzipped) attribute file.\n",
            "    * 'ranges='<file> specifies a gene/range list file.\n",
            "    (Both source types can be specified simultaneously.)  The following options\n",
            "    are also supported:\n",
            "    * 'filter='<file> causes only variants within one of the ranges in the file\n",
            "      to be included in the new report.\n",
            "    * 'snps='<file> causes only variants named in the file to be included in\n",
            "      the new report.\n",
            "    * The 'NA' modifier causes unannotated variants to have 'NA' instead of '.'\n",
            "      in the new report's ANNOT column, while the 'prune' modifier excludes\n",
            "      them entirely.\n",
            "    * The 'block' modifier replaces the single ANNOT column with a 0/1-coded\n",
            "      column for each possible annotation.\n",
            "    * With 'ranges',\n",
            "      * 'subset='<file> causes only intervals named in the subset file to be\n",
            "        loaded from the ranges file.\n",
            "      * interval annotations normally come with a parenthesized signed distance\n",
            "        to the interval boundary (0 if the variant is located inside the\n",
            "        interval; this is always true without --border).  They can be excluded\n",
            "        with the 'minimal' modifier.\n",
            "      * the 'distance' modifier adds 'DIST' and 'SGN' columns describing signed\n",
            "        distance to the nearest interval.\n",
            "    * When --pfilter is present, high p-values are filtered out.\n",
            "\n",
            "  --clump <PLINK report filename(s)...>\n",
            "    Process association analysis report(s) with 'SNP' and p-value columns,\n",
            "    organizing results by LD-based clumps.  Multiple filenames can be separated\n",
            "    by spaces or commas.\n",
            "\n",
            "  --gene-report <PLINK report> <gene range file>\n",
            "    Generate a gene-based report from a variant-based report.\n",
            "    * When --pfilter is present, high p-values are filtered out.\n",
            "    * When --extract (without 'range') is present, only variants named in the\n",
            "      --extract file are considered.\n",
            "\n",
            "  --meta-analysis <PLINK report filenames...>\n",
            "  --meta-analysis <PLINK report filenames...> + [{logscale | qt}]\n",
            "                  [{no-map | no-allele}] ['study'] ['report-all']\n",
            "                  ['weighted-z']\n",
            "    Perform a meta-analysis on several variant-based reports with 'SNP' and\n",
            "    'SE' fields.\n",
            "    * Normally, an 'OR' odds ratio field must also be present in each input\n",
            "      file.  With 'logscale', 'BETA' log-odds values/regression coefficients\n",
            "      are expected instead, but the generated report will still contain odds\n",
            "      ratio estimates.  With 'qt', both input and output values are regression\n",
            "      betas.\n",
            "    * 'CHR', 'BP', and 'A1' fields are also normally required.  'no-map' causes\n",
            "      them to all be ignored, while 'no-allele' causes just 'A1' to be ignored.\n",
            "    * If 'A2' fields are present, and neither 'no-map' nor 'no-allele' was\n",
            "      specified, A1/A2 allele flips are handled properly.  Otherwise, A1\n",
            "      mismatches are thrown out.\n",
            "    * 'study' causes study-specific effect estimates to be collated in the\n",
            "      meta-analysis report.\n",
            "    * 'report-all' causes variants present in only a single input file to be\n",
            "      included in the meta-analysis report.\n",
            "    * 'weighted-z' requests weighted Z-score-based p-values (as computed by the\n",
            "      Abecasis Lab's METAL software) in addition to the usual inverse\n",
            "      variance-based analysis.  This requires P and effective sample size\n",
            "      fields.\n",
            "    * When --extract (without 'range') is present, only variants named in the\n",
            "      --extract file are considered.\n",
            "    * Unless 'no-map' is specified, chromosome filters are also respected.\n",
            "\n",
            "  --fast-epistasis [{boost | joint-effects | no-ueki}] ['case-only']\n",
            "                   [{set-by-set | set-by-all}] ['nop']\n",
            "  --epistasis [{set-by-set | set-by-all}]\n",
            "    Scan for epistatic interactions.  --fast-epistasis inspects 3x3 joint\n",
            "    genotype count tables and only applies to case/control phenotypes, while\n",
            "    --epistasis performs linear or logistic regression.\n",
            "    * By default, --fast-epistasis uses the PLINK 1.07 allele-based test.  Two\n",
            "      newer tests are now supported: 'boost' invokes the likelihood ratio test\n",
            "      introduced by Wan X et al. (2010) BOOST: A Fast Approach to Detecting\n",
            "      Gene-Gene Interactions in Genome-wide Case-Control Studies, while\n",
            "      'joint-effects' applies the joint effects test introduced in Ueki M,\n",
            "      Cordell HJ (2012) Improved statistics for genome-wide interaction\n",
            "      analysis.\n",
            "    * The original --fast-epistasis test normally applies the variance and\n",
            "      empty cell corrections suggested by Ueki and Cordell's paper.  To disable\n",
            "      them, use the 'no-ueki' modifier.\n",
            "    * 'case-only' requests a case-only instead of a case/control test.\n",
            "    * By default, all pairs of variants across the entire genome are tested.\n",
            "      To just test pairs of variants within a single set, add the 'set-by-set'\n",
            "      modifier and load exactly one set with --set/--make-set; with exactly two\n",
            "      sets loaded, all variants in one set are tested against all variants in\n",
            "      the other.  'set-by-all' tests all variants in one set against the entire\n",
            "      genome instead.\n",
            "    * 'nop' strips p-values from the main report.\n",
            "    * These computations can be subdivided with --parallel; however...\n",
            "  --epistasis-summary-merge <common file prefix> <ct>\n",
            "    When a --[fast-]epistasis job is subdivided with --parallel, the main\n",
            "    report can be assembled at the end by applying Unix 'cat' in the usual\n",
            "    manner, but the .summary.1, .summary.2, ... files may require a specialized\n",
            "    merge.  --epistasis-summary-merge takes care of the latter.\n",
            "\n",
            "  --twolocus <variant ID> <variant ID>\n",
            "    Two-locus joint genotype count report.\n",
            "\n",
            "  --score <filename> [i] [j] [k] ['header'] [{sum | no-sum}]\n",
            "          [{no-mean-imputation | center}] ['include-cnt'] ['double-dosage']\n",
            "    Apply a linear scoring system to each sample.\n",
            "    The input file should have one line per scored variant.  Variant IDs are\n",
            "    read from column #i, allele codes are read from column #j, and scores are\n",
            "    read from column #k, where i defaults to 1, j defaults to i+1, and k\n",
            "    defaults to j+1.\n",
            "    * The 'header' modifier causes the first nonempty line of the input file to\n",
            "      be ignored; otherwise, --score assumes there is no header line.\n",
            "    * By default, final scores are averages of the valid per-variant scores.\n",
            "      The 'sum' modifier causes sums to be reported instead.  (This cannot be\n",
            "      used with 'no-mean-imputation'.  And for backward compatibility, 'sum' is\n",
            "      automatically on with dosage data unless 'no-sum' is specified.)\n",
            "    * By default, copies of the unnamed allele contribute zero to score, while\n",
            "      missing genotypes contribute an amount proportional to the loaded (via\n",
            "      --read-freq) or imputed allele frequency.  To throw out missing\n",
            "      observations instead (decreasing the denominator in the final average\n",
            "      when this happens), use the 'no-mean-imputation' modifier.\n",
            "    * Alternatively, you can use the 'center' modifier to shift all scores to\n",
            "      mean zero.\n",
            "    * This command can be used with dosage data.  By default, the 'CNT' column\n",
            "      is omitted from the output file in this case; use 'include-cnt' to keep\n",
            "      it.  Also, note that scores are multiplied by 0..1 dosages, not 0..2\n",
            "      diploid allele counts, unless the 'double-dosage' modifier is present.\n",
            "\n",
            "  --R <R script file> ['debug']\n",
            "    Connect to a Rserve (preferably version 1.7 or later) background process,\n",
            "    and execute the Rplink function defined in the input file.  (Unless the\n",
            "    'debug' modifier is present; in that case, the R commands that PLINK would\n",
            "    have tried to execute are logged to a file.)\n",
            "\n",
            "  --write-var-ranges <block ct>\n",
            "    Divide the set of variants into equal-size blocks.  (Can be used with\n",
            "    --snps to split a job across multiple machines.)\n",
            "\n",
            "The following other flags are supported.  (Order of operations is described at\n",
            "https://www.cog-genomics.org/plink/1.9/order .)\n",
            "  --script <fname> : Include command-line options from file.\n",
            "  --rerun [log]    : Rerun commands in log (default 'plink.log').\n",
            "  --version        : Display only version number before exiting.\n",
            "  --silent         : Suppress output to console.\n",
            "  --gplink         : Reserved for interoperation with gPLINK.\n",
            "  --missing-genotype <char> : Set missing genotype code (normally '0').\n",
            "  --double-id          : Set both FIDs and IIDs to the VCF/BCF sample ID.\n",
            "  --const-fid [ID]     : Set all FIDs to the given constant (default '0').\n",
            "  --id-delim [d]       : Parse sample IDs as <FID><d><IID> (default delim '_').\n",
            "  --vcf-idspace-to <c> : Convert spaces in sample IDs to the given character.\n",
            "  --biallelic-only ['strict'] ['list'] : Skip VCF variants with 2+ ALT alleles.\n",
            "  --vcf-min-qual <val>           : Skip VCF variants with low/missing QUAL.\n",
            "  --vcf-filter [exception(s)...] : Skip variants which have FILTER failures.\n",
            "  --vcf-require-gt               : Skip variants with no GT field.\n",
            "  --vcf-min-gq <val>             : No-call a genotype when GQ is below the\n",
            "                                   given threshold.\n",
            "  --vcf-min-gp <val>             : No-call a genotype when 0-1 scaled GP is\n",
            "                                   below the given threshold.\n",
            "  --vcf-half-call <m>  : Specify how '0/.' and similar VCF GT values should be\n",
            "                         handled.  The following four modes are supported:\n",
            "                         * 'error'/'e' (default) errors out and reports line #.\n",
            "                         * 'haploid'/'h' treats them as haploid calls.\n",
            "                         * 'missing'/'m' treats them as missing.\n",
            "                         * 'reference'/'r' treats the missing value as 0.\n",
            "  --oxford-single-chr <chr nm> : Specify single-chromosome .gen file with\n",
            "                                 ignorable first column.\n",
            "  --oxford-pheno-name <col nm> : Import named phenotype from the .sample file.\n",
            "  --hard-call-threshold <val>  : When an Oxford-format fileset is loaded, calls\n",
            "  --hard-call-threshold random   with uncertainty level greater than 0.1 are\n",
            "                                 normally treated as missing.  You can adjust\n",
            "                                 this threshold by providing a numeric\n",
            "                                 parameter, or randomize all calls with\n",
            "                                 'random'.\n",
            "  --missing-code [string list] : Comma-delimited list of missing phenotype\n",
            "    (alias: --missing_code)      values for Oxford-format filesets (def. 'NA').\n",
            "  --simulate-ncases <num>   : Set --simulate case count (default 1000).\n",
            "  --simulate-ncontrols <n>  : Set --simulate control count (default 1000).\n",
            "  --simulate-prevalence <p> : Set --simulate disease prevalence (default 0.01).\n",
            "  --simulate-n <num>        : Set --simulate-qt sample count (default 1000).\n",
            "  --simulate-label <prefix> : Set --simulate[-qt] FID/IID name prefix.\n",
            "  --simulate-missing <freq> : Set --simulate[-qt] missing genotype frequency.\n",
            "  --allow-extra-chr ['0']   : Permit unrecognized chromosome codes.  The '0'\n",
            "    (alias: --aec)            modifier causes them to be treated as if they had\n",
            "                              been set to zero.\n",
            "  --chr-set <autosome ct> ['no-x'] ['no-y'] ['no-xy'] ['no-mt'] :\n",
            "    Specify a nonhuman chromosome set.  The first parameter sets the number of\n",
            "    diploid autosome pairs if positive, or haploid chromosomes if negative.\n",
            "    Given diploid autosomes, the remaining modifiers indicate the absence of\n",
            "    the named non-autosomal chromosomes.\n",
            "  --cow/--dog/--horse/--mouse/--rice/--sheep : Shortcuts for those species.\n",
            "  --autosome-num <value>    : Alias for \"--chr-set <value> no-y no-xy no-mt\".\n",
            "  --cm-map <fname pattern> [chr] : Use SHAPEIT-format recombination maps to set\n",
            "                                   centimorgan positions.  To process more than\n",
            "                                   one chromosome, include a '@' in the first\n",
            "                                   parameter where the chrom. number belongs,\n",
            "                                   e.g. 'genetic_map_chr@_combined_b37.txt'.\n",
            "  --zero-cms         : Zero out centimorgan positions.\n",
            "  --allow-no-samples : Allow the input fileset to contain no samples.\n",
            "  --allow-no-vars    : Allow the input fileset to contain no variants.\n",
            "  --pheno <fname>  : Load phenotype data from the specified file, instead of\n",
            "                     using the values in the main input fileset.\n",
            "  --all-pheno      : For basic association tests, loop through all phenotypes\n",
            "                     in --pheno file.\n",
            "  --mpheno <n>     : Load phenotype from column (n+2) in --pheno file.\n",
            "  --pheno-name <c> : If --pheno file has a header row, use column with the\n",
            "                     given name.\n",
            "  --pheno-merge    : When the main input fileset contains an phenotype value\n",
            "                     for a sample, but the --pheno file does not, use the\n",
            "                     original value instead of treating the phenotype as\n",
            "                     missing.\n",
            "  --missing-phenotype <v> : Set missing phenotype value (normally -9).\n",
            "  --1                     : Expect case/control phenotypes to be coded as\n",
            "                            0 = control, 1 = case, instead of the usual\n",
            "                            0 = missing, 1 = control, 2 = case.  This also\n",
            "                            forces phenotypes to be interpreted as case/ctrl.\n",
            "  --make-pheno <fn> <val> : Define a new case/control phenotype.  If the val\n",
            "                            parameter is '*', all samples listed in the given\n",
            "                            file are cases, and everyone else is a control.\n",
            "                            (Note that, in some shells, it is necessary to\n",
            "                            surround the * with quotes.)\n",
            "                            Otherwise, all samples with third column entry\n",
            "                            equal to the val parameter are cases, and all other\n",
            "                            samples mentioned in the file are controls.\n",
            "  --tail-pheno <Lt> [Hbt] : Downcode a scalar phenotype to a case/control\n",
            "                            phenotype.  All samples with phenotype values\n",
            "                            greater than Hbt are cases, and all with values\n",
            "                            less than or equal to Lt are controls.  If Hbt is\n",
            "                            unspecified, it is equal to Lt; otherwise,\n",
            "                            in-between phenotype values are set to missing.\n",
            "  --covar <filename> ['keep-pheno-on-missing-cov'] : Specify covariate file.\n",
            "  --covar-name <...>       : Specify covariate(s) in --covar file by name.\n",
            "                             Separate multiple names with spaces or commas, and\n",
            "                             use dashes to designate ranges.\n",
            "  --covar-number <...>     : Specify covariate(s) in --covar file by index.\n",
            "  --no-const-covar         : Exclude constant covariates.\n",
            "  --allow-no-covars        : Allow no covariates to be loaded from --covar\n",
            "                             file.\n",
            "  --within <f> ['keep-NA'] : Specify initial cluster assignments.\n",
            "  --mwithin <n>            : Load cluster assignments from column n+2.\n",
            "  --family                 : Create a cluster for each family ID.\n",
            "  --loop-assoc <f> ['keep-NA']  : Run specified case/control association\n",
            "                                  commands once for each cluster in the file,\n",
            "                                  using cluster membership as the phenotype.\n",
            "  --set <filename>              : Load sets from a .set file.\n",
            "  --set-names <name(s)...>      : Load only sets named on the command line.\n",
            "                                  Use spaces to separate multiple names.\n",
            "  --subset <filename>           : Load only sets named in the given text file.\n",
            "  --set-collapse-all <set name> : Merge all sets.\n",
            "  --complement-sets             : Invert all sets.  (Names gain 'C_' prefixes.)\n",
            "  --make-set-complement-all <s> : --set-collapse-all + inversion.\n",
            "  --make-set <filename>         : Define sets from a list of named bp ranges.\n",
            "  --make-set-border <kbs>       : Stretch regions in --make-set file.\n",
            "  --make-set-collapse-group     : Define sets from groups instead of sets in\n",
            "                                  --make-set file.\n",
            "  --keep <filename>       : Exclude all samples not named in the file.\n",
            "  --remove <filename>     : Exclude all samples named in the file.\n",
            "  --keep-fam <filename>   : Exclude all families not named in the file.\n",
            "  --remove-fam <filename> : Exclude all families named in the file.\n",
            "  --extract ['range'] <f> : Exclude all variants not named in the file.\n",
            "  --exclude ['range'] <f> : Exclude all variants named in the file.\n",
            "  --keep-clusters <filename>          : These can be used individually or in\n",
            "  --keep-cluster-names <name(s)...>     combination to define a list of\n",
            "                                        clusters to keep; all samples not in a\n",
            "                                        cluster in that list are then excluded.\n",
            "                                        Use spaces to separate cluster names\n",
            "                                        for --keep-cluster-names.\n",
            "  --remove-clusters <filename>        : Exclude all clusters named in the file.\n",
            "  --remove-cluster-names <name(s)...> : Exclude the named clusters.\n",
            "  --gene <sets...> : Exclude variants not in a set named on the command line.\n",
            "                     (Separate multiple set names with spaces.)\n",
            "  --gene-all       : Exclude variants which aren't a member of any set.  (PLINK\n",
            "                     1.07 automatically did this under some circumstances.)\n",
            "  --attrib <f> [att lst] : Given a file assigning attributes to variants, and a\n",
            "  --attrib-indiv <f> [a]   comma-delimited list (with no whitespace) of\n",
            "                           attribute names, remove variants/samples which are\n",
            "                           either missing from the file or don't have any of\n",
            "                           the listed attributes.  If some attribute names in\n",
            "                           the list are preceded by '-', they are treated as\n",
            "                           \"negative match conditions\" instead: variants with\n",
            "                           at least one negative match attribute are removed.\n",
            "                           The first character in the list cannot be a '-', due\n",
            "                           to how command-line parsing works; add a comma in\n",
            "                           front to get around this.\n",
            "  --chr <chrs...>  : Exclude all variants not on the given chromosome(s).\n",
            "                     Valid choices for humans are 0 (unplaced), 1-22, X, Y, XY,\n",
            "                     and MT.  Separate multiple chromosomes with spaces and/or\n",
            "                     commas, and use a dash (no adjacent spaces permitted) to\n",
            "                     denote a range, e.g. \"--chr 1-4, 22, xy\".\n",
            "  --not-chr <...>  : Reverse of --chr (exclude variants on listed chromosomes).\n",
            "  --autosome       : Exclude all non-autosomal variants.\n",
            "  --autosome-xy    : Exclude all non-autosomal variants, except those with\n",
            "                     chromosome code XY (pseudo-autosomal region of X).\n",
            "  --snps-only ['just-acgt'] : Exclude non-SNP variants.  By default, SNP = both\n",
            "                              allele codes are single-character; 'just-acgt'\n",
            "                              restricts codes to {A,C,G,T,a,c,g,t,<missing>}.\n",
            "  --from <var ID>  : Use ID(s) to specify a variant range to load.  When used\n",
            "  --to   <var ID>    together, both variants must be on the same chromosome.\n",
            "  --snp  <var ID>  : Specify a single variant to load.\n",
            "  --exclude-snp <> : Specify a single variant to exclude.\n",
            "  --window  <kbs>  : With --snp or --exclude-snp, loads/excludes all variants\n",
            "                     within half the specified kb distance of the named one.\n",
            "  --from-bp <pos>  : Use physical position(s) to define a variant range to\n",
            "  --to-bp   <pos>    load.  --from-kb/--to-kb/--from-mb/--to-mb allow decimal\n",
            "  --from-kb <pos>    values.  You must also specify a single chromosome (using\n",
            "  --to-kb   <pos>    e.g. --chr) when using these flags.\n",
            "  --from-mb <pos>\n",
            "  --to-mb   <pos>\n",
            "  --snps <var IDs...>  : Use IDs to specify variant range(s) to load or\n",
            "  --exclude-snps <...>   exclude.  E.g. \"--snps rs1111-rs2222, rs3333, rs4444\".\n",
            "  --thin <p>       : Randomly remove variants, retaining each with prob. p.\n",
            "  --thin-count <n> : Randomly remove variants until n of them remain.\n",
            "  --bp-space <bps> : Remove variants so that each pair is no closer than the\n",
            "                     given bp distance.  (Equivalent to VCFtools --thin.)\n",
            "  --thin-indiv <p>         : Randomly remove samples, retaining with prob. p.\n",
            "  --thin-indiv-count <n>   : Randomly remove samples until n of them remain.\n",
            "  --filter <f> <val(s)...> : Exclude all samples without a 3rd column entry in\n",
            "                             the given file matching one of the given\n",
            "                             space-separated value(s).\n",
            "  --mfilter <n>            : Match against (n+2)th column instead.\n",
            "  --geno [val]     : Exclude variants with missing call frequencies greater\n",
            "                     than a threshold (default 0.1).  (Note that the default\n",
            "                     threshold is only applied if --geno is invoked without a\n",
            "                     parameter; when --geno is not invoked, no per-variant\n",
            "                     missing call frequency ceiling is enforced at all.  Other\n",
            "                     inclusion/exclusion default thresholds work the same way.)\n",
            "  --mind [val]     : Exclude samples with missing call frequencies greater than\n",
            "                     a threshold (default 0.1).\n",
            "  --oblig-missing <f1> <f2> : Specify blocks of missing genotype calls for\n",
            "                              --geno/--mind to ignore.  The first file should\n",
            "                              have variant IDs in the first column and block\n",
            "                              IDs in the second, while the second file should\n",
            "                              have FIDs in the first column, IIDs in the\n",
            "                              second, and block IDs in the third.\n",
            "  --prune             : Remove samples with missing phenotypes.\n",
            "  --maf [freq]        : Exclude variants with minor allele frequency lower than\n",
            "                        a threshold (default 0.01).\n",
            "  --max-maf <freq>    : Exclude variants with MAF greater than the threshold.\n",
            "  --mac <ct>          : Exclude variants with minor allele count lower than the\n",
            "    (alias: --min-ac)   given threshold.\n",
            "  --max-mac <ct>      : Exclude variants with minor allele count greater than\n",
            "    (alias: --max-ac)   the given threshold.\n",
            "  --maf-succ       : Rule of succession MAF estimation (used in EIGENSOFT).\n",
            "                     Given j observations of one allele and k >= j observations\n",
            "                     of the other, infer a MAF of (j+1) / (j+k+2), rather than\n",
            "                     the default j / (j+k).\n",
            "  --read-freq <fn> : Estimate MAFs and heterozygote frequencies from the given\n",
            "                     --freq[x] report, instead of the input fileset.\n",
            "  --hwe <p> ['midp'] ['include-nonctrl'] : Exclude variants with Hardy-Weinberg\n",
            "                                           equilibrium exact test p-values\n",
            "                                           below a threshold.\n",
            "  --me <t> <v> ['var-first'] : Filter out trios and variants with Mendel error\n",
            "                               rates exceeding the given thresholds.\n",
            "  --me-exclude-one [ratio]   : Make --me exclude only one sample per trio.\n",
            "  --qual-scores <f> [qcol] [IDcol] [skip] : Filter out variants with\n",
            "                                            out-of-range quality scores.\n",
            "                                            Default range is now [0, \\infty ).\n",
            "  --qual-threshold <min qual score>       : Set --qual-scores range floor.\n",
            "  --qual-max-threshold <max qual score>   : Set --qual-scores range ceiling.\n",
            "  --allow-no-sex   : Do not treat ambiguous-sex samples as having missing\n",
            "                     phenotypes in analysis commands.  (Automatic /w --no-sex.)\n",
            "  --must-have-sex  : Force ambiguous-sex phenotypes to missing on\n",
            "                     --make-bed/--make-just-fam/--recode/--write-covar.\n",
            "  --filter-cases       : Include only cases in the current analysis.\n",
            "  --filter-controls    : Include only controls.\n",
            "  --filter-males       : Include only males.\n",
            "  --filter-females     : Include only females.\n",
            "  --filter-founders    : Include only founders.\n",
            "  --filter-nonfounders : Include only nonfounders.\n",
            "  --nonfounders        : Include nonfounders in allele freq/HWE calculations.\n",
            "  --make-founders ['require-2-missing'] ['first'] :\n",
            "    Clear parental IDs for those with 1+ missing parent(s).\n",
            "  --recode-allele <fn> : With --recode A/A-transpose/AD, count alleles named in\n",
            "                         the file (otherwise A1 alleles are always counted).\n",
            "  --output-chr <MT code> : Set chromosome coding scheme in output files by\n",
            "                           providing the desired human mitochondrial code.\n",
            "                           (Options are '26', 'M', 'MT', '0M', 'chr26', 'chrM',\n",
            "                           and 'chrMT'.)\n",
            "  --output-missing-genotype <ch> : Set the code used to represent missing\n",
            "                                   genotypes in output files (normally the\n",
            "                                   --missing-genotype value).\n",
            "  --output-missing-phenotype <s> : Set the string used to represent missing\n",
            "                                   phenotypes in output files (normally the\n",
            "                                   --missing-phenotype value).\n",
            "  --zero-cluster <f> : In combination with --within/--family, set blocks of\n",
            "                       genotype calls to missing.  The input file should have\n",
            "                       variant IDs in the first column and cluster IDs in the\n",
            "                       second.  This must now be used with --make-bed and no\n",
            "                       other output commands.\n",
            "  --set-hh-missing       : Cause --make-bed and --recode to set heterozygous\n",
            "                           haploid genotypes to missing.\n",
            "  --set-mixed-mt-missing : Cause --make-bed and --recode to set mixed MT\n",
            "                           genotypes to missing.\n",
            "  --split-x <bp1> <bp2> ['no-fail']\n",
            "  --split-x <build> ['no-fail'] :\n",
            "    Changes chromosome code of all chrX variants with bp position <= bp1 or >=\n",
            "    bp2 to XY.  The following build codes are supported as shorthand:\n",
            "    * 'b36'/'hg18' = NCBI 36, 2709521/154584237\n",
            "    * 'b37'/'hg19' = GRCh37, 2699520/154931044\n",
            "    * 'b38'/'hg38' = GRCh38, 2781479/155701383\n",
            "    By default, PLINK errors out when no variants would be affected by\n",
            "    --split-x; the 'no-fail' modifier (useful in scripts) overrides this.\n",
            "  --merge-x ['no-fail'] : Merge XY chromosome back with X.\n",
            "  --set-me-missing      : Cause --make-bed to set Mendel errors to missing.\n",
            "  --fill-missing-a2     : Cause --make-bed to replace all missing calls with\n",
            "                          homozygous A2 calls.\n",
            "  --set-missing-var-ids <t>   : Given a template string with a '@' where the\n",
            "                                chromosome code should go and '#' where the bp\n",
            "                                coordinate belongs, --set-missing-var-ids\n",
            "                                assigns chromosome-and-bp-based IDs to unnamed\n",
            "                                variants.\n",
            "                                You may also use '$1' and '$2' to refer to\n",
            "                                allele names in the template string, and in\n",
            "                                fact this becomes essential when multiple\n",
            "                                variants share the same coordinate.\n",
            "  --new-id-max-allele-len <n> : Specify maximum number of leading characters\n",
            "                                from allele names to include in new variant IDs\n",
            "                                (default 23).\n",
            "  --missing-var-code <string> : Change unnamed variant code (default '.').\n",
            "  --update-chr  <f> [chrcol] [IDcol]  [skip] : Update variant chromosome codes.\n",
            "  --update-cm   <f> [cmcol]  [IDcol]  [skip] : Update centimorgan positions.\n",
            "  --update-map  <f> [bpcol]  [IDcol]  [skip] : Update variant bp positions.\n",
            "  --update-name <f> [newcol] [oldcol] [skip] : Update variant IDs.\n",
            "  --update-alleles <fname> : Update variant allele codes.\n",
            "  --allele1234 ['multichar'] : Interpret/recode A/C/G/T alleles as 1/2/3/4.\n",
            "                               With 'multichar', converts all A/C/G/Ts in\n",
            "                               allele names to 1/2/3/4s.\n",
            "  --alleleACGT ['multichar'] : Reverse of --allele1234.\n",
            "  --update-ids <f>     : Update sample IDs.\n",
            "  --update-parents <f> : Update parental IDs.\n",
            "  --update-sex <f> [n] : Update sexes.  Sex (1 or M = male, 2 or F = female, 0\n",
            "                         = missing) is loaded from column n+2 (default n is 1).\n",
            "  --flip <filename>    : Flip alleles (A<->T, C<->G) for SNP IDs in the file.\n",
            "  --flip-subset <fn>   : Only apply --flip to samples in --flip-subset file.\n",
            "  --flip-scan-window <ct+1> : Set --flip-scan max variant ct dist. (def. 10).\n",
            "  --flip-scan-window-kb <x> : Set --flip-scan max kb distance (default 1000).\n",
            "  --flip-scan-threshold <x> : Set --flip-scan min correlation (default 0.5).\n",
            "  --keep-allele-order  : Keep the allele order defined in the .bim file,\n",
            "  --real-ref-alleles     instead of forcing A2 to be the major allele.\n",
            "                         --real-ref-alleles also removes 'PR' from the INFO\n",
            "                         values emitted by --recode vcf{,-fid,-iid}.\n",
            "  --a1-allele <f> [a1col] [IDcol] [skip] : Force alleles in the file to A1.\n",
            "  --a2-allele <filename> [a2col] [IDcol] [skip] :\n",
            "    Force alleles in the file to A2.  (\"--a2-allele <VCF filename> 4 3 '#'\",\n",
            "    which scrapes reference allele assignments from a VCF file, is especially\n",
            "    useful.)\n",
            "  --indiv-sort <m> [f] : Specify FID/IID sort order.  The following four modes\n",
            "                         are supported:\n",
            "                         * 'none'/'0' keeps samples in the order they were\n",
            "                           loaded.  Default for non-merge operations.\n",
            "                         * 'natural'/'n' invokes 'natural sort', e.g.\n",
            "                           'id2' < 'ID3' < 'id10'.  Default when merging.\n",
            "                         * 'ascii'/'a' sorts in ASCII order, e.g.\n",
            "                           'ID3' < 'id10' < 'id2'.\n",
            "                         * 'file'/'f' uses the order in the given file (named\n",
            "                           in the second parameter).\n",
            "                         For now, only --merge/--bmerge/--merge-list and\n",
            "                         --make-bed/--make-just-fam respect this flag.\n",
            "  --with-phenotype ['no-parents'] [{no-sex | female-2}] :\n",
            "    Include more sample info in new .cov file.\n",
            "  --dummy-coding [N] ['no-round'] : Split categorical variables (n categories,\n",
            "                                    2 < n <= N, default N is 49) into n-1\n",
            "                                    binary dummy variables when writing\n",
            "                                    covariate file.\n",
            "  --merge-mode <n>   : Adjust --[b]merge/--merge-list behavior based on a\n",
            "                       numeric code.\n",
            "                       1 (default) = ignore missing calls, otherwise difference\n",
            "                                     -> missing\n",
            "                       2 = only overwrite originally missing calls\n",
            "                       3 = only overwrite when nonmissing in new file\n",
            "                       4/5 = never overwrite and always overwrite, respectively\n",
            "                       6 = report all mismatching calls without merging\n",
            "                       7 = report mismatching nonmissing calls without merging\n",
            "  --merge-equal-pos  : With --merge/--bmerge/--merge-list, merge variants with\n",
            "                       different names but identical positions.  (Exception:\n",
            "                       same-position chromosome code 0 variants aren't merged.)\n",
            "  --mendel-duos      : Make Mendel error checks consider samples with only one\n",
            "                       parent in the dataset.\n",
            "  --mendel-multigen  : Make Mendel error checks consider (great-)grandparental\n",
            "                       genotypes when parental genotype data is missing.\n",
            "  --ld-window <ct+1> : Set --r/--r2 max variant ct pairwise distance (usu. 10).\n",
            "  --ld-window-kb <x> : Set --r/--r2 max kb pairwise distance (usually 1000).\n",
            "  --ld-window-cm <x> : Set --r/--r2 max centimorgan pairwise distance.\n",
            "  --ld-window-r2 <x> : Set threshold for --r2 report inclusion (usually 0.2).\n",
            "  --ld-snp <var ID>  : Set first variant in all --r/--r2 pairs.\n",
            "  --ld-snps <vID...> : Restrict first --r/--r2 variant to the given ranges.\n",
            "  --ld-snp-list <f>  : Restrict first --r/--r2 var. to those named in the file.\n",
            "  --list-all         : Generate the 'all' mode report when using --show-tags in\n",
            "                       file mode.\n",
            "  --tag-kb <kbs>     : Set --show-tags max tag kb distance (default 250).\n",
            "  --tag-r2 <val>     : Set --show-tags min tag r-squared (default 0.8)\n",
            "  --tag-mode2        : Use two-column --show-tags (file mode) I/O format.\n",
            "  --ld-xchr <code>   : Set chrX model for --indep[-pairwise], --r/--r2,\n",
            "                       --flip-scan, and --show-tags.\n",
            "                       1 (default) = males coded 0/1, females 0/1/2 (A1 dosage)\n",
            "                       2 = males coded 0/2\n",
            "                       3 = males coded 0/2, but females given double weighting\n",
            "  --blocks-max-kb <kbs>      : Set --blocks maximum haploblock span (def. 200).\n",
            "  --blocks-min-maf <cutoff>  : Adjust --blocks MAF minimum (default 0.05).\n",
            "  --blocks-strong-lowci <x>  : Set --blocks \"strong LD\" CI thresholds (defaults\n",
            "  --blocks-strong-highci <x>   0.70 and 0.98).\n",
            "  --blocks-recomb-highci <x> : Set 'recombination' CI threshold (default 0.90).\n",
            "  --blocks-inform-frac <x>   : Force haploblock <strong LD pairs>:<total\n",
            "                               informative pairs> ratios to be larger than this\n",
            "                               value (default 0.95).\n",
            "  --distance-wts exp=<x>        : When computing genomic distances, assign each\n",
            "                                  variant a weight of (2q(1-q))^{-x}, where q\n",
            "                                  is the loaded or inferred MAF.\n",
            "  --read-dists <dist file> [id file] : Load a triangular binary distance matrix\n",
            "                                       instead of recalculating from scratch.\n",
            "  --ppc-gap <val>    : Minimum number of base pairs, in thousands, between\n",
            "                       informative pairs of markers used in --genome PPC test.\n",
            "                       500 if unspecified.\n",
            "  --min <cutoff>     : Specify minimum PI_HAT for inclusion in --genome report.\n",
            "  --max <cutoff>     : Specify maximum PI_HAT for inclusion in --genome report.\n",
            "  --homozyg-match <> : Set minimum concordance across jointly homozygous\n",
            "                       variants for a pairwise allelic match to be declared.\n",
            "  --pool-size <ct>   : Set minimum size of pools in \"--homozyg group\" report.\n",
            "  --read-genome <fn> : Load --genome report for --cluster/--neighbour, instead\n",
            "                       of recalculating IBS and PPC test p-values from scratch.\n",
            "  --ppc <p-val>    : Specify minimum PPC test p-value within a cluster.\n",
            "  --mc <max size>  : Specify maximum cluster size.\n",
            "  --mcc <c1> <c2>  : Specify maximum case and control counts per cluster.\n",
            "  --K <min count>  : Specify minimum cluster count.\n",
            "  --ibm <val>      : Specify minimum identity-by-missingness.\n",
            "  --match <f> [mv] : Use covariate values to restrict clustering.  Without\n",
            "                     --match-type, two samples can only be in the same cluster\n",
            "                     if all covariates match.  The optional second parameter\n",
            "                     specifies a covariate value to treat as missing.\n",
            "  --match-type <f> : Refine interpretation of --match file.  The --match-type\n",
            "                     file is expected to be a single line with as many entries\n",
            "                     as the --match file has covariates; '0' entries specify\n",
            "                     \"negative matches\" (i.e. samples with equal covariate\n",
            "                     values cannot be in the same cluster), '1' entries specify\n",
            "                     \"positive matches\" (default), and '-1' causes the\n",
            "                     corresponding covariate to be ignored.\n",
            "  --qmatch <f> [m] : Force all members of a cluster to have similar\n",
            "  --qt <fname>       quantitative covariate values.  The --qmatch file contains\n",
            "                     the covariate values, while the --qt file is a list of\n",
            "                     nonnegative tolerances (and '-1's marking covariates to\n",
            "                     skip).\n",
            "  --pca-cluster-names <...> : These can be used individually or in combination\n",
            "  --pca-clusters <fname>      to define a list of clusters to use in the basic\n",
            "                              --pca computation.  (--pca-cluster-names expects\n",
            "                              a space-delimited sequence of cluster names,\n",
            "                              while --pca-clusters expects a file with one\n",
            "                              cluster name per line.)  All samples outside\n",
            "                              those clusters will then be projected on to the\n",
            "                              calculated PCs.\n",
            "  --mds-plot <dims> ['by-cluster'] ['eigendecomp'] ['eigvals'] :\n",
            "    Multidimensional scaling analysis.  Requires --cluster.\n",
            "  --cell <thresh>  : Skip some --model tests when a contingency table entry is\n",
            "                     smaller than the given threshold.\n",
            "  --condition <var ID> [{dominant | recessive}] : Add one variant as a --linear\n",
            "                                                  or --logistic covariate.\n",
            "  --condition-list <f> [{dominant | recessive}] : Add variants named in the\n",
            "                                                  file as --linear/--logistic\n",
            "                                                  covariates.\n",
            "  --parameters <...>  : Include only the given covariates/interactions in the\n",
            "                        --linear/--logistic models, identified by a list of\n",
            "                        1-based indices and/or ranges of them.\n",
            "  --tests <all> [...] : Perform a (joint) test on the specified term(s) in the\n",
            "                        --linear/--logistic model, identified by 1-based\n",
            "                        indices and/or ranges of them.  If permutation was\n",
            "                        requested, it is based on this test.\n",
            "                        * Note that, when --parameters is also present, the\n",
            "                          indices refer to the terms remaining AFTER pruning by\n",
            "                          --parameters.\n",
            "                        * You can use \"--tests all\" to include all terms.\n",
            "  --vif <max VIF>     : Set VIF threshold for --linear multicollinearity check\n",
            "                        (default 50).\n",
            "  --xchr-model <code> : Set the X chromosome --linear/--logistic model.\n",
            "                        0 = skip sex and haploid chromosomes\n",
            "                        1 (default) = add sex as a covariate on X chromosome\n",
            "                        2 = code male genotypes 0/2 instead of 0/1\n",
            "                        3 = test for interaction between genotype and sex\n",
            "  --lasso-select-covars [cov(s)...] : Subject some or all covariates to LASSO\n",
            "                                      model selection.\n",
            "  --adjust ['gc'] ['log10'] ['qq-plot'] : Report some multiple-testing\n",
            "                                          corrections.\n",
            "  --lambda <val>   : Set genomic control lambda for --adjust.\n",
            "  --ci <size>      : Report confidence intervals for odds ratios.\n",
            "  --pfilter <val>  : Filter out association test results with higher p-values.\n",
            "  --aperm <min perms - 1> [max perms] [alpha] [beta] [init interval] [slope] :\n",
            "    Set up to six parameters controlling adaptive permutation tests.\n",
            "    * The first two control the minimum and maximum number of permutations that\n",
            "      may be run for each variant; default values are 5 and 1000000.\n",
            "    * The next two control the early termination condition.  A\n",
            "      100% * (1 - beta/2T) confidence interval is calculated for each empirical\n",
            "      p-value, where T is the total number of variants; whenever this\n",
            "      confidence interval doesn't contain alpha, the variant is exempted from\n",
            "      further permutation testing.  Default values are 0 and 1e-4.\n",
            "    * The last two control when the early termination condition is checked.  If\n",
            "      a check occurs at permutation #p, the next check occurs after\n",
            "      <slope>p + <init interval> more permutations (rounded down).  Default\n",
            "      initial interval is 1, and default slope is 0.001.\n",
            "  --mperm-save     : Save best max(T) permutation test statistics.\n",
            "  --mperm-save-all : Save all max(T) permutation test statistics.\n",
            "  --set-p <p-val>        : Adjust set test significant variant p-value ceiling\n",
            "                           (default 0.05).\n",
            "  --set-r2 [v] ['write'] : Adjust set test significant variant pairwise r^2\n",
            "                           ceiling (default 0.5).  'write' causes violating\n",
            "                          pairs to be dumped to <output prefix>.ldset.\n",
            "  --set-max <ct>         : Adjust set test maximum # of significant variants\n",
            "                           considered per set (default 5).\n",
            "  --set-test-lambda <v>  : Specify genomic control correction for set test.\n",
            "  --border <kbs>            : Extend --annotate range intervals by given # kbs.\n",
            "  --annotate-snp-field <nm> : Set --annotate variant ID field name.\n",
            "  --clump-p1 <pval> : Set --clump index var. p-value ceiling (default 1e-4).\n",
            "  --clump-p2 <pval> : Set --clump secondary p-value threshold (default 0.01).\n",
            "  --clump-r2 <r^2>  : Set --clump r^2 threshold (default 0.5).\n",
            "  --clump-kb <kbs>  : Set --clump kb radius (default 250).\n",
            "  --clump-snp-field <n...>  : Set --clump variant ID field name (default\n",
            "                              'SNP').  With multiple field names, earlier names\n",
            "                              take precedence over later ones.\n",
            "  --clump-field <name...>   : Set --clump p-value field name (default 'P').\n",
            "  --clump-allow-overlap     : Let --clump non-index vars. join multiple clumps.\n",
            "  --clump-verbose           : Request extended --clump report.\n",
            "  --clump-annotate <hdr...> : Include named extra fields in --clump-verbose and\n",
            "                              --clump-best reports.  (Field names can be\n",
            "                              separated with spaces or commas.)\n",
            "  --clump-range <filename>  : Report overlaps between clumps and regions.\n",
            "  --clump-range-border <kb> : Stretch regions in --clump-range file.\n",
            "  --clump-index-first       : Extract --clump index vars. from only first file.\n",
            "  --clump-replicate         : Exclude clumps which contain secondary results\n",
            "                              from only one file.\n",
            "  --clump-best              : Report best proxy for each --clump index var.\n",
            "  --meta-analysis-chr-field <n...> : Set --meta-analysis chromosome, variant\n",
            "  --meta-analysis-snp-field <n...>   ID, position, A1/A2 allele, p-value,\n",
            "  --meta-analysis-bp-field <n...>    standard error, and/or effective sample\n",
            "  --meta-analysis-a1-field <n...>    size field names.\n",
            "  --meta-analysis-a2-field <n...>    Defaults are 'CHR', 'SNP', 'BP', 'A1',\n",
            "  --meta-analysis-p-field <n...>     'A2', 'P', 'SE', and 'NMISS',\n",
            "  --meta-analysis-se-field <n...>    respectively.  When multiple parameters\n",
            "  --meta-analysis-ess-field <n...>   are given to these flags, earlier names\n",
            "                                     take precedence over later ones.\n",
            "                                     Note that, if the numbers of cases and\n",
            "                                     controls are unequal, effective sample\n",
            "                                     size should be\n",
            "                                       4 / (1/<# cases> + 1/<# controls>).\n",
            "  --meta-analysis-report-dups      : When a variant appears multiple times in\n",
            "                                     in the same file, report that.\n",
            "  --gene-list-border <kbs>   : Extend --gene-report regions by given # of kbs.\n",
            "  --gene-subset <filename>   : Specify gene name subset for --gene-report.\n",
            "  --gene-report-snp-field <> : Set --gene-report variant ID field name (default\n",
            "                               'SNP').  Only relevant with --extract.\n",
            "  --gap <kbs>      : Set \"--fast-epistasis case-only\" min. gap (default 1000).\n",
            "  --epi1 <p-value> : Set --[fast-]epistasis reporting threshold (default\n",
            "                     5e-6 for 'boost', 1e-4 otherwise).\n",
            "  --epi2 <p-value> : Set threshold for contributing to SIG_E count (def. 0.01).\n",
            "  --je-cellmin <n> : Set required number of observations per 3x3x2 contingency\n",
            "                     table cell for joint-effects test (default 5).\n",
            "  --q-score-range <range file> <data file> [i] [j] ['header'] :\n",
            "    Apply --score to subset(s) of variants in the primary score list based\n",
            "    on e.g. p-value ranges.\n",
            "    * The first file should have range labels in the first column, p-value\n",
            "      lower bounds in the second column, and upper bounds in the third column.\n",
            "      Lines with too few entries, or nonnumeric values in the second or third\n",
            "      column, are ignored.\n",
            "    * The second file should contain a variant ID and a p-value on each\n",
            "      nonempty line (except possibly the first).  Variant IDs are read from\n",
            "      column #i and p-values are read from column #j, where i defaults to 1 and\n",
            "      j defaults to i+1.  The 'header' modifier causes the first nonempty line\n",
            "      of this file to be skipped.\n",
            "  --R-port <port #>  : Connect to Rserve on a port other than 6311.\n",
            "  --R-host <host>    : Connect to Rserve host.\n",
            "  --R-socket <sock>  : Connect to Rserve socket.\n",
            "  --parallel <k> <n> : Divide the output matrix into n pieces, and only compute\n",
            "                       the kth piece.  The primary output file will have the\n",
            "                       piece number included in its name, e.g. plink.rel.13 or\n",
            "                       plink.rel.13.gz if k is 13.  Concatenating these files\n",
            "                       in order will yield the full matrix of interest.  (Yes,\n",
            "                       this can be done before unzipping.)\n",
            "                       N.B. This generally cannot be used to directly write a\n",
            "                       symmetric square matrix.  Choose square0 or triangle\n",
            "                       shape instead, and postprocess as necessary.\n",
            "  --memory <val>     : Set size, in MB, of initial workspace malloc attempt.\n",
            "                       (Practically mandatory when using GNU parallel.)\n",
            "  --threads <val>    : Set maximum number of concurrent threads.\n",
            "                       This has one known limitation: some BLAS/LAPACK linear\n",
            "                       algebra operations are multithreaded in a way that PLINK\n",
            "                       cannot control.  If this is problematic, you should\n",
            "                       recompile against single-threaded BLAS/LAPACK.\n",
            "  --d <char>         : Change variant/covariate range delimiter (normally '-').\n",
            "  --seed <val...>    : Set random number seed(s).  Each value must be an\n",
            "                       integer between 0 and 4294967295 inclusive.\n",
            "  --perm-batch-size <val> : Set number of permutations per batch for some\n",
            "                            permutation tests.\n",
            "  --output-min-p <p> : Specify minimum p-value to write to reports.\n",
            "  --debug            : Use slower, more crash-resistant logging method.\n",
            "\n",
            "Primary methods paper:\n",
            "Chang CC, Chow CC, Tellier LCAM, Vattikuti S, Purcell SM, Lee JJ (2015)\n",
            "Second-generation PLINK: rising to the challenge of larger and richer datasets.\n",
            "GigaScience, 4.\n",
            "\n",
            "For further documentation and support, consult the main webpage\n",
            "(https://www.cog-genomics.org/plink/1.9 ) and/or the mailing list\n",
            "(https://groups.google.com/d/forum/plink2-users ).\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qu2BIQ5p8Qry"
      },
      "source": [
        "Parabéns!!! Se tudo deu certo você conseguiu instalar o Plink!"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "_AjvLY5P9FXY"
      },
      "source": [
        "## Plink e estudo de associação"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "B-qCmBN59P34"
      },
      "source": [
        "Agora chegou a hora de trabalhar com o plink para resolver um caso de estudo de associação. Temos dois modelos. O modelo 1 e o modelo 2. Siga as instruções para rodar o plink aqui no google colab, e no final teremos um manhatan plot sobre o resultado final.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "078QwRFD_W_V"
      },
      "source": [],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "EHzWdEQd940z"
      },
      "source": [
        "---\n",
        "\n",
        "### Estudo de caso:\n",
        "\n",
        "No interior do Rio Grande do Norte uma familia apresentou indivíduos com manifestação clínica de traços de transtorno bipolar. A familia vive em uma comunidade pobre e em uma cidade com poucos habitantes. Todos os idosos da cidade (fundadores) foram genotipados e estão servindo como controle, enquanto os casos podem variar de acordo com a hipótese. Nesse cenário criei duas hipóteses que poderiam explicar os casos de transtorno bipolar, codifiquei de forma diferente as amostras.\n",
        "\n"
      ]
    }
  ]
}